{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# pySCENIC protocol: PBMC10k\n",
    "\n",
    "September 2019\n",
    "\n",
    "Dataset: *10k PBMCs from a Healthy Donor* available from 10x Genomics ([here](https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.0.0/pbmc_10k_v3))."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# import dependencies\n",
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import scanpy as sc\n",
    "import loompy as lp\n",
    "from MulticoreTSNE import MulticoreTSNE as TSNE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SCENIC steps"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 1: Gene regulatory network inference, and generation of co-expression modules\n",
    "#### Phase Ia: GRN inference using the GRNBoost2 algorithm\n",
    "\n",
    "For this step the CLI version of SCENIC is used. This step can be deployed on an High Performance Computing system. We use the counts matrix (without log transformation or further processing) from the loom file we wrote earlier.\n",
    "_Output:_ List of adjacencies between a TF and its targets stored in `ADJACENCIES_FNAME`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# transcription factors list\n",
    "\n",
    "f_loom_path_scenic = \"/node210data/gyeongdaekim/pyscenic/cnu_lung/cd8.loom\"\n",
    "f_tfs = \"/node210data/gyeongdaekim/pyscenic/hs_hgnc_tfs.txt\"   # human\n",
    "f_out_file = \"/node210data/gyeongdaekim/pyscenic/cnu_lung/cd8.tsv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2024-03-13 19:43:33,635 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.\n",
      "\n",
      "2024-03-13 19:44:10,739 - pyscenic.cli.pyscenic - INFO - Inferring regulatory networks.\n",
      "/home/gyeongdaekim/anaconda3/envs/pyscenic/lib/python3.7/site-packages/distributed/node.py:161: UserWarning: Port 8787 is already in use.\n",
      "Perhaps you already have a cluster running?\n",
      "Hosting the HTTP server on port 38519 instead\n",
      "  f\"Port {expected} is already in use.\\n\"\n",
      "Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.\n",
      "Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.\n",
      "Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.\n",
      "Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.\n",
      "Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.\n",
      "Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.\n",
      "Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.\n",
      "Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.\n",
      "Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.\n",
      "Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.\n",
      "preparing dask client\n",
      "parsing input\n",
      "creating dask graph\n",
      "9 partitions\n",
      "computing dask graph\n"
     ]
    }
   ],
   "source": [
    "!pyscenic grn {f_loom_path_scenic} {f_tfs} -o {f_out_file} --num_workers 9"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "read in the adjacencies matrix:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "adjacencies = pd.read_csv(\"/node210data/gyeongdaekim/pyscenic/thyroid/new_M1_M2/m2_cd163_po.tsv\", index_col=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>target</th>\n",
       "      <th>importance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RPL35</td>\n",
       "      <td>RPL36</td>\n",
       "      <td>211.385697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>YBX1</td>\n",
       "      <td>FTL</td>\n",
       "      <td>210.755757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RPS4X</td>\n",
       "      <td>RPL26</td>\n",
       "      <td>209.175938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YBX1</td>\n",
       "      <td>SH3BGRL3</td>\n",
       "      <td>204.886261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RPL35</td>\n",
       "      <td>RPL36A</td>\n",
       "      <td>199.054813</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      TF    target  importance\n",
       "0  RPL35     RPL36  211.385697\n",
       "1   YBX1       FTL  210.755757\n",
       "2  RPS4X     RPL26  209.175938\n",
       "3   YBX1  SH3BGRL3  204.886261\n",
       "4  RPL35    RPL36A  199.054813"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adjacencies.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 2-3: Regulon prediction aka cisTarget from CLI\n",
    "\n",
    "For this step the CLI version of SCENIC is used. This step can be deployed on an High Performance Computing system.\n",
    "\n",
    "_Output:_ List of adjacencies between a TF and its targets stored in `MOTIFS_FNAME`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "locations for ranking databases, and motif annotations:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "# ranking databases\n",
    "f_db_glob = \"/node210data/gyeongdaekim/pyscenic/rcistarget/*.feather\"\n",
    "f_db_names = ' '.join( glob.glob(f_db_glob) )\n",
    "\n",
    "# motif databases\n",
    "f_motif_path = \"/node210data/gyeongdaekim/pyscenic/rcistarget/motifs-v9-nr.hgnc-m0.001-o0.0.tbl\"\n",
    "\n",
    "\n",
    "# loom file directory\n",
    "# loom file directory\n",
    "f_loom_path_scenic = \"/node210data/gyeongdaekim/pyscenic/thyroid/new_M1_M2/m2_cd163_po.loom\"\n",
    "\n",
    "f_out_adj_path = \"/node210data/gyeongdaekim/pyscenic/thyroid/new_M1_M2/m2_cd163_po.tsv\"\n",
    "\n",
    "f_out_reg_path = \"/node210data/gyeongdaekim/pyscenic/thyroid/new_M1_M2/m2_cd163_po_reg.csv\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here, we use the `--mask_dropouts` option, which affects how the correlation between TF and target genes is calculated during module creation. It is important to note that prior to pySCENIC v0.9.18, the default behavior was to mask dropouts, while in v0.9.18 and later, the correlation is performed using the entire set of cells (including those with zero expression). When using the `modules_from_adjacencies` function directly in python instead of via the command line, the `rho_mask_dropouts` option can be used to control this."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/gyeongdaekim/anaconda3/lib/python3.7/site-packages/dask/dataframe/utils.py:14: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n",
      "\n",
      "2022-10-06 22:52:55,112 - pyscenic.cli.pyscenic - INFO - Creating modules.\n",
      "\n",
      "2022-10-06 22:52:56,874 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.\n",
      "\n",
      "2022-10-06 22:53:18,300 - pyscenic.utils - INFO - Calculating Pearson correlations.\n",
      "\n",
      "2022-10-06 22:53:19,295 - pyscenic.utils - WARNING - Note on correlation calculation: the default behaviour for calculating the correlations has changed after pySCENIC verion 0.9.16. Previously, the default was to calculate the correlation between a TF and target gene using only cells with non-zero expression values (mask_dropouts=True). The current default is now to use all cells to match the behavior of the R verision of SCENIC. The original settings can be retained by setting 'rho_mask_dropouts=True' in the modules_from_adjacencies function, or '--mask_dropouts' from the CLI.\n",
      "\tDropout masking is currently set to [True].\n",
      "\n",
      "2022-10-06 22:53:55,209 - pyscenic.utils - INFO - Creating modules.\n",
      "\n",
      "2022-10-06 22:55:40,250 - pyscenic.cli.pyscenic - INFO - Loading databases.\n",
      "\n",
      "2022-10-06 22:55:40,250 - pyscenic.cli.pyscenic - INFO - Calculating regulons.\n",
      "[                                        ] | 0% Completed | 23.7s\n",
      "2022-10-06 22:56:06,261 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF354C could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 35.6s\n",
      "2022-10-06 22:56:18,069 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for FOXS1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[                                        ] | 2% Completed | 54.0s\n",
      "2022-10-06 22:56:36,569 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for AGAP2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[                                        ] | 2% Completed | 54.3s\n",
      "2022-10-06 22:56:36,844 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for AHCTF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[                                        ] | 2% Completed |  1min  1.5s\n",
      "2022-10-06 22:56:44,075 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for E4F1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[                                        ] | 2% Completed |  1min  8.2s\n",
      "2022-10-06 22:56:50,765 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for CBX7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##                                      ] | 6% Completed |  1min 37.7s\n",
      "2022-10-06 22:57:20,268 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for POU6F1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##                                      ] | 6% Completed |  1min 41.4s\n",
      "2022-10-06 22:57:24,051 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for PTCD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##                                      ] | 6% Completed |  1min 46.6s\n",
      "2022-10-06 22:57:29,097 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for THRA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###                                     ] | 8% Completed |  1min 55.3s\n",
      "2022-10-06 22:57:37,832 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for DIABLO could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###                                     ] | 8% Completed |  1min 57.8s\n",
      "2022-10-06 22:57:40,386 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZFP82 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###                                     ] | 8% Completed |  2min  0.9s\n",
      "2022-10-06 22:57:43,455 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF333 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###                                     ] | 8% Completed |  2min  1.6s\n",
      "2022-10-06 22:57:44,199 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF35 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###                                     ] | 8% Completed |  2min  2.0s\n",
      "2022-10-06 22:57:44,569 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF354C could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###                                     ] | 9% Completed |  2min  5.5s\n",
      "2022-10-06 22:57:48,005 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for NR2F6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###                                     ] | 9% Completed |  2min  8.0s\n",
      "2022-10-06 22:57:50,646 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for OLIG1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###                                     ] | 9% Completed |  2min  8.4s\n",
      "2022-10-06 22:57:50,940 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for MLXIP could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###                                     ] | 9% Completed |  2min  8.8s\n",
      "2022-10-06 22:57:51,356 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF577 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed |  2min  9.7s\n",
      "2022-10-06 22:57:52,278 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF589 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed |  2min 10.8s\n",
      "2022-10-06 22:57:53,436 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF75A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed |  2min 13.5s\n",
      "2022-10-06 22:57:55,997 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF525 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed |  2min 13.7s\n",
      "2022-10-06 22:57:56,257 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF599 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed |  2min 14.0s\n",
      "2022-10-06 22:57:56,591 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF181 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed |  2min 15.4s\n",
      "2022-10-06 22:57:57,932 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF184 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####                                    ] | 11% Completed |  2min 20.8s\n",
      "2022-10-06 22:58:03,370 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZBTB11 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####                                    ] | 11% Completed |  2min 23.4s\n",
      "2022-10-06 22:58:05,998 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF248 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####                                    ] | 11% Completed |  2min 23.5s\n",
      "2022-10-06 22:58:06,098 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF846 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[####                                    ] | 11% Completed |  2min 24.6s\n",
      "2022-10-06 22:58:07,240 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZBTB26 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####                                    ] | 12% Completed |  2min 26.8s\n",
      "2022-10-06 22:58:09,365 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF254 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[######                                  ] | 15% Completed |  2min 33.0s\n",
      "2022-10-06 22:58:15,626 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for AHCTF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[######                                  ] | 15% Completed |  2min 33.4s\n",
      "2022-10-06 22:58:15,937 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for PEG3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[######                                  ] | 15% Completed |  2min 36.6s\n",
      "2022-10-06 22:58:19,185 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for MLXIP could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[######                                  ] | 15% Completed |  2min 36.7s\n",
      "2022-10-06 22:58:19,270 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for MLXIPL could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[######                                  ] | 17% Completed |  2min 40.3s\n",
      "2022-10-06 22:58:22,857 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for MRPL2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 17% Completed |  2min 41.4s\n",
      "2022-10-06 22:58:24,056 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for STAT3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 43.0s\n",
      "2022-10-06 22:58:25,624 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for BPTF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 43.6s\n",
      "2022-10-06 22:58:26,129 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZBTB11 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 45.2s\n",
      "2022-10-06 22:58:27,742 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for TBX18 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 46.4s\n",
      "2022-10-06 22:58:28,934 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for GMEB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 47.9s\n",
      "2022-10-06 22:58:30,494 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZBTB26 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 48.4s\n",
      "2022-10-06 22:58:30,928 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF430 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 52.3s\n",
      "2022-10-06 22:58:34,864 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for PPP1R10 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 55.6s\n",
      "2022-10-06 22:58:38,176 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF184 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 55.7s\n",
      "2022-10-06 22:58:38,332 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for CBX7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 56.2s\n",
      "2022-10-06 22:58:38,759 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for CDK2AP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 56.5s\n",
      "2022-10-06 22:58:39,071 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for SF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 57.2s\n",
      "2022-10-06 22:58:39,771 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for PTCD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  2min 59.6s\n",
      "2022-10-06 22:58:42,244 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF562 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  3min  0.8s\n",
      "2022-10-06 22:58:43,377 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF577 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:43,423 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF24 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  3.1s\n",
      "2022-10-06 22:58:45,720 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF599 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  4.5s\n",
      "2022-10-06 22:58:47,065 - pyscenic.transform - WARNING - Less than 80% of the genes in CBX3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  4.6s\n",
      "2022-10-06 22:58:47,140 - pyscenic.transform - WARNING - Less than 80% of the genes in CCDC25 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  4.7s\n",
      "2022-10-06 22:58:47,210 - pyscenic.transform - WARNING - Less than 80% of the genes in CCNT2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  5.0s\n",
      "2022-10-06 22:58:47,576 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF263 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  5.1s\n",
      "2022-10-06 22:58:47,628 - pyscenic.transform - WARNING - Less than 80% of the genes in CDC5L could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:47,704 - pyscenic.transform - WARNING - Less than 80% of the genes in CDK2AP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  5.6s\n",
      "2022-10-06 22:58:48,194 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF846 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  7.0s\n",
      "2022-10-06 22:58:49,530 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPB could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  7.1s\n",
      "2022-10-06 22:58:49,648 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPD could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[########                                ] | 20% Completed |  3min  7.2s\n",
      "2022-10-06 22:58:49,728 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZSCAN25 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:49,813 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPZ could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  7.3s\n",
      "2022-10-06 22:58:49,884 - pyscenic.transform - WARNING - Less than 80% of the genes in CERS2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  7.5s\n",
      "2022-10-06 22:58:50,082 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZSCAN5A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  7.6s\n",
      "2022-10-06 22:58:50,181 - pyscenic.transform - WARNING - Less than 80% of the genes in CHD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  7.7s\n",
      "2022-10-06 22:58:50,271 - pyscenic.transform - WARNING - Less than 80% of the genes in CHD2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  7.8s\n",
      "2022-10-06 22:58:50,334 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF34 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:50,373 - pyscenic.transform - WARNING - Less than 80% of the genes in CHURC1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  7.9s\n",
      "2022-10-06 22:58:50,464 - pyscenic.transform - WARNING - Less than 80% of the genes in CLK1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  8.0s\n",
      "2022-10-06 22:58:50,538 - pyscenic.transform - WARNING - Less than 80% of the genes in CNOT4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:50,595 - pyscenic.transform - WARNING - Less than 80% of the genes in ACAA1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:50,617 - pyscenic.transform - WARNING - Less than 80% of the genes in CPSF4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  8.2s\n",
      "2022-10-06 22:58:50,769 - pyscenic.transform - WARNING - Less than 80% of the genes in CREB5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  8.3s\n",
      "2022-10-06 22:58:50,846 - pyscenic.transform - WARNING - Less than 80% of the genes in CREBL2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  9.2s\n",
      "2022-10-06 22:58:51,789 - pyscenic.transform - WARNING - Less than 80% of the genes in APEX1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  9.3s\n",
      "2022-10-06 22:58:51,858 - pyscenic.transform - WARNING - Less than 80% of the genes in ARG2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  9.5s\n",
      "2022-10-06 22:58:52,039 - pyscenic.transform - WARNING - Less than 80% of the genes in ARID5B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  9.6s\n",
      "2022-10-06 22:58:52,193 - pyscenic.transform - WARNING - Less than 80% of the genes in ASCL2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min  9.7s\n",
      "2022-10-06 22:58:52,260 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  3min 12.1s\n",
      "2022-10-06 22:58:54,660 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF669 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:54,674 - pyscenic.transform - WARNING - Less than 80% of the genes in CSNK2B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 12.2s\n",
      "2022-10-06 22:58:54,830 - pyscenic.transform - WARNING - Less than 80% of the genes in CTBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 12.3s\n",
      "2022-10-06 22:58:54,915 - pyscenic.transform - WARNING - Less than 80% of the genes in CTBP2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 13.2s\n",
      "2022-10-06 22:58:55,757 - pyscenic.transform - WARNING - Less than 80% of the genes in H1FX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 13.5s\n",
      "2022-10-06 22:58:56,050 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 13.6s\n",
      "2022-10-06 22:58:56,166 - pyscenic.transform - WARNING - Less than 80% of the genes in HBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 13.7s\n",
      "2022-10-06 22:58:56,291 - pyscenic.transform - WARNING - Less than 80% of the genes in CTCF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 13.9s\n",
      "2022-10-06 22:58:56,415 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for SNAPC4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:56,426 - pyscenic.transform - WARNING - Less than 80% of the genes in CTNNB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 14.1s\n",
      "2022-10-06 22:58:56,632 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for SND1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:56,680 - pyscenic.transform - WARNING - Less than 80% of the genes in CYCS could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 14.3s\n",
      "2022-10-06 22:58:56,905 - pyscenic.transform - WARNING - Less than 80% of the genes in DAZAP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 14.4s\n",
      "2022-10-06 22:58:56,983 - pyscenic.transform - WARNING - Less than 80% of the genes in DBP could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 14.8s\n",
      "2022-10-06 22:58:57,363 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for SOX4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 14.9s\n",
      "2022-10-06 22:58:57,429 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 15.5s\n",
      "2022-10-06 22:58:58,071 - pyscenic.transform - WARNING - Less than 80% of the genes in HDAC2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[########                                ] | 21% Completed |  3min 15.6s\n",
      "2022-10-06 22:58:58,152 - pyscenic.transform - WARNING - Less than 80% of the genes in HDAC3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 15.7s\n",
      "2022-10-06 22:58:58,244 - pyscenic.transform - WARNING - Less than 80% of the genes in HES1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 15.8s\n",
      "2022-10-06 22:58:58,335 - pyscenic.transform - WARNING - Less than 80% of the genes in HES4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:58,365 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF682 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 15.9s\n",
      "2022-10-06 22:58:58,506 - pyscenic.transform - WARNING - Less than 80% of the genes in HHEX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 16.0s\n",
      "2022-10-06 22:58:58,608 - pyscenic.transform - WARNING - Less than 80% of the genes in DHX36 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 16.1s\n",
      "2022-10-06 22:58:58,679 - pyscenic.transform - WARNING - Less than 80% of the genes in DMAP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:58,687 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:58,687 - pyscenic.transform - WARNING - Less than 80% of the genes in HIF1A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 16.2s\n",
      "2022-10-06 22:58:58,760 - pyscenic.transform - WARNING - Less than 80% of the genes in HIST2H2BE could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:58,768 - pyscenic.transform - WARNING - Less than 80% of the genes in DNAJC21 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:58,778 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF6B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 16.3s\n",
      "2022-10-06 22:58:58,855 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:58,917 - pyscenic.transform - WARNING - Less than 80% of the genes in HIVEP3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 16.4s\n",
      "2022-10-06 22:58:59,002 - pyscenic.transform - WARNING - Less than 80% of the genes in BACH1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:59,022 - pyscenic.transform - WARNING - Less than 80% of the genes in DUSP22 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 16.5s\n",
      "2022-10-06 22:58:59,082 - pyscenic.transform - WARNING - Less than 80% of the genes in HMG20B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 16.6s\n",
      "2022-10-06 22:58:59,157 - pyscenic.transform - WARNING - Less than 80% of the genes in BAD could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:58:59,177 - pyscenic.transform - WARNING - Less than 80% of the genes in HMGA1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 17.7s\n",
      "2022-10-06 22:59:00,320 - pyscenic.transform - WARNING - Less than 80% of the genes in HNRNPA0 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 18.2s\n",
      "2022-10-06 22:59:00,796 - pyscenic.transform - WARNING - Less than 80% of the genes in HNRNPH3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 18.3s\n",
      "2022-10-06 22:59:00,867 - pyscenic.transform - WARNING - Less than 80% of the genes in HOMEZ could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 18.5s\n",
      "2022-10-06 22:59:01,096 - pyscenic.transform - WARNING - Less than 80% of the genes in HOXB6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 18.6s\n",
      "2022-10-06 22:59:01,183 - pyscenic.transform - WARNING - Less than 80% of the genes in HP1BP3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 18.7s\n",
      "2022-10-06 22:59:01,281 - pyscenic.transform - WARNING - Less than 80% of the genes in HSPA5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed |  3min 19.0s\n",
      "2022-10-06 22:59:01,581 - pyscenic.transform - WARNING - Less than 80% of the genes in ID2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 22% Completed |  3min 19.9s\n",
      "2022-10-06 22:59:02,430 - pyscenic.transform - WARNING - Less than 80% of the genes in EEF1D could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[########                                ] | 22% Completed |  3min 20.1s\n",
      "2022-10-06 22:59:02,625 - pyscenic.transform - WARNING - Less than 80% of the genes in BAX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:02,715 - pyscenic.transform - WARNING - Less than 80% of the genes in BBX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 20.2s\n",
      "2022-10-06 22:59:02,791 - pyscenic.transform - WARNING - Less than 80% of the genes in BCL11A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 20.8s\n",
      "2022-10-06 22:59:03,349 - pyscenic.transform - WARNING - Less than 80% of the genes in IKZF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 20.9s\n",
      "2022-10-06 22:59:03,429 - pyscenic.transform - WARNING - Less than 80% of the genes in IKZF4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 21.0s\n",
      "2022-10-06 22:59:03,515 - pyscenic.transform - WARNING - Less than 80% of the genes in ILF2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 21.1s\n",
      "2022-10-06 22:59:03,700 - pyscenic.transform - WARNING - Less than 80% of the genes in RAB2A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 21.2s\n",
      "2022-10-06 22:59:03,795 - pyscenic.transform - WARNING - Less than 80% of the genes in RAB7A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 21.4s\n",
      "2022-10-06 22:59:03,990 - pyscenic.transform - WARNING - Less than 80% of the genes in EGR1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 21.9s\n",
      "2022-10-06 22:59:04,494 - pyscenic.transform - WARNING - Less than 80% of the genes in MGA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[#########                               ] | 23% Completed |  3min 22.2s\n",
      "2022-10-06 22:59:04,759 - pyscenic.transform - WARNING - Less than 80% of the genes in MLX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 22.3s\n",
      "2022-10-06 22:59:04,829 - pyscenic.transform - WARNING - Less than 80% of the genes in MLXIP could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:04,904 - pyscenic.transform - WARNING - Less than 80% of the genes in MRPL1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 22.4s\n",
      "2022-10-06 22:59:04,980 - pyscenic.transform - WARNING - Less than 80% of the genes in MRPL2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 22.6s\n",
      "2022-10-06 22:59:05,171 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 22.7s\n",
      "2022-10-06 22:59:05,233 - pyscenic.transform - WARNING - Less than 80% of the genes in MSRA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:05,245 - pyscenic.transform - WARNING - Less than 80% of the genes in EGR2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:05,248 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 22.8s\n",
      "2022-10-06 22:59:05,344 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 23.5s\n",
      "2022-10-06 22:59:06,044 - pyscenic.transform - WARNING - Less than 80% of the genes in BCL3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 23.6s\n",
      "2022-10-06 22:59:06,212 - pyscenic.transform - WARNING - Less than 80% of the genes in BCLAF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 24.1s\n",
      "2022-10-06 22:59:06,678 - pyscenic.transform - WARNING - Less than 80% of the genes in ELF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 24.4s\n",
      "2022-10-06 22:59:06,972 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 25.0s\n",
      "2022-10-06 22:59:07,594 - pyscenic.transform - WARNING - Less than 80% of the genes in EP300 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 25.1s\n",
      "2022-10-06 22:59:07,616 - pyscenic.transform - WARNING - Less than 80% of the genes in BHLHE40 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 25.2s\n",
      "2022-10-06 22:59:07,716 - pyscenic.transform - WARNING - Less than 80% of the genes in BHLHE41 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:07,804 - pyscenic.transform - WARNING - Less than 80% of the genes in BPTF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 25.3s\n",
      "2022-10-06 22:59:07,842 - pyscenic.transform - WARNING - Less than 80% of the genes in TFAM could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 25.7s\n",
      "2022-10-06 22:59:08,236 - pyscenic.transform - WARNING - Less than 80% of the genes in MTF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 25.8s\n",
      "2022-10-06 22:59:08,320 - pyscenic.transform - WARNING - Less than 80% of the genes in MXD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 26.3s\n",
      "2022-10-06 22:59:08,822 - pyscenic.transform - WARNING - Less than 80% of the genes in ESRRA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  3min 26.4s\n",
      "2022-10-06 22:59:08,921 - pyscenic.transform - WARNING - Less than 80% of the genes in ETFB could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 26.5s\n",
      "2022-10-06 22:59:09,116 - pyscenic.transform - WARNING - Less than 80% of the genes in ETS2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 26.6s\n",
      "2022-10-06 22:59:09,191 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 26.7s\n",
      "2022-10-06 22:59:09,274 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 26.8s\n",
      "2022-10-06 22:59:09,354 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:09,429 - pyscenic.transform - WARNING - Less than 80% of the genes in EWSR1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 26.9s\n",
      "2022-10-06 22:59:09,497 - pyscenic.transform - WARNING - Less than 80% of the genes in EXOSC3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 27.0s\n",
      "2022-10-06 22:59:09,584 - pyscenic.transform - WARNING - Less than 80% of the genes in EZR could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:09,647 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for THAP5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:09,661 - pyscenic.transform - WARNING - Less than 80% of the genes in FHL2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 27.2s\n",
      "2022-10-06 22:59:09,734 - pyscenic.transform - WARNING - Less than 80% of the genes in FIP1L1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 27.4s\n",
      "2022-10-06 22:59:09,891 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF451 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 27.9s\n",
      "2022-10-06 22:59:10,422 - pyscenic.transform - WARNING - Less than 80% of the genes in RBBP5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 28.0s\n",
      "2022-10-06 22:59:10,516 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM17 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 28.2s\n",
      "2022-10-06 22:59:10,773 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM42 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[#########                               ] | 24% Completed |  3min 28.3s\n",
      "2022-10-06 22:59:10,859 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM8A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 28.4s\n",
      "2022-10-06 22:59:10,939 - pyscenic.transform - WARNING - Less than 80% of the genes in C19orf25 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:10,948 - pyscenic.transform - WARNING - Less than 80% of the genes in RBMS1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 28.5s\n",
      "2022-10-06 22:59:11,003 - pyscenic.transform - WARNING - Less than 80% of the genes in TFDP2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:11,072 - pyscenic.transform - WARNING - Less than 80% of the genes in TFEB could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 28.7s\n",
      "2022-10-06 22:59:11,241 - pyscenic.transform - WARNING - Less than 80% of the genes in CAT could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed |  3min 28.8s\n",
      "2022-10-06 22:59:11,288 - pyscenic.transform - WARNING - Less than 80% of the genes in TGIF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:11,321 - pyscenic.transform - WARNING - Less than 80% of the genes in CBFB could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:11,363 - pyscenic.transform - WARNING - Less than 80% of the genes in THAP5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 28.9s\n",
      "2022-10-06 22:59:11,445 - pyscenic.transform - WARNING - Less than 80% of the genes in THOC2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:11,478 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF467 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 29.0s\n",
      "2022-10-06 22:59:11,515 - pyscenic.transform - WARNING - Less than 80% of the genes in TIA1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:11,551 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF493 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 29.1s\n",
      "2022-10-06 22:59:11,601 - pyscenic.transform - WARNING - Less than 80% of the genes in TMEM33 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:11,625 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF503 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 29.2s\n",
      "2022-10-06 22:59:11,781 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF524 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 29.5s\n",
      "2022-10-06 22:59:12,106 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF581 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 29.6s\n",
      "2022-10-06 22:59:12,182 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF585B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 29.9s\n",
      "2022-10-06 22:59:12,412 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF622 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:12,446 - pyscenic.transform - WARNING - Less than 80% of the genes in CCDC25 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 30.1s\n",
      "2022-10-06 22:59:12,577 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF641 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:12,666 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF652 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 30.2s\n",
      "2022-10-06 22:59:12,735 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF654 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 30.3s\n",
      "2022-10-06 22:59:12,815 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF655 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 30.8s\n",
      "2022-10-06 22:59:13,296 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for MAGEF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:13,371 - pyscenic.transform - WARNING - Less than 80% of the genes in TP53 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 30.9s\n",
      "2022-10-06 22:59:13,397 - pyscenic.transform - WARNING - Less than 80% of the genes in CDC5L could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 31.0s\n",
      "2022-10-06 22:59:13,478 - pyscenic.transform - WARNING - Less than 80% of the genes in CDK2AP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 31.3s\n",
      "2022-10-06 22:59:13,854 - pyscenic.transform - WARNING - Less than 80% of the genes in JAZF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 31.5s\n",
      "2022-10-06 22:59:13,984 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPD could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:14,069 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPG could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 31.6s\n",
      "2022-10-06 22:59:14,155 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPZ could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 33.4s\n",
      "2022-10-06 22:59:15,912 - pyscenic.transform - WARNING - Less than 80% of the genes in CERS2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 33.5s\n",
      "2022-10-06 22:59:16,062 - pyscenic.transform - WARNING - Less than 80% of the genes in CERS5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 33.8s\n",
      "2022-10-06 22:59:16,306 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF677 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 33.9s\n",
      "2022-10-06 22:59:16,391 - pyscenic.transform - WARNING - Less than 80% of the genes in CHD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed |  3min 34.0s\n",
      "2022-10-06 22:59:16,551 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF688 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[##########                              ] | 26% Completed |  3min 34.2s\n",
      "2022-10-06 22:59:16,719 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF708 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 34.3s\n",
      "2022-10-06 22:59:16,862 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF720 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:16,927 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF785 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 34.5s\n",
      "2022-10-06 22:59:17,068 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF791 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 34.6s\n",
      "2022-10-06 22:59:17,243 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF808 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 34.8s\n",
      "2022-10-06 22:59:17,328 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF880 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 34.9s\n",
      "2022-10-06 22:59:17,483 - pyscenic.transform - WARNING - Less than 80% of the genes in ZSCAN25 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 35.0s\n",
      "2022-10-06 22:59:17,553 - pyscenic.transform - WARNING - Less than 80% of the genes in ZZZ3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 35.1s\n",
      "2022-10-06 22:59:17,604 - pyscenic.transform - WARNING - Less than 80% of the genes in TRIM69 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 35.2s\n",
      "2022-10-06 22:59:17,711 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXD4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:17,717 - pyscenic.transform - WARNING - Less than 80% of the genes in ACAA1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:17,765 - pyscenic.transform - WARNING - Less than 80% of the genes in TRMT1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 35.3s\n",
      "2022-10-06 22:59:17,795 - pyscenic.transform - WARNING - Less than 80% of the genes in ACO1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:17,842 - pyscenic.transform - WARNING - Less than 80% of the genes in TROVE2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:17,885 - pyscenic.transform - WARNING - Less than 80% of the genes in ADNP could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 35.4s\n",
      "2022-10-06 22:59:17,919 - pyscenic.transform - WARNING - Less than 80% of the genes in TRPS1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:17,972 - pyscenic.transform - WARNING - Less than 80% of the genes in CNOT4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 35.5s\n",
      "2022-10-06 22:59:17,996 - pyscenic.transform - WARNING - Less than 80% of the genes in TSC22D4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:18,055 - pyscenic.transform - WARNING - Less than 80% of the genes in AFF4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:18,066 - pyscenic.transform - WARNING - Less than 80% of the genes in TSN could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 35.6s\n",
      "2022-10-06 22:59:18,135 - pyscenic.transform - WARNING - Less than 80% of the genes in AGGF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:18,143 - pyscenic.transform - WARNING - Less than 80% of the genes in CREB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:18,159 - pyscenic.transform - WARNING - Less than 80% of the genes in U2AF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 35.7s\n",
      "2022-10-06 22:59:18,209 - pyscenic.transform - WARNING - Less than 80% of the genes in AHCTF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:18,217 - pyscenic.transform - WARNING - Less than 80% of the genes in CREB3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 35.8s\n",
      "2022-10-06 22:59:18,384 - pyscenic.transform - WARNING - Less than 80% of the genes in CREB5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 35.9s\n",
      "2022-10-06 22:59:18,478 - pyscenic.transform - WARNING - Less than 80% of the genes in CREBL2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 36.0s\n",
      "2022-10-06 22:59:18,547 - pyscenic.transform - WARNING - Less than 80% of the genes in CREBZF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 36.8s\n",
      "2022-10-06 22:59:19,297 - pyscenic.transform - WARNING - Less than 80% of the genes in UBE2K could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:19,372 - pyscenic.transform - WARNING - Less than 80% of the genes in JUND could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 36.9s\n",
      "2022-10-06 22:59:19,398 - pyscenic.transform - WARNING - Less than 80% of the genes in UBXN1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 37.1s\n",
      "2022-10-06 22:59:19,613 - pyscenic.transform - WARNING - Less than 80% of the genes in MYC could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:19,618 - pyscenic.transform - WARNING - Less than 80% of the genes in APEX1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:19,693 - pyscenic.transform - WARNING - Less than 80% of the genes in APEX2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:19,694 - pyscenic.transform - WARNING - Less than 80% of the genes in MYLK could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 37.2s\n",
      "2022-10-06 22:59:19,763 - pyscenic.transform - WARNING - Less than 80% of the genes in ARFGAP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 37.3s\n",
      "2022-10-06 22:59:19,800 - pyscenic.transform - WARNING - Less than 80% of the genes in KDM5A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:19,841 - pyscenic.transform - WARNING - Less than 80% of the genes in ARG2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:19,875 - pyscenic.transform - WARNING - Less than 80% of the genes in KDM7A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[##########                              ] | 26% Completed |  3min 37.4s\n",
      "2022-10-06 22:59:19,914 - pyscenic.transform - WARNING - Less than 80% of the genes in ARID3A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:19,947 - pyscenic.transform - WARNING - Less than 80% of the genes in KIAA0907 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 37.5s\n",
      "2022-10-06 22:59:20,018 - pyscenic.transform - WARNING - Less than 80% of the genes in USP39 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:20,095 - pyscenic.transform - WARNING - Less than 80% of the genes in UTP18 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 37.6s\n",
      "2022-10-06 22:59:20,113 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF10 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:20,169 - pyscenic.transform - WARNING - Less than 80% of the genes in VAMP3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:20,181 - pyscenic.transform - WARNING - Less than 80% of the genes in ARID5B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:20,196 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF13 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 37.7s\n",
      "2022-10-06 22:59:20,240 - pyscenic.transform - WARNING - Less than 80% of the genes in VDR could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 37.8s\n",
      "2022-10-06 22:59:20,385 - pyscenic.transform - WARNING - Less than 80% of the genes in VPS72 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 37.9s\n",
      "2022-10-06 22:59:20,427 - pyscenic.transform - WARNING - Less than 80% of the genes in ASCC1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:20,477 - pyscenic.transform - WARNING - Less than 80% of the genes in XBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 38.0s\n",
      "2022-10-06 22:59:20,514 - pyscenic.transform - WARNING - Less than 80% of the genes in ASCL2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:20,550 - pyscenic.transform - WARNING - Less than 80% of the genes in XPA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 38.2s\n",
      "2022-10-06 22:59:20,725 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXN3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 38.3s\n",
      "2022-10-06 22:59:20,810 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXO1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 38.5s\n",
      "2022-10-06 22:59:21,077 - pyscenic.transform - WARNING - Less than 80% of the genes in RFXANK could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 38.8s\n",
      "2022-10-06 22:59:21,318 - pyscenic.transform - WARNING - Less than 80% of the genes in RNASEH2C could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 38.9s\n",
      "2022-10-06 22:59:21,412 - pyscenic.transform - WARNING - Less than 80% of the genes in RNF114 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:21,464 - pyscenic.transform - WARNING - Less than 80% of the genes in YOD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 39.3s\n",
      "2022-10-06 22:59:21,857 - pyscenic.transform - WARNING - Less than 80% of the genes in YY1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 39.4s\n",
      "2022-10-06 22:59:21,933 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 39.5s\n",
      "2022-10-06 22:59:22,005 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB11 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:22,067 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXO3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 39.6s\n",
      "2022-10-06 22:59:22,162 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 39.8s\n",
      "2022-10-06 22:59:22,345 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXN3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:22,401 - pyscenic.transform - WARNING - Less than 80% of the genes in FUBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 40.0s\n",
      "2022-10-06 22:59:22,593 - pyscenic.transform - WARNING - Less than 80% of the genes in CRTC2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 40.3s\n",
      "2022-10-06 22:59:22,882 - pyscenic.transform - WARNING - Less than 80% of the genes in CTBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 40.6s\n",
      "2022-10-06 22:59:23,128 - pyscenic.transform - WARNING - Less than 80% of the genes in NCOR1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:23,200 - pyscenic.transform - WARNING - Less than 80% of the genes in NCOR2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 41.1s\n",
      "2022-10-06 22:59:23,624 - pyscenic.transform - WARNING - Less than 80% of the genes in RREB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:23,642 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXO1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 41.2s\n",
      "2022-10-06 22:59:23,712 - pyscenic.transform - WARNING - Less than 80% of the genes in RUFY3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:23,741 - pyscenic.transform - WARNING - Less than 80% of the genes in KMT2A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 41.3s\n",
      "2022-10-06 22:59:23,843 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 41.4s\n",
      "2022-10-06 22:59:23,916 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 42.1s\n",
      "2022-10-06 22:59:24,686 - pyscenic.transform - WARNING - Less than 80% of the genes in NELFE could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[##########                              ] | 26% Completed |  3min 42.2s\n",
      "2022-10-06 22:59:24,767 - pyscenic.transform - WARNING - Less than 80% of the genes in NF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:24,836 - pyscenic.transform - WARNING - Less than 80% of the genes in NFAT5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 42.4s\n",
      "2022-10-06 22:59:24,987 - pyscenic.transform - WARNING - Less than 80% of the genes in RUNX1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 42.8s\n",
      "2022-10-06 22:59:25,410 - pyscenic.transform - WARNING - Less than 80% of the genes in LEF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 43.0s\n",
      "2022-10-06 22:59:25,558 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB25 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:25,566 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:25,582 - pyscenic.transform - WARNING - Less than 80% of the genes in GADD45A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 43.1s\n",
      "2022-10-06 22:59:25,626 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB38 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:25,698 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 43.2s\n",
      "2022-10-06 22:59:25,777 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB41 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 43.3s\n",
      "2022-10-06 22:59:25,857 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB43 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 43.4s\n",
      "2022-10-06 22:59:25,938 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB44 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 43.7s\n",
      "2022-10-06 22:59:26,229 - pyscenic.transform - WARNING - Less than 80% of the genes in CTNNB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:26,248 - pyscenic.transform - WARNING - Less than 80% of the genes in ZDHHC24 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 44.3s\n",
      "2022-10-06 22:59:26,839 - pyscenic.transform - WARNING - Less than 80% of the genes in LMO2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 44.4s\n",
      "2022-10-06 22:59:26,932 - pyscenic.transform - WARNING - Less than 80% of the genes in LRRFIP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:27,014 - pyscenic.transform - WARNING - Less than 80% of the genes in LSM6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 44.5s\n",
      "2022-10-06 22:59:27,039 - pyscenic.transform - WARNING - Less than 80% of the genes in FUBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:27,105 - pyscenic.transform - WARNING - Less than 80% of the genes in LUZP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 44.6s\n",
      "2022-10-06 22:59:27,175 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:27,176 - pyscenic.transform - WARNING - Less than 80% of the genes in LYL1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 44.7s\n",
      "2022-10-06 22:59:27,284 - pyscenic.transform - WARNING - Less than 80% of the genes in MAF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 44.8s\n",
      "2022-10-06 22:59:27,423 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 45.0s\n",
      "2022-10-06 22:59:27,614 - pyscenic.transform - WARNING - Less than 80% of the genes in MAFF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 45.2s\n",
      "2022-10-06 22:59:27,796 - pyscenic.transform - WARNING - Less than 80% of the genes in BACH1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:27,812 - pyscenic.transform - WARNING - Less than 80% of the genes in CUX1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 45.3s\n",
      "2022-10-06 22:59:27,887 - pyscenic.transform - WARNING - Less than 80% of the genes in CXXC1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 45.4s\n",
      "2022-10-06 22:59:27,957 - pyscenic.transform - WARNING - Less than 80% of the genes in BAD could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:27,980 - pyscenic.transform - WARNING - Less than 80% of the genes in CYB5R1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 45.5s\n",
      "2022-10-06 22:59:28,098 - pyscenic.transform - WARNING - Less than 80% of the genes in CYCS could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 45.7s\n",
      "2022-10-06 22:59:28,291 - pyscenic.transform - WARNING - Less than 80% of the genes in SAP30 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 45.9s\n",
      "2022-10-06 22:59:28,456 - pyscenic.transform - WARNING - Less than 80% of the genes in SEMA4A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:28,521 - pyscenic.transform - WARNING - Less than 80% of the genes in DBP could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 46.1s\n",
      "2022-10-06 22:59:28,693 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 46.2s\n",
      "2022-10-06 22:59:28,767 - pyscenic.transform - WARNING - Less than 80% of the genes in GIT2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 46.3s\n",
      "2022-10-06 22:59:28,856 - pyscenic.transform - WARNING - Less than 80% of the genes in DHX36 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 46.4s\n",
      "2022-10-06 22:59:28,998 - pyscenic.transform - WARNING - Less than 80% of the genes in DMAP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:29,013 - pyscenic.transform - WARNING - Less than 80% of the genes in GRHPR could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[##########                              ] | 26% Completed |  3min 46.5s\n",
      "2022-10-06 22:59:29,046 - pyscenic.transform - WARNING - Less than 80% of the genes in MAFG could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:29,081 - pyscenic.transform - WARNING - Less than 80% of the genes in DNAJC21 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:29,106 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2A2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:29,124 - pyscenic.transform - WARNING - Less than 80% of the genes in MAGEF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 46.6s\n",
      "2022-10-06 22:59:29,183 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:29,198 - pyscenic.transform - WARNING - Less than 80% of the genes in MAGOH could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 46.7s\n",
      "2022-10-06 22:59:29,277 - pyscenic.transform - WARNING - Less than 80% of the genes in DNTTIP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 46.8s\n",
      "2022-10-06 22:59:29,339 - pyscenic.transform - WARNING - Less than 80% of the genes in ZEB2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:29,425 - pyscenic.transform - WARNING - Less than 80% of the genes in ZFHX3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 46.9s\n",
      "2022-10-06 22:59:29,439 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2I could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 47.0s\n",
      "2022-10-06 22:59:29,541 - pyscenic.transform - WARNING - Less than 80% of the genes in DUSP22 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:29,627 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF3A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 47.1s\n",
      "2022-10-06 22:59:29,728 - pyscenic.transform - WARNING - Less than 80% of the genes in ZKSCAN1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 47.2s\n",
      "2022-10-06 22:59:29,804 - pyscenic.transform - WARNING - Less than 80% of the genes in ZMAT2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:29,834 - pyscenic.transform - WARNING - Less than 80% of the genes in SETDB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  3min 47.3s\n",
      "2022-10-06 22:59:29,921 - pyscenic.transform - WARNING - Less than 80% of the genes in SF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##########                              ] | 27% Completed |  3min 47.4s\n",
      "2022-10-06 22:59:29,998 - pyscenic.transform - WARNING - Less than 80% of the genes in SF3B1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:30,024 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 47.8s\n",
      "2022-10-06 22:59:30,334 - pyscenic.transform - WARNING - Less than 80% of the genes in GABPB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 47.9s\n",
      "2022-10-06 22:59:30,436 - pyscenic.transform - WARNING - Less than 80% of the genes in GADD45A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 48.0s\n",
      "2022-10-06 22:59:30,579 - pyscenic.transform - WARNING - Less than 80% of the genes in MAX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 48.1s\n",
      "2022-10-06 22:59:30,648 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 48.2s\n",
      "2022-10-06 22:59:30,727 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:30,816 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 48.3s\n",
      "2022-10-06 22:59:30,895 - pyscenic.transform - WARNING - Less than 80% of the genes in MBNL2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 48.5s\n",
      "2022-10-06 22:59:31,134 - pyscenic.transform - WARNING - Less than 80% of the genes in MECP2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 48.7s\n",
      "2022-10-06 22:59:31,208 - pyscenic.transform - WARNING - Less than 80% of the genes in NFIA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,213 - pyscenic.transform - WARNING - Less than 80% of the genes in MED30 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,244 - pyscenic.transform - WARNING - Less than 80% of the genes in NCOR1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 48.8s\n",
      "2022-10-06 22:59:31,291 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,346 - pyscenic.transform - WARNING - Less than 80% of the genes in NCOR2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,386 - pyscenic.transform - WARNING - Less than 80% of the genes in ZMIZ1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 48.9s\n",
      "2022-10-06 22:59:31,405 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2C could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,433 - pyscenic.transform - WARNING - Less than 80% of the genes in BAX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,459 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF100 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,476 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2D could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 49.0s\n",
      "2022-10-06 22:59:31,529 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF12 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,541 - pyscenic.transform - WARNING - Less than 80% of the genes in BBX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,601 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF131 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 49.1s\n",
      "2022-10-06 22:59:31,624 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,630 - pyscenic.transform - WARNING - Less than 80% of the genes in BCL11A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-10-06 22:59:31,696 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 49.3s\n",
      "2022-10-06 22:59:31,858 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,907 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF165 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:31,942 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAP2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  3min 49.4s\n",
      "2022-10-06 22:59:32,025 - pyscenic.transform - WARNING - Less than 80% of the genes in SMARCA1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:32,057 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed |  3min 49.5s\n",
      "2022-10-06 22:59:32,141 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF207 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed |  3min 49.7s\n",
      "2022-10-06 22:59:32,189 - pyscenic.transform - WARNING - Less than 80% of the genes in SMARCB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:32,218 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF22 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:32,263 - pyscenic.transform - WARNING - Less than 80% of the genes in SMARCC1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed |  3min 49.8s\n",
      "2022-10-06 22:59:32,287 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF224 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:32,357 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF226 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed |  3min 49.9s\n",
      "2022-10-06 22:59:32,442 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF24 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed |  3min 50.0s\n",
      "2022-10-06 22:59:32,604 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF267 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed |  3min 50.2s\n",
      "2022-10-06 22:59:32,764 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF280D could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:32,769 - pyscenic.transform - WARNING - Less than 80% of the genes in E2F3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:32,769 - pyscenic.transform - WARNING - Less than 80% of the genes in NFIC could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed |  3min 50.3s\n",
      "2022-10-06 22:59:32,849 - pyscenic.transform - WARNING - Less than 80% of the genes in NFIL3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed |  3min 50.4s\n",
      "2022-10-06 22:59:32,990 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF302 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed |  3min 50.5s\n",
      "2022-10-06 22:59:33,067 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF32 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:33,154 - pyscenic.transform - WARNING - Less than 80% of the genes in NELFB could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 50.6s\n",
      "2022-10-06 22:59:33,232 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF331 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:33,254 - pyscenic.transform - WARNING - Less than 80% of the genes in NELFE could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 50.8s\n",
      "2022-10-06 22:59:33,343 - pyscenic.transform - WARNING - Less than 80% of the genes in NF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 50.9s\n",
      "2022-10-06 22:59:33,437 - pyscenic.transform - WARNING - Less than 80% of the genes in NFAT5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 51.0s\n",
      "2022-10-06 22:59:33,549 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF385A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 51.1s\n",
      "2022-10-06 22:59:33,624 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF394 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:33,679 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAI1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:33,694 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF429 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 51.2s\n",
      "2022-10-06 22:59:33,761 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF43 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:33,798 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF9 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 51.3s\n",
      "2022-10-06 22:59:33,831 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF430 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:33,836 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAPC5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:33,892 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB11 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:33,908 - pyscenic.transform - WARNING - Less than 80% of the genes in SND1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 51.4s\n",
      "2022-10-06 22:59:33,992 - pyscenic.transform - WARNING - Less than 80% of the genes in SNRNP70 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 51.5s\n",
      "2022-10-06 22:59:34,009 - pyscenic.transform - WARNING - Less than 80% of the genes in E2F4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:34,033 - pyscenic.transform - WARNING - Less than 80% of the genes in JAZF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:34,090 - pyscenic.transform - WARNING - Less than 80% of the genes in SNRPB2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 51.6s\n",
      "2022-10-06 22:59:34,161 - pyscenic.transform - WARNING - Less than 80% of the genes in SOCS4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 51.8s\n",
      "2022-10-06 22:59:34,407 - pyscenic.transform - WARNING - Less than 80% of the genes in SOX4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 51.9s\n",
      "2022-10-06 22:59:34,470 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for NR2F6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 52.0s\n",
      "2022-10-06 22:59:34,596 - pyscenic.transform - WARNING - Less than 80% of the genes in EGR1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:34,601 - pyscenic.transform - WARNING - Less than 80% of the genes in SP100 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 52.3s\n",
      "2022-10-06 22:59:34,918 - pyscenic.transform - WARNING - Less than 80% of the genes in BCL3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 52.5s\n",
      "2022-10-06 22:59:35,070 - pyscenic.transform - WARNING - Less than 80% of the genes in GMEB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:35,124 - pyscenic.transform - WARNING - Less than 80% of the genes in BCLAF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 52.8s\n",
      "2022-10-06 22:59:35,414 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2A1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 52.9s\n",
      "2022-10-06 22:59:35,516 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2A2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 53.0s\n",
      "2022-10-06 22:59:35,606 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 53.1s\n",
      "2022-10-06 22:59:35,700 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2F1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 53.4s\n",
      "2022-10-06 22:59:35,890 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2I could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 53.5s\n",
      "2022-10-06 22:59:36,013 - pyscenic.transform - WARNING - Less than 80% of the genes in EGR2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:36,016 - pyscenic.transform - WARNING - Less than 80% of the genes in SP3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:36,061 - pyscenic.transform - WARNING - Less than 80% of the genes in NFKB2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 53.6s\n",
      "2022-10-06 22:59:36,095 - pyscenic.transform - WARNING - Less than 80% of the genes in SPAG7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:36,096 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF3A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:36,169 - pyscenic.transform - WARNING - Less than 80% of the genes in SPATS2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 53.7s\n",
      "2022-10-06 22:59:36,169 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF429 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:36,244 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF43 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:36,256 - pyscenic.transform - WARNING - Less than 80% of the genes in GTPBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 53.8s\n",
      "2022-10-06 22:59:36,325 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF430 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:36,369 - pyscenic.transform - WARNING - Less than 80% of the genes in EIF5A2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 53.9s\n",
      "2022-10-06 22:59:36,416 - pyscenic.transform - WARNING - Less than 80% of the genes in GZF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 54.0s\n",
      "2022-10-06 22:59:36,484 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF438 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:36,511 - pyscenic.transform - WARNING - Less than 80% of the genes in H1FX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:36,554 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF439 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 54.2s\n",
      "2022-10-06 22:59:36,717 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF451 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 54.3s\n",
      "2022-10-06 22:59:36,773 - pyscenic.transform - WARNING - Less than 80% of the genes in BHLHE40 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 54.4s\n",
      "2022-10-06 22:59:36,895 - pyscenic.transform - WARNING - Less than 80% of the genes in BHLHE41 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 54.5s\n",
      "2022-10-06 22:59:36,999 - pyscenic.transform - WARNING - Less than 80% of the genes in BPTF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 54.6s\n",
      "2022-10-06 22:59:37,109 - pyscenic.transform - WARNING - Less than 80% of the genes in HBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 54.9s\n",
      "2022-10-06 22:59:37,441 - pyscenic.transform - WARNING - Less than 80% of the genes in HDAC1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 55.4s\n",
      "2022-10-06 22:59:37,971 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF467 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 55.5s\n",
      "2022-10-06 22:59:38,025 - pyscenic.transform - WARNING - Less than 80% of the genes in ELK3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 55.6s\n",
      "2022-10-06 22:59:38,099 - pyscenic.transform - WARNING - Less than 80% of the genes in ELK4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[############                            ] | 30% Completed |  3min 55.7s\n",
      "2022-10-06 22:59:38,191 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF493 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:38,276 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF503 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 55.8s\n",
      "2022-10-06 22:59:38,352 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF506 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 56.0s\n",
      "2022-10-06 22:59:38,499 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF510 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:38,568 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF513 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 56.1s\n",
      "2022-10-06 22:59:38,649 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF516 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 56.2s\n",
      "2022-10-06 22:59:38,739 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF524 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:38,751 - pyscenic.transform - WARNING - Less than 80% of the genes in HDAC2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 56.3s\n",
      "2022-10-06 22:59:38,836 - pyscenic.transform - WARNING - Less than 80% of the genes in HDAC3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  3min 56.4s\n",
      "2022-10-06 22:59:38,909 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF529 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 56.9s\n",
      "2022-10-06 22:59:39,425 - pyscenic.transform - WARNING - Less than 80% of the genes in KDM5A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:39,435 - pyscenic.transform - WARNING - Less than 80% of the genes in NOC2L could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:39,521 - pyscenic.transform - WARNING - Less than 80% of the genes in NONO could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 57.0s\n",
      "2022-10-06 22:59:39,561 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB33 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:39,591 - pyscenic.transform - WARNING - Less than 80% of the genes in KDM7A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 57.1s\n",
      "2022-10-06 22:59:39,673 - pyscenic.transform - WARNING - Less than 80% of the genes in NFE2L1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:39,687 - pyscenic.transform - WARNING - Less than 80% of the genes in KIAA0907 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:39,730 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB38 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 57.3s\n",
      "2022-10-06 22:59:39,859 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:39,889 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF10 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:39,949 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB41 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 57.5s\n",
      "2022-10-06 22:59:39,985 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF13 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 57.6s\n",
      "2022-10-06 22:59:40,120 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB43 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 57.7s\n",
      "2022-10-06 22:59:40,221 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB44 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:40,229 - pyscenic.transform - WARNING - Less than 80% of the genes in HES1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 57.8s\n",
      "2022-10-06 22:59:40,369 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB48 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:40,401 - pyscenic.transform - WARNING - Less than 80% of the genes in HES6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 58.0s\n",
      "2022-10-06 22:59:40,588 - pyscenic.transform - WARNING - Less than 80% of the genes in HHEX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 58.3s\n",
      "2022-10-06 22:59:40,915 - pyscenic.transform - WARNING - Less than 80% of the genes in RBBP5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 58.5s\n",
      "2022-10-06 22:59:41,117 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM22 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 58.7s\n",
      "2022-10-06 22:59:41,331 - pyscenic.transform - WARNING - Less than 80% of the genes in ZSCAN25 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 58.8s\n",
      "2022-10-06 22:59:41,387 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:41,404 - pyscenic.transform - WARNING - Less than 80% of the genes in ZXDC could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 58.9s\n",
      "2022-10-06 22:59:41,485 - pyscenic.transform - WARNING - Less than 80% of the genes in ZZZ3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:41,501 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM8A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 59.8s\n",
      "2022-10-06 22:59:42,422 - pyscenic.transform - WARNING - Less than 80% of the genes in SPR could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  3min 59.9s\n",
      "2022-10-06 22:59:42,563 - pyscenic.transform - WARNING - Less than 80% of the genes in EP300 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  0.3s\n",
      "2022-10-06 22:59:42,887 - pyscenic.transform - WARNING - Less than 80% of the genes in AGMAT could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:42,918 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF580 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[############                            ] | 31% Completed |  4min  0.5s\n",
      "2022-10-06 22:59:43,105 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,124 - pyscenic.transform - WARNING - Less than 80% of the genes in NR3C1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  0.7s\n",
      "2022-10-06 22:59:43,228 - pyscenic.transform - WARNING - Less than 80% of the genes in NR4A1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,262 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT5B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  0.8s\n",
      "2022-10-06 22:59:43,267 - pyscenic.transform - WARNING - Less than 80% of the genes in RCOR1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,338 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  0.9s\n",
      "2022-10-06 22:59:43,405 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF611 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  1.0s\n",
      "2022-10-06 22:59:43,519 - pyscenic.transform - WARNING - Less than 80% of the genes in STUB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  1.1s\n",
      "2022-10-06 22:59:43,622 - pyscenic.transform - WARNING - Less than 80% of the genes in NUP107 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,632 - pyscenic.transform - WARNING - Less than 80% of the genes in SUCLG1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,634 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF622 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,658 - pyscenic.transform - WARNING - Less than 80% of the genes in SRF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  1.2s\n",
      "2022-10-06 22:59:43,684 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,713 - pyscenic.transform - WARNING - Less than 80% of the genes in NFE2L3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,750 - pyscenic.transform - WARNING - Less than 80% of the genes in SRP9 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  1.3s\n",
      "2022-10-06 22:59:43,797 - pyscenic.transform - WARNING - Less than 80% of the genes in NFIA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,818 - pyscenic.transform - WARNING - Less than 80% of the genes in SUZ12 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,861 - pyscenic.transform - WARNING - Less than 80% of the genes in ERF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  1.4s\n",
      "2022-10-06 22:59:43,876 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF626 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:43,947 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF639 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  1.5s\n",
      "2022-10-06 22:59:43,996 - pyscenic.transform - WARNING - Less than 80% of the genes in NFIC could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:44,026 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF641 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  1.6s\n",
      "2022-10-06 22:59:44,095 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF649 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:44,133 - pyscenic.transform - WARNING - Less than 80% of the genes in PBX2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  1.7s\n",
      "2022-10-06 22:59:44,176 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:44,185 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF652 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:44,203 - pyscenic.transform - WARNING - Less than 80% of the genes in CAT could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  4min  1.8s\n",
      "2022-10-06 22:59:44,297 - pyscenic.transform - WARNING - Less than 80% of the genes in CBFB could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:44,354 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF655 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  2.3s\n",
      "2022-10-06 22:59:44,845 - pyscenic.transform - WARNING - Less than 80% of the genes in HIF1A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  2.4s\n",
      "2022-10-06 22:59:45,007 - pyscenic.transform - WARNING - Less than 80% of the genes in HIST2H2BE could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  2.6s\n",
      "2022-10-06 22:59:45,209 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB7B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  2.8s\n",
      "2022-10-06 22:59:45,382 - pyscenic.transform - WARNING - Less than 80% of the genes in ZC3H7A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  2.9s\n",
      "2022-10-06 22:59:45,473 - pyscenic.transform - WARNING - Less than 80% of the genes in ZCCHC17 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:45,557 - pyscenic.transform - WARNING - Less than 80% of the genes in ZDHHC24 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  3.2s\n",
      "2022-10-06 22:59:45,785 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF8 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:45,795 - pyscenic.transform - WARNING - Less than 80% of the genes in PBX3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  3.3s\n",
      "2022-10-06 22:59:45,868 - pyscenic.transform - WARNING - Less than 80% of the genes in PCK2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  3.4s\n",
      "2022-10-06 22:59:45,946 - pyscenic.transform - WARNING - Less than 80% of the genes in PDLIM5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  3.5s\n",
      "2022-10-06 22:59:46,096 - pyscenic.transform - WARNING - Less than 80% of the genes in PHLDA2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  3.6s\n",
      "2022-10-06 22:59:46,171 - pyscenic.transform - WARNING - Less than 80% of the genes in PHTF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  3.9s\n",
      "2022-10-06 22:59:46,427 - pyscenic.transform - WARNING - Less than 80% of the genes in HIVEP3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:46,493 - pyscenic.transform - WARNING - Less than 80% of the genes in HLCS could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  4.3s\n",
      "2022-10-06 22:59:46,820 - pyscenic.transform - WARNING - Less than 80% of the genes in PLAGL1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  4.4s\n",
      "2022-10-06 22:59:46,911 - pyscenic.transform - WARNING - Less than 80% of the genes in HMG20B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:46,973 - pyscenic.transform - WARNING - Less than 80% of the genes in ETFB could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  4.6s\n",
      "2022-10-06 22:59:47,135 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF9 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:47,163 - pyscenic.transform - WARNING - Less than 80% of the genes in ETS2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  4.7s\n",
      "2022-10-06 22:59:47,253 - pyscenic.transform - WARNING - Less than 80% of the genes in KMT2A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  4.9s\n",
      "2022-10-06 22:59:47,437 - pyscenic.transform - WARNING - Less than 80% of the genes in LARP4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  5.5s\n",
      "2022-10-06 22:59:48,099 - pyscenic.transform - WARNING - Less than 80% of the genes in NFKB2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  5.7s\n",
      "2022-10-06 22:59:48,261 - pyscenic.transform - WARNING - Less than 80% of the genes in PML could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:48,310 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF682 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  5.8s\n",
      "2022-10-06 22:59:48,335 - pyscenic.transform - WARNING - Less than 80% of the genes in POLD2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:48,389 - pyscenic.transform - WARNING - Less than 80% of the genes in HMGN3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:48,405 - pyscenic.transform - WARNING - Less than 80% of the genes in POLE3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  5.9s\n",
      "2022-10-06 22:59:48,474 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF687 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:48,481 - pyscenic.transform - WARNING - Less than 80% of the genes in POLE4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:48,485 - pyscenic.transform - WARNING - Less than 80% of the genes in HNRNPA0 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  6.2s\n",
      "2022-10-06 22:59:48,778 - pyscenic.transform - WARNING - Less than 80% of the genes in RELA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  6.5s\n",
      "2022-10-06 22:59:49,025 - pyscenic.transform - WARNING - Less than 80% of the genes in ZEB2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  6.7s\n",
      "2022-10-06 22:59:49,317 - pyscenic.transform - WARNING - Less than 80% of the genes in HNRNPH3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  6.9s\n",
      "2022-10-06 22:59:49,468 - pyscenic.transform - WARNING - Less than 80% of the genes in HOMEZ could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  7.4s\n",
      "2022-10-06 22:59:49,990 - pyscenic.transform - WARNING - Less than 80% of the genes in POLR2A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  7.5s\n",
      "2022-10-06 22:59:50,063 - pyscenic.transform - WARNING - Less than 80% of the genes in POLR3G could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:50,117 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF708 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  7.6s\n",
      "2022-10-06 22:59:50,147 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  7.7s\n",
      "2022-10-06 22:59:50,240 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  7.8s\n",
      "2022-10-06 22:59:50,338 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  7.9s\n",
      "2022-10-06 22:59:50,516 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF721 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  8.3s\n",
      "2022-10-06 22:59:50,903 - pyscenic.transform - WARNING - Less than 80% of the genes in LMO2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  8.6s\n",
      "2022-10-06 22:59:51,129 - pyscenic.transform - WARNING - Less than 80% of the genes in LSM6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  8.7s\n",
      "2022-10-06 22:59:51,232 - pyscenic.transform - WARNING - Less than 80% of the genes in LUZP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:51,283 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF791 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:51,318 - pyscenic.transform - WARNING - Less than 80% of the genes in LYL1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  8.8s\n",
      "2022-10-06 22:59:51,410 - pyscenic.transform - WARNING - Less than 80% of the genes in POU2AF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[############                            ] | 32% Completed |  4min  8.9s\n",
      "2022-10-06 22:59:51,483 - pyscenic.transform - WARNING - Less than 80% of the genes in POU2F2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  9.0s\n",
      "2022-10-06 22:59:51,621 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF813 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  9.1s\n",
      "2022-10-06 22:59:51,641 - pyscenic.transform - WARNING - Less than 80% of the genes in PPARG could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:51,718 - pyscenic.transform - WARNING - Less than 80% of the genes in PPP1R10 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  9.2s\n",
      "2022-10-06 22:59:51,772 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF821 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:51,794 - pyscenic.transform - WARNING - Less than 80% of the genes in PPP2R3B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  9.3s\n",
      "2022-10-06 22:59:51,871 - pyscenic.transform - WARNING - Less than 80% of the genes in PQBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:51,914 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF844 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  9.4s\n",
      "2022-10-06 22:59:51,962 - pyscenic.transform - WARNING - Less than 80% of the genes in PRDM1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:51,981 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF879 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:51,985 - pyscenic.transform - WARNING - Less than 80% of the genes in MAFF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  9.5s\n",
      "2022-10-06 22:59:52,052 - pyscenic.transform - WARNING - Less than 80% of the genes in PRDX5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:52,067 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF880 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  9.7s\n",
      "2022-10-06 22:59:52,294 - pyscenic.transform - WARNING - Less than 80% of the genes in ZFP14 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed |  4min  9.8s\n",
      "2022-10-06 22:59:52,357 - pyscenic.transform - WARNING - Less than 80% of the genes in ZRSR2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 10.1s\n",
      "2022-10-06 22:59:52,691 - pyscenic.transform - WARNING - Less than 80% of the genes in ZFP91 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 10.2s\n",
      "2022-10-06 22:59:52,765 - pyscenic.transform - WARNING - Less than 80% of the genes in ZFPM1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:52,848 - pyscenic.transform - WARNING - Less than 80% of the genes in ZFX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 10.4s\n",
      "2022-10-06 22:59:52,892 - pyscenic.transform - WARNING - Less than 80% of the genes in NONO could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:52,914 - pyscenic.transform - WARNING - Less than 80% of the genes in ZHX1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:52,939 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 10.5s\n",
      "2022-10-06 22:59:53,006 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT5A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 10.6s\n",
      "2022-10-06 22:59:53,102 - pyscenic.transform - WARNING - Less than 80% of the genes in ZKSCAN1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:53,159 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:53,173 - pyscenic.transform - WARNING - Less than 80% of the genes in ZKSCAN3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 10.7s\n",
      "2022-10-06 22:59:53,240 - pyscenic.transform - WARNING - Less than 80% of the genes in ZKSCAN5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 10.8s\n",
      "2022-10-06 22:59:53,312 - pyscenic.transform - WARNING - Less than 80% of the genes in STUB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:53,331 - pyscenic.transform - WARNING - Less than 80% of the genes in ZMAT2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:53,395 - pyscenic.transform - WARNING - Less than 80% of the genes in SUCLG1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:53,414 - pyscenic.transform - WARNING - Less than 80% of the genes in EWSR1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 10.9s\n",
      "2022-10-06 22:59:53,419 - pyscenic.transform - WARNING - Less than 80% of the genes in ZMIZ1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:53,472 - pyscenic.transform - WARNING - Less than 80% of the genes in MAFG could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:53,488 - pyscenic.transform - WARNING - Less than 80% of the genes in EXOSC3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:53,496 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF100 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 11.0s\n",
      "2022-10-06 22:59:53,563 - pyscenic.transform - WARNING - Less than 80% of the genes in SUZ12 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:53,567 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF12 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 11.2s\n",
      "2022-10-06 22:59:53,734 - pyscenic.transform - WARNING - Less than 80% of the genes in TAF9 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 11.3s\n",
      "2022-10-06 22:59:53,869 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF143 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 12.6s\n",
      "2022-10-06 22:59:55,158 - pyscenic.transform - WARNING - Less than 80% of the genes in TBL1XR1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-10-06 22:59:55,172 - pyscenic.transform - WARNING - Less than 80% of the genes in PRNP could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 12.7s\n",
      "2022-10-06 22:59:55,247 - pyscenic.transform - WARNING - Less than 80% of the genes in PSMA6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:55,326 - pyscenic.transform - WARNING - Less than 80% of the genes in PSMC2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 12.9s\n",
      "2022-10-06 22:59:55,468 - pyscenic.transform - WARNING - Less than 80% of the genes in RFXANK could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:55,495 - pyscenic.transform - WARNING - Less than 80% of the genes in PTPMT1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 13.0s\n",
      "2022-10-06 22:59:55,596 - pyscenic.transform - WARNING - Less than 80% of the genes in HP1BP3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 14.0s\n",
      "2022-10-06 22:59:56,622 - pyscenic.transform - WARNING - Less than 80% of the genes in MAGEF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:56,636 - pyscenic.transform - WARNING - Less than 80% of the genes in TCF7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 14.1s\n",
      "2022-10-06 22:59:56,677 - pyscenic.transform - WARNING - Less than 80% of the genes in EZR could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:56,715 - pyscenic.transform - WARNING - Less than 80% of the genes in MAGOH could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:56,715 - pyscenic.transform - WARNING - Less than 80% of the genes in TCF7L2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 14.2s\n",
      "2022-10-06 22:59:56,751 - pyscenic.transform - WARNING - Less than 80% of the genes in FANK1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:56,817 - pyscenic.transform - WARNING - Less than 80% of the genes in MAPK1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 14.3s\n",
      "2022-10-06 22:59:56,918 - pyscenic.transform - WARNING - Less than 80% of the genes in FHL2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 14.4s\n",
      "2022-10-06 22:59:57,003 - pyscenic.transform - WARNING - Less than 80% of the genes in FIP1L1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 15.1s\n",
      "2022-10-06 22:59:57,673 - pyscenic.transform - WARNING - Less than 80% of the genes in NR3C1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 15.3s\n",
      "2022-10-06 22:59:57,894 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF207 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 15.4s\n",
      "2022-10-06 22:59:57,968 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF217 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 15.5s\n",
      "2022-10-06 22:59:58,077 - pyscenic.transform - WARNING - Less than 80% of the genes in MAX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:58,119 - pyscenic.transform - WARNING - Less than 80% of the genes in TCF3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:58,137 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF22 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 15.7s\n",
      "2022-10-06 22:59:58,298 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF224 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:58,335 - pyscenic.transform - WARNING - Less than 80% of the genes in FLI1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 15.8s\n",
      "2022-10-06 22:59:58,387 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF226 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 16.1s\n",
      "2022-10-06 22:59:58,673 - pyscenic.transform - WARNING - Less than 80% of the genes in HSPA5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:58,697 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF24 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:58,698 - pyscenic.transform - WARNING - Less than 80% of the genes in ODC1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 16.3s\n",
      "2022-10-06 22:59:58,893 - pyscenic.transform - WARNING - Less than 80% of the genes in RNASEH2C could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 16.8s\n",
      "2022-10-06 22:59:59,334 - pyscenic.transform - WARNING - Less than 80% of the genes in MAZ could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 16.9s\n",
      "2022-10-06 22:59:59,408 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:59,504 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 17.0s\n",
      "2022-10-06 22:59:59,506 - pyscenic.transform - WARNING - Less than 80% of the genes in TCF7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:59,524 - pyscenic.transform - WARNING - Less than 80% of the genes in PBX2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 22:59:59,600 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 17.6s\n",
      "2022-10-06 23:00:00,226 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 17.8s\n",
      "2022-10-06 23:00:00,354 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2C could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 17.9s\n",
      "2022-10-06 23:00:00,432 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2D could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 18.1s\n",
      "2022-10-06 23:00:00,648 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF263 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 19.4s\n",
      "2022-10-06 23:00:01,968 - pyscenic.transform - WARNING - Less than 80% of the genes in PBX3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 19.5s\n",
      "2022-10-06 23:00:02,056 - pyscenic.transform - WARNING - Less than 80% of the genes in PCK2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:02,087 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF274 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 19.6s\n",
      "2022-10-06 23:00:02,145 - pyscenic.transform - WARNING - Less than 80% of the genes in PDLIM5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:02,151 - pyscenic.transform - WARNING - Less than 80% of the genes in FOSL1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:02,169 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF276 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:02,229 - pyscenic.transform - WARNING - Less than 80% of the genes in PDS5A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 19.8s\n",
      "2022-10-06 23:00:02,350 - pyscenic.transform - WARNING - Less than 80% of the genes in ILF2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 19.9s\n",
      "2022-10-06 23:00:02,511 - pyscenic.transform - WARNING - Less than 80% of the genes in ING4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 20.2s\n",
      "2022-10-06 23:00:02,823 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF324 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 20.4s\n",
      "2022-10-06 23:00:03,027 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF331 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 20.6s\n",
      "2022-10-06 23:00:03,204 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF33A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  4min 20.7s\n",
      "2022-10-06 23:00:03,283 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF34 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 34% Completed |  4min 21.1s\n",
      "2022-10-06 23:00:03,608 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF358 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 34% Completed |  4min 21.4s\n",
      "2022-10-06 23:00:03,966 - pyscenic.transform - WARNING - Less than 80% of the genes in MIOS could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 34% Completed |  4min 21.6s\n",
      "2022-10-06 23:00:04,180 - pyscenic.transform - WARNING - Less than 80% of the genes in MLX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 34% Completed |  4min 21.7s\n",
      "2022-10-06 23:00:04,248 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF394 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:04,263 - pyscenic.transform - WARNING - Less than 80% of the genes in MLXIP could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#############                           ] | 34% Completed |  4min 21.8s\n",
      "2022-10-06 23:00:04,342 - pyscenic.transform - WARNING - Less than 80% of the genes in MLXIPL could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 35% Completed |  4min 23.1s\n",
      "2022-10-06 23:00:05,705 - pyscenic.transform - WARNING - Less than 80% of the genes in MNT could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 35% Completed |  4min 23.2s\n",
      "2022-10-06 23:00:05,792 - pyscenic.transform - WARNING - Less than 80% of the genes in MRPL1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 35% Completed |  4min 23.3s\n",
      "2022-10-06 23:00:05,859 - pyscenic.transform - WARNING - Less than 80% of the genes in TERF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:05,902 - pyscenic.transform - WARNING - Less than 80% of the genes in MRPL2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 35% Completed |  4min 23.4s\n",
      "2022-10-06 23:00:05,985 - pyscenic.transform - WARNING - Less than 80% of the genes in PHLDA2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:06,023 - pyscenic.transform - WARNING - Less than 80% of the genes in TFAM could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 35% Completed |  4min 23.7s\n",
      "2022-10-06 23:00:06,178 - pyscenic.transform - WARNING - Less than 80% of the genes in TFCP2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:06,253 - pyscenic.transform - WARNING - Less than 80% of the genes in TFCP2L1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 25.2s\n",
      "2022-10-06 23:00:07,779 - pyscenic.transform - WARNING - Less than 80% of the genes in PKNOX1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 25.6s\n",
      "2022-10-06 23:00:08,137 - pyscenic.transform - WARNING - Less than 80% of the genes in POLD2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 26.6s\n",
      "2022-10-06 23:00:09,227 - pyscenic.transform - WARNING - Less than 80% of the genes in MSRA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 26.9s\n",
      "2022-10-06 23:00:09,396 - pyscenic.transform - WARNING - Less than 80% of the genes in TFDP2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 27.0s\n",
      "2022-10-06 23:00:09,561 - pyscenic.transform - WARNING - Less than 80% of the genes in RREB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 27.1s\n",
      "2022-10-06 23:00:09,656 - pyscenic.transform - WARNING - Less than 80% of the genes in RUFY3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 27.5s\n",
      "2022-10-06 23:00:10,230 - pyscenic.transform - WARNING - Less than 80% of the genes in POLE3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 28.5s\n",
      "2022-10-06 23:00:11,097 - pyscenic.transform - WARNING - Less than 80% of the genes in RUNX1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 28.8s\n",
      "2022-10-06 23:00:11,421 - pyscenic.transform - WARNING - Less than 80% of the genes in RUVBL1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[##############                          ] | 36% Completed |  4min 29.7s\n",
      "2022-10-06 23:00:12,273 - pyscenic.transform - WARNING - Less than 80% of the genes in POLR2A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:12,327 - pyscenic.transform - WARNING - Less than 80% of the genes in MTF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 29.8s\n",
      "2022-10-06 23:00:12,407 - pyscenic.transform - WARNING - Less than 80% of the genes in TFEB could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 29.9s\n",
      "2022-10-06 23:00:12,510 - pyscenic.transform - WARNING - Less than 80% of the genes in MXD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 30.2s\n",
      "2022-10-06 23:00:12,756 - pyscenic.transform - WARNING - Less than 80% of the genes in TGIF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 31.5s\n",
      "2022-10-06 23:00:14,077 - pyscenic.transform - WARNING - Less than 80% of the genes in THAP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 32.7s\n",
      "2022-10-06 23:00:15,306 - pyscenic.transform - WARNING - Less than 80% of the genes in THAP11 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 32.8s\n",
      "2022-10-06 23:00:15,387 - pyscenic.transform - WARNING - Less than 80% of the genes in THAP5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 32.9s\n",
      "2022-10-06 23:00:15,477 - pyscenic.transform - WARNING - Less than 80% of the genes in THOC2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 33.1s\n",
      "2022-10-06 23:00:15,648 - pyscenic.transform - WARNING - Less than 80% of the genes in NR2F6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 33.5s\n",
      "2022-10-06 23:00:16,124 - pyscenic.transform - WARNING - Less than 80% of the genes in POU2F1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 35.2s\n",
      "2022-10-06 23:00:17,836 - pyscenic.transform - WARNING - Less than 80% of the genes in SAP30 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 35.3s\n",
      "2022-10-06 23:00:17,866 - pyscenic.transform - WARNING - Less than 80% of the genes in HSF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 35.4s\n",
      "2022-10-06 23:00:18,027 - pyscenic.transform - WARNING - Less than 80% of the genes in HSF4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 35.7s\n",
      "2022-10-06 23:00:18,352 - pyscenic.transform - WARNING - Less than 80% of the genes in POU2F2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 35.9s\n",
      "2022-10-06 23:00:18,502 - pyscenic.transform - WARNING - Less than 80% of the genes in TIA1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:18,515 - pyscenic.transform - WARNING - Less than 80% of the genes in PPARA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 36.0s\n",
      "2022-10-06 23:00:18,650 - pyscenic.transform - WARNING - Less than 80% of the genes in TIMM8A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 36.2s\n",
      "2022-10-06 23:00:18,790 - pyscenic.transform - WARNING - Less than 80% of the genes in PPP1R10 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 36.4s\n",
      "2022-10-06 23:00:19,038 - pyscenic.transform - WARNING - Less than 80% of the genes in PQBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 36.5s\n",
      "2022-10-06 23:00:19,147 - pyscenic.transform - WARNING - Less than 80% of the genes in PRDM1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 36.6s\n",
      "2022-10-06 23:00:19,184 - pyscenic.transform - WARNING - Less than 80% of the genes in MYC could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 36.7s\n",
      "2022-10-06 23:00:19,358 - pyscenic.transform - WARNING - Less than 80% of the genes in PRDX5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 36.9s\n",
      "2022-10-06 23:00:19,475 - pyscenic.transform - WARNING - Less than 80% of the genes in OLIG1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 37.2s\n",
      "2022-10-06 23:00:19,851 - pyscenic.transform - WARNING - Less than 80% of the genes in PSMA6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 37.3s\n",
      "2022-10-06 23:00:19,897 - pyscenic.transform - WARNING - Less than 80% of the genes in TMEM33 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 37.5s\n",
      "2022-10-06 23:00:20,053 - pyscenic.transform - WARNING - Less than 80% of the genes in TOPORS could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 37.6s\n",
      "2022-10-06 23:00:20,138 - pyscenic.transform - WARNING - Less than 80% of the genes in TP53 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:20,159 - pyscenic.transform - WARNING - Less than 80% of the genes in PTPMT1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  4min 37.7s\n",
      "2022-10-06 23:00:20,257 - pyscenic.transform - WARNING - Less than 80% of the genes in PURA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 37% Completed |  4min 38.1s\n",
      "2022-10-06 23:00:20,659 - pyscenic.transform - WARNING - Less than 80% of the genes in RAB2A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 37% Completed |  4min 38.2s\n",
      "2022-10-06 23:00:20,780 - pyscenic.transform - WARNING - Less than 80% of the genes in RAB7A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 37% Completed |  4min 38.5s\n",
      "2022-10-06 23:00:21,125 - pyscenic.transform - WARNING - Less than 80% of the genes in SF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 37% Completed |  4min 38.7s\n",
      "2022-10-06 23:00:21,214 - pyscenic.transform - WARNING - Less than 80% of the genes in SF3B1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 41.7s\n",
      "2022-10-06 23:00:24,243 - pyscenic.transform - WARNING - Less than 80% of the genes in TRAF4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[###############                         ] | 38% Completed |  4min 41.8s\n",
      "2022-10-06 23:00:24,320 - pyscenic.transform - WARNING - Less than 80% of the genes in TRIM21 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 42.0s\n",
      "2022-10-06 23:00:24,516 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF143 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 44.1s\n",
      "2022-10-06 23:00:26,657 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF200 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 44.9s\n",
      "2022-10-06 23:00:27,404 - pyscenic.transform - WARNING - Less than 80% of the genes in TRIM33 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 45.0s\n",
      "2022-10-06 23:00:27,494 - pyscenic.transform - WARNING - Less than 80% of the genes in TRIM69 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 45.1s\n",
      "2022-10-06 23:00:27,680 - pyscenic.transform - WARNING - Less than 80% of the genes in TRMT1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 45.3s\n",
      "2022-10-06 23:00:27,814 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXO4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:27,856 - pyscenic.transform - WARNING - Less than 80% of the genes in TRPS1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 45.4s\n",
      "2022-10-06 23:00:27,945 - pyscenic.transform - WARNING - Less than 80% of the genes in TSC22D4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 45.5s\n",
      "2022-10-06 23:00:28,028 - pyscenic.transform - WARNING - Less than 80% of the genes in TSN could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 45.6s\n",
      "2022-10-06 23:00:28,107 - pyscenic.transform - WARNING - Less than 80% of the genes in TSNAX could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 45.7s\n",
      "2022-10-06 23:00:28,211 - pyscenic.transform - WARNING - Less than 80% of the genes in U2AF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 46.2s\n",
      "2022-10-06 23:00:28,736 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 47.5s\n",
      "2022-10-06 23:00:30,024 - pyscenic.transform - WARNING - Less than 80% of the genes in UBE2K could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:30,098 - pyscenic.transform - WARNING - Less than 80% of the genes in UBE2V1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 47.7s\n",
      "2022-10-06 23:00:30,245 - pyscenic.transform - WARNING - Less than 80% of the genes in UBTF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 47.8s\n",
      "2022-10-06 23:00:30,370 - pyscenic.transform - WARNING - Less than 80% of the genes in UBXN1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 48.3s\n",
      "2022-10-06 23:00:30,810 - pyscenic.transform - WARNING - Less than 80% of the genes in RFX1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 49.0s\n",
      "2022-10-06 23:00:31,563 - pyscenic.transform - WARNING - Less than 80% of the genes in MYNN could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 49.1s\n",
      "2022-10-06 23:00:31,673 - pyscenic.transform - WARNING - Less than 80% of the genes in USP39 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 49.2s\n",
      "2022-10-06 23:00:31,760 - pyscenic.transform - WARNING - Less than 80% of the genes in UTP18 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 49.3s\n",
      "2022-10-06 23:00:31,848 - pyscenic.transform - WARNING - Less than 80% of the genes in VAMP3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 49.4s\n",
      "2022-10-06 23:00:31,927 - pyscenic.transform - WARNING - Less than 80% of the genes in VDR could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 49.5s\n",
      "2022-10-06 23:00:32,074 - pyscenic.transform - WARNING - Less than 80% of the genes in VEZF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 49.6s\n",
      "2022-10-06 23:00:32,153 - pyscenic.transform - WARNING - Less than 80% of the genes in VPS4B could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 49.7s\n",
      "2022-10-06 23:00:32,233 - pyscenic.transform - WARNING - Less than 80% of the genes in VPS72 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 49.9s\n",
      "2022-10-06 23:00:32,411 - pyscenic.transform - WARNING - Less than 80% of the genes in XBP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:32,493 - pyscenic.transform - WARNING - Less than 80% of the genes in XPA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:32,495 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF248 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 50.5s\n",
      "2022-10-06 23:00:33,012 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF254 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:33,035 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 51.7s\n",
      "2022-10-06 23:00:34,273 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 51.9s\n",
      "2022-10-06 23:00:34,448 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 52.0s\n",
      "2022-10-06 23:00:34,546 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAP2 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 54.2s\n",
      "2022-10-06 23:00:36,762 - pyscenic.transform - WARNING - Less than 80% of the genes in YEATS4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 54.3s\n",
      "2022-10-06 23:00:36,842 - pyscenic.transform - WARNING - Less than 80% of the genes in YOD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[###############                         ] | 38% Completed |  4min 56.0s\n",
      "2022-10-06 23:00:38,608 - pyscenic.transform - WARNING - Less than 80% of the genes in YY1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 56.1s\n",
      "2022-10-06 23:00:38,681 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBED1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 56.2s\n",
      "2022-10-06 23:00:38,778 - pyscenic.transform - WARNING - Less than 80% of the genes in CBX7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 56.3s\n",
      "2022-10-06 23:00:38,886 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF317 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 56.5s\n",
      "2022-10-06 23:00:39,124 - pyscenic.transform - WARNING - Less than 80% of the genes in SMARCB1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  4min 56.6s\n",
      "2022-10-06 23:00:39,213 - pyscenic.transform - WARNING - Less than 80% of the genes in SMARCC1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  4min 56.9s\n",
      "2022-10-06 23:00:39,381 - pyscenic.transform - WARNING - Less than 80% of the genes in SMUG1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:39,472 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAI1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  4min 57.1s\n",
      "2022-10-06 23:00:39,687 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAPC4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  4min 57.2s\n",
      "2022-10-06 23:00:39,761 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAPC5 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  4min 57.3s\n",
      "2022-10-06 23:00:39,845 - pyscenic.transform - WARNING - Less than 80% of the genes in SND1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  4min 57.4s\n",
      "2022-10-06 23:00:39,936 - pyscenic.transform - WARNING - Less than 80% of the genes in SNRNP70 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  4min 57.5s\n",
      "2022-10-06 23:00:40,132 - pyscenic.transform - WARNING - Less than 80% of the genes in SOCS4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  4min 58.0s\n",
      "2022-10-06 23:00:40,570 - pyscenic.transform - WARNING - Less than 80% of the genes in SOX4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  4min 58.4s\n",
      "2022-10-06 23:00:41,008 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF35 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  4min 59.8s\n",
      "2022-10-06 23:00:42,361 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF429 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  5min  0.0s\n",
      "2022-10-06 23:00:42,641 - pyscenic.transform - WARNING - Less than 80% of the genes in GLI4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed |  5min  0.6s\n",
      "2022-10-06 23:00:43,215 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF354C could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  2.2s\n",
      "2022-10-06 23:00:44,848 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF598 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  2.3s\n",
      "2022-10-06 23:00:44,939 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF599 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  2.4s\n",
      "2022-10-06 23:00:45,020 - pyscenic.transform - WARNING - Less than 80% of the genes in SP1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  2.8s\n",
      "2022-10-06 23:00:45,356 - pyscenic.transform - WARNING - Less than 80% of the genes in SP3 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  3.6s\n",
      "2022-10-06 23:00:46,235 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBED1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  4.0s\n",
      "2022-10-06 23:00:46,590 - pyscenic.transform - WARNING - Less than 80% of the genes in SP4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:00:46,620 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB11 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  4.1s\n",
      "2022-10-06 23:00:46,684 - pyscenic.transform - WARNING - Less than 80% of the genes in SPAG7 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  5.7s\n",
      "2022-10-06 23:00:48,203 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF669 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  6.2s\n",
      "2022-10-06 23:00:48,725 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF514 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  8.3s\n",
      "2022-10-06 23:00:50,848 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF562 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  8.4s\n",
      "2022-10-06 23:00:50,900 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB26 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  9.3s\n",
      "2022-10-06 23:00:51,857 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF577 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min  9.5s\n",
      "2022-10-06 23:00:52,067 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF580 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min 10.8s\n",
      "2022-10-06 23:00:53,345 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF75A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min 12.3s\n",
      "2022-10-06 23:00:54,808 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF846 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min 13.1s\n",
      "2022-10-06 23:00:55,684 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF93 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min 14.0s\n",
      "2022-10-06 23:00:56,539 - pyscenic.transform - WARNING - Less than 80% of the genes in ZSCAN29 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 40% Completed |  5min 14.3s\n",
      "2022-10-06 23:00:56,813 - pyscenic.transform - WARNING - Less than 80% of the genes in ZSCAN5A could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 41% Completed |  5min 15.0s\n",
      "2022-10-06 23:00:57,538 - pyscenic.transform - WARNING - Less than 80% of the genes in PTCD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 41% Completed |  5min 16.1s\n",
      "2022-10-06 23:00:58,654 - pyscenic.transform - WARNING - Less than 80% of the genes in SRBD1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[################                        ] | 41% Completed |  5min 17.0s\n",
      "2022-10-06 23:00:59,504 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAPC4 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#################                       ] | 44% Completed |  5min 23.2s\n",
      "2022-10-06 23:01:05,850 - pyscenic.transform - WARNING - Less than 80% of the genes in SRF could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[#################                       ] | 44% Completed |  5min 25.1s\n",
      "2022-10-06 23:01:07,713 - pyscenic.transform - WARNING - Less than 80% of the genes in HES6 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##################                      ] | 46% Completed |  5min 29.7s\n",
      "2022-10-06 23:01:12,358 - pyscenic.transform - WARNING - Less than 80% of the genes in TBX18 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[##################                      ] | 46% Completed |  5min 40.6s\n",
      "2022-10-06 23:01:23,185 - pyscenic.transform - WARNING - Less than 80% of the genes in TERF1 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###################                     ] | 49% Completed |  5min 58.2s\n",
      "2022-10-06 23:01:40,733 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF354C could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################                     ] | 49% Completed |  6min  0.2s\n",
      "2022-10-06 23:01:42,744 - pyscenic.transform - WARNING - Less than 80% of the genes in THRA could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[###################                     ] | 49% Completed |  6min  1.4s\n",
      "2022-10-06 23:01:43,934 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for FOXS1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################                    ] | 50% Completed |  6min  8.4s\n",
      "2022-10-06 23:01:50,916 - pyscenic.transform - WARNING - Less than 80% of the genes in TRIM33 could be mapped to hg19-500bp-upstream-7species.mc9nr. Skipping this module.\n",
      "[####################                    ] | 52% Completed |  6min 29.8s\n",
      "2022-10-06 23:02:12,420 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for AGAP2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################                    ] | 52% Completed |  6min 30.1s\n",
      "2022-10-06 23:02:12,695 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for AHCTF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################                    ] | 52% Completed |  6min 36.3s\n",
      "2022-10-06 23:02:18,794 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for E4F1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 53% Completed |  6min 47.5s\n",
      "2022-10-06 23:02:29,989 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for CBX7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed |  7min  0.8s\n",
      "2022-10-06 23:02:43,363 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for POU6F1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed |  7min 10.6s\n",
      "2022-10-06 23:02:53,097 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for PTCD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed |  7min 12.4s\n",
      "2022-10-06 23:02:54,961 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for THRA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 57% Completed |  7min 23.6s\n",
      "2022-10-06 23:03:06,184 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZFP82 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 57% Completed |  7min 29.7s\n",
      "2022-10-06 23:03:12,188 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for NR2F6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 58% Completed |  7min 34.6s\n",
      "2022-10-06 23:03:17,191 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF181 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 58% Completed |  7min 35.6s\n",
      "2022-10-06 23:03:18,175 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for DIABLO could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 58% Completed |  7min 35.8s\n",
      "2022-10-06 23:03:18,370 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for MLXIP could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 58% Completed |  7min 35.9s\n",
      "2022-10-06 23:03:18,575 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF184 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 58% Completed |  7min 38.5s\n",
      "2022-10-06 23:03:21,099 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for OLIG1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed |  7min 43.0s\n",
      "2022-10-06 23:03:25,572 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF75A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed |  7min 44.4s\n",
      "2022-10-06 23:03:26,964 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF248 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed |  7min 45.0s\n",
      "2022-10-06 23:03:27,607 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF333 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed |  7min 45.1s\n",
      "2022-10-06 23:03:27,691 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF577 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed |  7min 45.7s\n",
      "2022-10-06 23:03:28,363 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF35 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[########################                ] | 60% Completed |  7min 46.1s\n",
      "2022-10-06 23:03:28,627 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF589 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed |  7min 46.2s\n",
      "2022-10-06 23:03:28,730 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF354C could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed |  7min 46.6s\n",
      "2022-10-06 23:03:29,199 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF599 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed |  7min 47.8s\n",
      "2022-10-06 23:03:30,353 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF254 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed |  7min 49.8s\n",
      "2022-10-06 23:03:32,447 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF846 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed |  7min 54.6s\n",
      "2022-10-06 23:03:37,128 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZBTB11 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed |  7min 55.5s\n",
      "2022-10-06 23:03:38,109 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZBTB26 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################                ] | 62% Completed |  7min 56.7s\n",
      "2022-10-06 23:03:39,206 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for AHCTF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#########################               ] | 64% Completed |  8min  1.1s\n",
      "2022-10-06 23:03:43,706 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF525 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed |  8min  6.5s\n",
      "2022-10-06 23:03:49,072 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for PEG3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed |  8min  6.7s\n",
      "2022-10-06 23:03:49,220 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for MLXIP could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed |  8min  6.8s\n",
      "2022-10-06 23:03:49,270 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for STAT3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:03:49,308 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for MLXIPL could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed |  8min 10.0s\n",
      "2022-10-06 23:03:52,583 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for MRPL2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 67% Completed |  8min 17.5s\n",
      "2022-10-06 23:04:00,017 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for GMEB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 67% Completed |  8min 19.3s\n",
      "2022-10-06 23:04:01,841 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF184 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 67% Completed |  8min 21.4s\n",
      "2022-10-06 23:04:03,991 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZBTB11 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 67% Completed |  8min 22.3s\n",
      "2022-10-06 23:04:04,871 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZBTB26 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 67% Completed |  8min 24.0s\n",
      "2022-10-06 23:04:06,634 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for SF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed |  8min 24.6s\n",
      "2022-10-06 23:04:07,133 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF24 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed |  8min 28.2s\n",
      "2022-10-06 23:04:10,774 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF263 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed |  8min 30.1s\n",
      "2022-10-06 23:04:12,644 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF430 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed |  8min 30.9s\n",
      "2022-10-06 23:04:13,448 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for PPP1R10 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed |  8min 33.8s\n",
      "2022-10-06 23:04:16,389 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF34 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed |  8min 35.4s\n",
      "2022-10-06 23:04:17,928 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for PTCD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed |  8min 37.6s\n",
      "2022-10-06 23:04:20,130 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for BPTF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed |  8min 38.1s\n",
      "2022-10-06 23:04:20,727 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF562 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed |  8min 40.6s\n",
      "2022-10-06 23:04:23,149 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF577 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 42.3s\n",
      "2022-10-06 23:04:24,900 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF599 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 42.7s\n",
      "2022-10-06 23:04:25,350 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for TBX18 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 43.0s\n",
      "2022-10-06 23:04:25,595 - pyscenic.transform - WARNING - Less than 80% of the genes in CBX3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 43.1s\n",
      "2022-10-06 23:04:25,667 - pyscenic.transform - WARNING - Less than 80% of the genes in CCDC25 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:25,738 - pyscenic.transform - WARNING - Less than 80% of the genes in CCNT2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[############################            ] | 70% Completed |  8min 43.6s\n",
      "2022-10-06 23:04:26,188 - pyscenic.transform - WARNING - Less than 80% of the genes in CDC5L could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:26,264 - pyscenic.transform - WARNING - Less than 80% of the genes in CDK2AP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 44.1s\n",
      "2022-10-06 23:04:26,568 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPB could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 45.4s\n",
      "2022-10-06 23:04:27,884 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF669 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 45.7s\n",
      "2022-10-06 23:04:28,241 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPD could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 45.9s\n",
      "2022-10-06 23:04:28,411 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPZ could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:28,477 - pyscenic.transform - WARNING - Less than 80% of the genes in CERS2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 46.1s\n",
      "2022-10-06 23:04:28,583 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF682 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 46.2s\n",
      "2022-10-06 23:04:28,760 - pyscenic.transform - WARNING - Less than 80% of the genes in CHD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 46.3s\n",
      "2022-10-06 23:04:28,848 - pyscenic.transform - WARNING - Less than 80% of the genes in CHD2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 46.4s\n",
      "2022-10-06 23:04:28,951 - pyscenic.transform - WARNING - Less than 80% of the genes in CHURC1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 46.5s\n",
      "2022-10-06 23:04:29,042 - pyscenic.transform - WARNING - Less than 80% of the genes in CLK1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 46.6s\n",
      "2022-10-06 23:04:29,115 - pyscenic.transform - WARNING - Less than 80% of the genes in CNOT4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 47.2s\n",
      "2022-10-06 23:04:29,780 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for SNAPC4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 47.3s\n",
      "2022-10-06 23:04:29,853 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for CBX7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 47.4s\n",
      "2022-10-06 23:04:29,946 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for SND1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 47.7s\n",
      "2022-10-06 23:04:30,277 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for CDK2AP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 47.8s\n",
      "2022-10-06 23:04:30,354 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZNF846 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 48.2s\n",
      "2022-10-06 23:04:30,705 - pyscenic.transform - WARNING - Less than 80% of the genes in CPSF4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 49.4s\n",
      "2022-10-06 23:04:31,908 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZSCAN25 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 49.7s\n",
      "2022-10-06 23:04:32,244 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for ZSCAN5A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 50.2s\n",
      "2022-10-06 23:04:32,764 - pyscenic.transform - WARNING - Less than 80% of the genes in ACAA1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 51.4s\n",
      "2022-10-06 23:04:33,964 - pyscenic.transform - WARNING - Less than 80% of the genes in APEX1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 51.5s\n",
      "2022-10-06 23:04:34,036 - pyscenic.transform - WARNING - Less than 80% of the genes in ARG2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 51.7s\n",
      "2022-10-06 23:04:34,258 - pyscenic.transform - WARNING - Less than 80% of the genes in ARID5B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed |  8min 52.8s\n",
      "2022-10-06 23:04:35,340 - pyscenic.transform - WARNING - Less than 80% of the genes in CREB5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 72% Completed |  8min 53.8s\n",
      "2022-10-06 23:04:36,347 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for SOX4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 72% Completed |  8min 53.9s\n",
      "2022-10-06 23:04:36,506 - pyscenic.transform - WARNING - Less than 80% of the genes in H1FX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 72% Completed |  8min 54.0s\n",
      "2022-10-06 23:04:36,608 - pyscenic.transform - WARNING - Less than 80% of the genes in CREBL2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[############################            ] | 72% Completed |  8min 54.4s\n",
      "2022-10-06 23:04:36,897 - pyscenic.transform - WARNING - Less than 80% of the genes in HBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 54.6s\n",
      "2022-10-06 23:04:37,152 - pyscenic.transform - WARNING - Less than 80% of the genes in HDAC2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 54.7s\n",
      "2022-10-06 23:04:37,228 - pyscenic.transform - WARNING - Less than 80% of the genes in HDAC3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 54.8s\n",
      "2022-10-06 23:04:37,323 - pyscenic.transform - WARNING - Less than 80% of the genes in HES1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 54.9s\n",
      "2022-10-06 23:04:37,415 - pyscenic.transform - WARNING - Less than 80% of the genes in HES4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-10-06 23:04:37,456 - pyscenic.transform - WARNING - Less than 80% of the genes in ASCL2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 55.0s\n",
      "2022-10-06 23:04:37,590 - pyscenic.transform - WARNING - Less than 80% of the genes in HHEX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 55.2s\n",
      "2022-10-06 23:04:37,785 - pyscenic.transform - WARNING - Less than 80% of the genes in HIF1A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 55.3s\n",
      "2022-10-06 23:04:37,865 - pyscenic.transform - WARNING - Less than 80% of the genes in HIST2H2BE could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 55.5s\n",
      "2022-10-06 23:04:38,077 - pyscenic.transform - WARNING - Less than 80% of the genes in RAB2A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 55.6s\n",
      "2022-10-06 23:04:38,168 - pyscenic.transform - WARNING - Less than 80% of the genes in RAB7A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 55.9s\n",
      "2022-10-06 23:04:38,492 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for THAP5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 56.1s\n",
      "2022-10-06 23:04:38,708 - pyscenic.transform - WARNING - Less than 80% of the genes in RBBP5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:38,745 - pyscenic.transform - WARNING - Less than 80% of the genes in MGA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 56.3s\n",
      "2022-10-06 23:04:38,804 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:38,814 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM17 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 56.5s\n",
      "2022-10-06 23:04:39,062 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM42 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 56.6s\n",
      "2022-10-06 23:04:39,146 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM8A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 56.7s\n",
      "2022-10-06 23:04:39,233 - pyscenic.transform - WARNING - Less than 80% of the genes in RBMS1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 57.6s\n",
      "2022-10-06 23:04:40,140 - pyscenic.transform - WARNING - Less than 80% of the genes in CSNK2B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 57.7s\n",
      "2022-10-06 23:04:40,270 - pyscenic.transform - WARNING - Less than 80% of the genes in MLX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 57.8s\n",
      "2022-10-06 23:04:40,287 - pyscenic.transform - WARNING - Less than 80% of the genes in CTBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:40,343 - pyscenic.transform - WARNING - Less than 80% of the genes in MLXIP could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 57.9s\n",
      "2022-10-06 23:04:40,375 - pyscenic.transform - WARNING - Less than 80% of the genes in CTBP2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:40,422 - pyscenic.transform - WARNING - Less than 80% of the genes in MRPL1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 58.0s\n",
      "2022-10-06 23:04:40,496 - pyscenic.transform - WARNING - Less than 80% of the genes in MRPL2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 58.2s\n",
      "2022-10-06 23:04:40,762 - pyscenic.transform - WARNING - Less than 80% of the genes in MSRA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 58.4s\n",
      "2022-10-06 23:04:40,912 - pyscenic.transform - WARNING - Less than 80% of the genes in MTF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 58.5s\n",
      "2022-10-06 23:04:40,997 - pyscenic.transform - WARNING - Less than 80% of the genes in MXD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 59.0s\n",
      "2022-10-06 23:04:41,515 - pyscenic.transform - WARNING - Less than 80% of the genes in HIVEP3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 59.2s\n",
      "2022-10-06 23:04:41,683 - pyscenic.transform - WARNING - Less than 80% of the genes in HMG20B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:41,772 - pyscenic.transform - WARNING - Less than 80% of the genes in HMGA1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 59.3s\n",
      "2022-10-06 23:04:41,851 - pyscenic.transform - WARNING - Less than 80% of the genes in CTCF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 59.4s\n",
      "2022-10-06 23:04:41,957 - pyscenic.transform - WARNING - Less than 80% of the genes in CTNNB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 59.8s\n",
      "2022-10-06 23:04:42,308 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  8min 59.9s\n",
      "2022-10-06 23:04:42,430 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  0.3s\n",
      "2022-10-06 23:04:42,802 - pyscenic.transform - WARNING - Less than 80% of the genes in HNRNPA0 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  0.8s\n",
      "2022-10-06 23:04:43,315 - pyscenic.transform - WARNING - Less than 80% of the genes in HNRNPH3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  0.9s\n",
      "2022-10-06 23:04:43,392 - pyscenic.transform - WARNING - Less than 80% of the genes in HOMEZ could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  1.1s\n",
      "2022-10-06 23:04:43,630 - pyscenic.transform - WARNING - Less than 80% of the genes in HOXB6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  1.2s\n",
      "2022-10-06 23:04:43,736 - pyscenic.transform - WARNING - Less than 80% of the genes in HP1BP3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:43,744 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  1.3s\n",
      "2022-10-06 23:04:43,830 - pyscenic.transform - WARNING - Less than 80% of the genes in HSPA5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:43,848 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF6B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  1.6s\n",
      "2022-10-06 23:04:44,145 - pyscenic.transform - WARNING - Less than 80% of the genes in ID2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  1.7s\n",
      "2022-10-06 23:04:44,324 - pyscenic.transform - WARNING - Less than 80% of the genes in IKZF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  1.8s\n",
      "2022-10-06 23:04:44,396 - pyscenic.transform - WARNING - Less than 80% of the genes in IKZF4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  1.9s\n",
      "2022-10-06 23:04:44,479 - pyscenic.transform - WARNING - Less than 80% of the genes in ILF2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  2.1s\n",
      "2022-10-06 23:04:44,711 - pyscenic.transform - WARNING - Less than 80% of the genes in MYC could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  2.2s\n",
      "2022-10-06 23:04:44,793 - pyscenic.transform - WARNING - Less than 80% of the genes in MYLK could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  2.6s\n",
      "2022-10-06 23:04:45,089 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  2.7s\n",
      "2022-10-06 23:04:45,219 - pyscenic.transform - WARNING - Less than 80% of the genes in CYCS could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  2.9s\n",
      "2022-10-06 23:04:45,461 - pyscenic.transform - WARNING - Less than 80% of the genes in DAZAP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  3.4s\n",
      "2022-10-06 23:04:45,924 - pyscenic.transform - WARNING - Less than 80% of the genes in TFAM could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  3.9s\n",
      "2022-10-06 23:04:46,434 - pyscenic.transform - WARNING - Less than 80% of the genes in BACH1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:46,509 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for MAGEF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  4.1s\n",
      "2022-10-06 23:04:46,708 - pyscenic.transform - WARNING - Less than 80% of the genes in DBP could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  4.6s\n",
      "2022-10-06 23:04:47,222 - pyscenic.transform - WARNING - Less than 80% of the genes in RFXANK could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  4.9s\n",
      "2022-10-06 23:04:47,463 - pyscenic.transform - WARNING - Less than 80% of the genes in RNASEH2C could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  5.0s\n",
      "2022-10-06 23:04:47,552 - pyscenic.transform - WARNING - Less than 80% of the genes in RNF114 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  5.5s\n",
      "2022-10-06 23:04:48,110 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for NR2F6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  6.0s\n",
      "2022-10-06 23:04:48,539 - pyscenic.transform - WARNING - Less than 80% of the genes in NCOR1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:48,620 - pyscenic.transform - WARNING - Less than 80% of the genes in NCOR2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  6.1s\n",
      "2022-10-06 23:04:48,706 - pyscenic.transform - WARNING - Less than 80% of the genes in NELFE could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  6.2s\n",
      "2022-10-06 23:04:48,784 - pyscenic.transform - WARNING - Less than 80% of the genes in NF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  6.3s\n",
      "2022-10-06 23:04:48,855 - pyscenic.transform - WARNING - Less than 80% of the genes in NFAT5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed |  9min  6.4s\n",
      "2022-10-06 23:04:48,945 - pyscenic.transform - WARNING - Less than 80% of the genes in TFDP2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:49,019 - pyscenic.transform - WARNING - Less than 80% of the genes in TFEB could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  6.8s\n",
      "2022-10-06 23:04:49,443 - pyscenic.transform - WARNING - Less than 80% of the genes in BAD could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  7.1s\n",
      "2022-10-06 23:04:49,644 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  7.3s\n",
      "2022-10-06 23:04:49,856 - pyscenic.transform - WARNING - Less than 80% of the genes in DHX36 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  7.4s\n",
      "2022-10-06 23:04:49,924 - pyscenic.transform - WARNING - Less than 80% of the genes in DMAP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:49,992 - pyscenic.transform - WARNING - Less than 80% of the genes in DNAJC21 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  7.5s\n",
      "2022-10-06 23:04:50,032 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF451 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  7.7s\n",
      "2022-10-06 23:04:50,225 - pyscenic.transform - WARNING - Less than 80% of the genes in DUSP22 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[#############################           ] | 74% Completed |  9min  7.9s\n",
      "2022-10-06 23:04:50,452 - pyscenic.transform - WARNING - Less than 80% of the genes in TGIF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  8.0s\n",
      "2022-10-06 23:04:50,526 - pyscenic.transform - WARNING - Less than 80% of the genes in THAP5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:50,603 - pyscenic.transform - WARNING - Less than 80% of the genes in THOC2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  8.1s\n",
      "2022-10-06 23:04:50,675 - pyscenic.transform - WARNING - Less than 80% of the genes in TIA1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  8.2s\n",
      "2022-10-06 23:04:50,751 - pyscenic.transform - WARNING - Less than 80% of the genes in TMEM33 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  8.3s\n",
      "2022-10-06 23:04:50,823 - pyscenic.transform - WARNING - Less than 80% of the genes in TP53 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  8.4s\n",
      "2022-10-06 23:04:50,938 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  9.1s\n",
      "2022-10-06 23:04:51,688 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF467 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  9.2s\n",
      "2022-10-06 23:04:51,774 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF493 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  9.3s\n",
      "2022-10-06 23:04:51,856 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF503 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  9.5s\n",
      "2022-10-06 23:04:52,018 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF524 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  9.6s\n",
      "2022-10-06 23:04:52,201 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  9.7s\n",
      "2022-10-06 23:04:52,280 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  9.8s\n",
      "2022-10-06 23:04:52,357 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF581 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min  9.9s\n",
      "2022-10-06 23:04:52,432 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF585B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#############################           ] | 74% Completed |  9min 10.0s\n",
      "2022-10-06 23:04:52,560 - pyscenic.transform - WARNING - Less than 80% of the genes in BAX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 10.1s\n",
      "2022-10-06 23:04:52,652 - pyscenic.transform - WARNING - Less than 80% of the genes in BBX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:52,672 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF622 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 10.2s\n",
      "2022-10-06 23:04:52,732 - pyscenic.transform - WARNING - Less than 80% of the genes in BCL11A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 10.3s\n",
      "2022-10-06 23:04:52,841 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF641 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:52,928 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF652 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 10.4s\n",
      "2022-10-06 23:04:52,950 - pyscenic.transform - WARNING - Less than 80% of the genes in RREB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:52,998 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF654 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:53,037 - pyscenic.transform - WARNING - Less than 80% of the genes in RUFY3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 10.5s\n",
      "2022-10-06 23:04:53,086 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF655 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 10.9s\n",
      "2022-10-06 23:04:53,397 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF677 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 11.1s\n",
      "2022-10-06 23:04:53,662 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF688 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 11.2s\n",
      "2022-10-06 23:04:53,718 - pyscenic.transform - WARNING - Less than 80% of the genes in CCDC25 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 11.3s\n",
      "2022-10-06 23:04:53,794 - pyscenic.transform - WARNING - Less than 80% of the genes in EEF1D could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:53,848 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF708 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 11.5s\n",
      "2022-10-06 23:04:53,998 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF720 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:54,073 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF785 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 11.6s\n",
      "2022-10-06 23:04:54,112 - pyscenic.transform - WARNING - Less than 80% of the genes in BCL3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 11.7s\n",
      "2022-10-06 23:04:54,216 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF791 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:54,290 - pyscenic.transform - WARNING - Less than 80% of the genes in BCLAF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 11.9s\n",
      "2022-10-06 23:04:54,412 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF808 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-10-06 23:04:54,496 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF880 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 12.0s\n",
      "2022-10-06 23:04:54,555 - pyscenic.transform - WARNING - Less than 80% of the genes in RUNX1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 12.1s\n",
      "2022-10-06 23:04:54,638 - pyscenic.transform - WARNING - Less than 80% of the genes in ZSCAN25 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:54,642 - pyscenic.transform - WARNING - Less than 80% of the genes in CDC5L could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 12.2s\n",
      "2022-10-06 23:04:54,717 - pyscenic.transform - WARNING - Less than 80% of the genes in ZZZ3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:54,726 - pyscenic.transform - WARNING - Less than 80% of the genes in CDK2AP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 12.3s\n",
      "2022-10-06 23:04:54,877 - pyscenic.transform - WARNING - Less than 80% of the genes in ACAA1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 12.4s\n",
      "2022-10-06 23:04:54,906 - pyscenic.transform - WARNING - Less than 80% of the genes in TRIM69 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:54,956 - pyscenic.transform - WARNING - Less than 80% of the genes in ACO1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:55,047 - pyscenic.transform - WARNING - Less than 80% of the genes in ADNP could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:55,067 - pyscenic.transform - WARNING - Less than 80% of the genes in TRMT1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:55,143 - pyscenic.transform - WARNING - Less than 80% of the genes in TROVE2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:55,196 - pyscenic.transform - WARNING - Less than 80% of the genes in EGR1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:55,221 - pyscenic.transform - WARNING - Less than 80% of the genes in TRPS1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:55,233 - pyscenic.transform - WARNING - Less than 80% of the genes in AFF4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 12.7s\n",
      "2022-10-06 23:04:55,273 - pyscenic.transform - WARNING - Less than 80% of the genes in EGR2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:55,301 - pyscenic.transform - WARNING - Less than 80% of the genes in TSC22D4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:55,317 - pyscenic.transform - WARNING - Less than 80% of the genes in AGGF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 12.8s\n",
      "2022-10-06 23:04:55,368 - pyscenic.transform - WARNING - Less than 80% of the genes in TSN could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:55,398 - pyscenic.transform - WARNING - Less than 80% of the genes in AHCTF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed |  9min 12.9s\n",
      "2022-10-06 23:04:55,466 - pyscenic.transform - WARNING - Less than 80% of the genes in U2AF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:55,544 - pyscenic.transform - WARNING - Less than 80% of the genes in ELF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 13.0s\n",
      "2022-10-06 23:04:55,670 - pyscenic.transform - WARNING - Less than 80% of the genes in BHLHE40 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 13.2s\n",
      "2022-10-06 23:04:55,775 - pyscenic.transform - WARNING - Less than 80% of the genes in BHLHE41 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 13.3s\n",
      "2022-10-06 23:04:55,867 - pyscenic.transform - WARNING - Less than 80% of the genes in BPTF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 14.0s\n",
      "2022-10-06 23:04:56,583 - pyscenic.transform - WARNING - Less than 80% of the genes in UBE2K could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 14.1s\n",
      "2022-10-06 23:04:56,682 - pyscenic.transform - WARNING - Less than 80% of the genes in UBXN1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 14.3s\n",
      "2022-10-06 23:04:56,879 - pyscenic.transform - WARNING - Less than 80% of the genes in APEX1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 14.4s\n",
      "2022-10-06 23:04:56,951 - pyscenic.transform - WARNING - Less than 80% of the genes in APEX2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:57,021 - pyscenic.transform - WARNING - Less than 80% of the genes in ARFGAP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 14.5s\n",
      "2022-10-06 23:04:57,098 - pyscenic.transform - WARNING - Less than 80% of the genes in ARG2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 14.6s\n",
      "2022-10-06 23:04:57,176 - pyscenic.transform - WARNING - Less than 80% of the genes in ARID3A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 14.7s\n",
      "2022-10-06 23:04:57,285 - pyscenic.transform - WARNING - Less than 80% of the genes in USP39 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 14.8s\n",
      "2022-10-06 23:04:57,363 - pyscenic.transform - WARNING - Less than 80% of the genes in UTP18 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 14.9s\n",
      "2022-10-06 23:04:57,440 - pyscenic.transform - WARNING - Less than 80% of the genes in VAMP3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:57,461 - pyscenic.transform - WARNING - Less than 80% of the genes in ARID5B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:57,512 - pyscenic.transform - WARNING - Less than 80% of the genes in VDR could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 15.1s\n",
      "2022-10-06 23:04:57,657 - pyscenic.transform - WARNING - Less than 80% of the genes in VPS72 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 15.2s\n",
      "2022-10-06 23:04:57,801 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXD4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 15.4s\n",
      "2022-10-06 23:04:57,963 - pyscenic.transform - WARNING - Less than 80% of the genes in SAP30 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 15.5s\n",
      "2022-10-06 23:04:58,130 - pyscenic.transform - WARNING - Less than 80% of the genes in SEMA4A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 15.7s\n",
      "2022-10-06 23:04:58,300 - pyscenic.transform - WARNING - Less than 80% of the genes in NFIA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 15.8s\n",
      "2022-10-06 23:04:58,423 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPD could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 15.9s\n",
      "2022-10-06 23:04:58,480 - pyscenic.transform - WARNING - Less than 80% of the genes in NFIC could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:58,510 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPG could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 16.0s\n",
      "2022-10-06 23:04:58,556 - pyscenic.transform - WARNING - Less than 80% of the genes in NFIL3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:58,602 - pyscenic.transform - WARNING - Less than 80% of the genes in CEBPZ could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 16.2s\n",
      "2022-10-06 23:04:58,762 - pyscenic.transform - WARNING - Less than 80% of the genes in CERS2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 16.3s\n",
      "2022-10-06 23:04:58,877 - pyscenic.transform - WARNING - Less than 80% of the genes in C19orf25 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:58,907 - pyscenic.transform - WARNING - Less than 80% of the genes in CERS5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 16.4s\n",
      "2022-10-06 23:04:58,999 - pyscenic.transform - WARNING - Less than 80% of the genes in JAZF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 16.5s\n",
      "2022-10-06 23:04:59,050 - pyscenic.transform - WARNING - Less than 80% of the genes in XBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:59,122 - pyscenic.transform - WARNING - Less than 80% of the genes in XPA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 16.6s\n",
      "2022-10-06 23:04:59,170 - pyscenic.transform - WARNING - Less than 80% of the genes in CAT could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 16.7s\n",
      "2022-10-06 23:04:59,249 - pyscenic.transform - WARNING - Less than 80% of the genes in CBFB could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:59,261 - pyscenic.transform - WARNING - Less than 80% of the genes in CHD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 16.8s\n",
      "2022-10-06 23:04:59,395 - pyscenic.transform - WARNING - Less than 80% of the genes in SETDB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:59,452 - pyscenic.transform - WARNING - Less than 80% of the genes in EP300 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 16.9s\n",
      "2022-10-06 23:04:59,489 - pyscenic.transform - WARNING - Less than 80% of the genes in SF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:04:59,569 - pyscenic.transform - WARNING - Less than 80% of the genes in SF3B1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed |  9min 17.2s\n",
      "2022-10-06 23:04:59,688 - pyscenic.transform - WARNING - Less than 80% of the genes in CNOT4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##############################          ] | 77% Completed |  9min 17.7s\n",
      "2022-10-06 23:05:00,252 - pyscenic.transform - WARNING - Less than 80% of the genes in YOD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 18.1s\n",
      "2022-10-06 23:05:00,702 - pyscenic.transform - WARNING - Less than 80% of the genes in ESRRA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 18.3s\n",
      "2022-10-06 23:05:00,803 - pyscenic.transform - WARNING - Less than 80% of the genes in ETFB could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 18.6s\n",
      "2022-10-06 23:05:01,130 - pyscenic.transform - WARNING - Less than 80% of the genes in ASCC1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:01,212 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXN3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:01,224 - pyscenic.transform - WARNING - Less than 80% of the genes in ASCL2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 19.0s\n",
      "2022-10-06 23:05:01,598 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:01,627 - pyscenic.transform - WARNING - Less than 80% of the genes in NCOR1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 19.1s\n",
      "2022-10-06 23:05:01,717 - pyscenic.transform - WARNING - Less than 80% of the genes in NCOR2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 19.3s\n",
      "2022-10-06 23:05:01,867 - pyscenic.transform - WARNING - Less than 80% of the genes in NELFB could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:01,871 - pyscenic.transform - WARNING - Less than 80% of the genes in YY1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:01,952 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 19.4s\n",
      "2022-10-06 23:05:02,027 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB11 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 19.5s\n",
      "2022-10-06 23:05:02,146 - pyscenic.transform - WARNING - Less than 80% of the genes in ETS2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[###############################         ] | 77% Completed |  9min 19.7s\n",
      "2022-10-06 23:05:02,353 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB25 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 19.8s\n",
      "2022-10-06 23:05:02,423 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB38 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 19.9s\n",
      "2022-10-06 23:05:02,472 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXO1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:02,495 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 20.0s\n",
      "2022-10-06 23:05:02,573 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB41 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:02,655 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB43 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 20.1s\n",
      "2022-10-06 23:05:02,676 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:02,734 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB44 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 20.2s\n",
      "2022-10-06 23:05:02,775 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:02,788 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 20.3s\n",
      "2022-10-06 23:05:02,866 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:02,890 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 20.4s\n",
      "2022-10-06 23:05:03,034 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 20.5s\n",
      "2022-10-06 23:05:03,099 - pyscenic.transform - WARNING - Less than 80% of the genes in NFKB2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:03,123 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAP2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:03,140 - pyscenic.transform - WARNING - Less than 80% of the genes in NELFE could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 20.6s\n",
      "2022-10-06 23:05:03,209 - pyscenic.transform - WARNING - Less than 80% of the genes in SMARCA1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:03,229 - pyscenic.transform - WARNING - Less than 80% of the genes in NF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 20.8s\n",
      "2022-10-06 23:05:03,313 - pyscenic.transform - WARNING - Less than 80% of the genes in NFAT5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 21.0s\n",
      "2022-10-06 23:05:03,487 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 21.1s\n",
      "2022-10-06 23:05:03,575 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 21.5s\n",
      "2022-10-06 23:05:03,982 - pyscenic.transform - WARNING - Less than 80% of the genes in CREB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 21.7s\n",
      "2022-10-06 23:05:04,209 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 22.2s\n",
      "2022-10-06 23:05:04,742 - pyscenic.transform - WARNING - Less than 80% of the genes in SMARCB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 22.3s\n",
      "2022-10-06 23:05:04,814 - pyscenic.transform - WARNING - Less than 80% of the genes in SMARCC1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 22.4s\n",
      "2022-10-06 23:05:04,891 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:04,970 - pyscenic.transform - WARNING - Less than 80% of the genes in EWSR1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 22.5s\n",
      "2022-10-06 23:05:05,040 - pyscenic.transform - WARNING - Less than 80% of the genes in EXOSC3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 22.6s\n",
      "2022-10-06 23:05:05,130 - pyscenic.transform - WARNING - Less than 80% of the genes in EZR could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 22.7s\n",
      "2022-10-06 23:05:05,209 - pyscenic.transform - WARNING - Less than 80% of the genes in FHL2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:05,266 - pyscenic.transform - WARNING - Less than 80% of the genes in CREB3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed |  9min 22.8s\n",
      "2022-10-06 23:05:05,278 - pyscenic.transform - WARNING - Less than 80% of the genes in FIP1L1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 22.9s\n",
      "2022-10-06 23:05:05,442 - pyscenic.transform - WARNING - Less than 80% of the genes in CREB5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:05,461 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 23.0s\n",
      "2022-10-06 23:05:05,539 - pyscenic.transform - WARNING - Less than 80% of the genes in CREBL2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 23.1s\n",
      "2022-10-06 23:05:05,612 - pyscenic.transform - WARNING - Less than 80% of the genes in CREBZF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 23.4s\n",
      "2022-10-06 23:05:05,940 - pyscenic.transform - WARNING - Less than 80% of the genes in ZDHHC24 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[###############################         ] | 78% Completed |  9min 23.6s\n",
      "2022-10-06 23:05:06,194 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAI1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 23.7s\n",
      "2022-10-06 23:05:06,340 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAPC5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 23.9s\n",
      "2022-10-06 23:05:06,412 - pyscenic.transform - WARNING - Less than 80% of the genes in SND1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 24.0s\n",
      "2022-10-06 23:05:06,492 - pyscenic.transform - WARNING - Less than 80% of the genes in SNRNP70 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 24.1s\n",
      "2022-10-06 23:05:06,586 - pyscenic.transform - WARNING - Less than 80% of the genes in SNRPB2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:06,600 - pyscenic.transform - WARNING - Less than 80% of the genes in NFE2L1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:06,628 - pyscenic.transform - WARNING - Less than 80% of the genes in NOC2L could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:06,660 - pyscenic.transform - WARNING - Less than 80% of the genes in SOCS4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 24.2s\n",
      "2022-10-06 23:05:06,729 - pyscenic.transform - WARNING - Less than 80% of the genes in NONO could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 24.4s\n",
      "2022-10-06 23:05:06,893 - pyscenic.transform - WARNING - Less than 80% of the genes in SOX4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 24.6s\n",
      "2022-10-06 23:05:07,085 - pyscenic.transform - WARNING - Less than 80% of the genes in SP100 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 25.2s\n",
      "2022-10-06 23:05:07,696 - pyscenic.transform - WARNING - Less than 80% of the genes in JUND could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 26.0s\n",
      "2022-10-06 23:05:08,508 - pyscenic.transform - WARNING - Less than 80% of the genes in SP3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:08,585 - pyscenic.transform - WARNING - Less than 80% of the genes in SPAG7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 26.1s\n",
      "2022-10-06 23:05:08,657 - pyscenic.transform - WARNING - Less than 80% of the genes in SPATS2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 26.2s\n",
      "2022-10-06 23:05:08,732 - pyscenic.transform - WARNING - Less than 80% of the genes in FUBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 26.4s\n",
      "2022-10-06 23:05:08,953 - pyscenic.transform - WARNING - Less than 80% of the genes in ZEB2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 26.5s\n",
      "2022-10-06 23:05:09,036 - pyscenic.transform - WARNING - Less than 80% of the genes in ZFHX3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 26.7s\n",
      "2022-10-06 23:05:09,202 - pyscenic.transform - WARNING - Less than 80% of the genes in CRTC2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 26.8s\n",
      "2022-10-06 23:05:09,330 - pyscenic.transform - WARNING - Less than 80% of the genes in ZKSCAN1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 26.9s\n",
      "2022-10-06 23:05:09,409 - pyscenic.transform - WARNING - Less than 80% of the genes in ZMAT2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 27.0s\n",
      "2022-10-06 23:05:09,514 - pyscenic.transform - WARNING - Less than 80% of the genes in CTBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:09,566 - pyscenic.transform - WARNING - Less than 80% of the genes in ATF6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 27.4s\n",
      "2022-10-06 23:05:09,963 - pyscenic.transform - WARNING - Less than 80% of the genes in NFE2L3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 27.5s\n",
      "2022-10-06 23:05:10,048 - pyscenic.transform - WARNING - Less than 80% of the genes in NFIA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 27.7s\n",
      "2022-10-06 23:05:10,241 - pyscenic.transform - WARNING - Less than 80% of the genes in NFIC could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 28.1s\n",
      "2022-10-06 23:05:10,654 - pyscenic.transform - WARNING - Less than 80% of the genes in ZMIZ1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 28.2s\n",
      "2022-10-06 23:05:10,731 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF100 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:10,803 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF12 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 28.3s\n",
      "2022-10-06 23:05:10,877 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF131 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 28.4s\n",
      "2022-10-06 23:05:10,960 - pyscenic.transform - WARNING - Less than 80% of the genes in RBBP5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 28.5s\n",
      "2022-10-06 23:05:11,010 - pyscenic.transform - WARNING - Less than 80% of the genes in KDM5A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:11,083 - pyscenic.transform - WARNING - Less than 80% of the genes in KDM7A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 28.6s\n",
      "2022-10-06 23:05:11,154 - pyscenic.transform - WARNING - Less than 80% of the genes in KIAA0907 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 28.8s\n",
      "2022-10-06 23:05:11,320 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF10 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:11,404 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF13 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[###############################         ] | 78% Completed |  9min 29.3s\n",
      "2022-10-06 23:05:11,847 - pyscenic.transform - WARNING - Less than 80% of the genes in GABPB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 29.4s\n",
      "2022-10-06 23:05:11,957 - pyscenic.transform - WARNING - Less than 80% of the genes in KMT2A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:11,968 - pyscenic.transform - WARNING - Less than 80% of the genes in GADD45A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 29.7s\n",
      "2022-10-06 23:05:12,259 - pyscenic.transform - WARNING - Less than 80% of the genes in LEF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 29.8s\n",
      "2022-10-06 23:05:12,335 - pyscenic.transform - WARNING - Less than 80% of the genes in LMO2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:12,368 - pyscenic.transform - WARNING - Less than 80% of the genes in GMEB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 29.9s\n",
      "2022-10-06 23:05:12,426 - pyscenic.transform - WARNING - Less than 80% of the genes in LRRFIP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:12,510 - pyscenic.transform - WARNING - Less than 80% of the genes in LSM6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 30.0s\n",
      "2022-10-06 23:05:12,599 - pyscenic.transform - WARNING - Less than 80% of the genes in LUZP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 30.1s\n",
      "2022-10-06 23:05:12,662 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM22 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:12,675 - pyscenic.transform - WARNING - Less than 80% of the genes in LYL1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:12,692 - pyscenic.transform - WARNING - Less than 80% of the genes in CTNNB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:12,710 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2A1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 30.2s\n",
      "2022-10-06 23:05:12,785 - pyscenic.transform - WARNING - Less than 80% of the genes in CUX1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:12,793 - pyscenic.transform - WARNING - Less than 80% of the genes in MAF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:12,812 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2A2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 30.3s\n",
      "2022-10-06 23:05:12,862 - pyscenic.transform - WARNING - Less than 80% of the genes in CXXC1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:12,903 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 30.4s\n",
      "2022-10-06 23:05:12,950 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:12,954 - pyscenic.transform - WARNING - Less than 80% of the genes in CYB5R1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:12,997 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2F1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 30.5s\n",
      "2022-10-06 23:05:13,048 - pyscenic.transform - WARNING - Less than 80% of the genes in RBM8A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:13,078 - pyscenic.transform - WARNING - Less than 80% of the genes in CYCS could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:13,080 - pyscenic.transform - WARNING - Less than 80% of the genes in IRF9 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 30.6s\n",
      "2022-10-06 23:05:13,186 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2I could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 30.8s\n",
      "2022-10-06 23:05:13,319 - pyscenic.transform - WARNING - Less than 80% of the genes in JAZF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:13,394 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF3A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 30.9s\n",
      "2022-10-06 23:05:13,510 - pyscenic.transform - WARNING - Less than 80% of the genes in DBP could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 31.0s\n",
      "2022-10-06 23:05:13,552 - pyscenic.transform - WARNING - Less than 80% of the genes in GTPBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 31.1s\n",
      "2022-10-06 23:05:13,714 - pyscenic.transform - WARNING - Less than 80% of the genes in GZF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 31.2s\n",
      "2022-10-06 23:05:13,809 - pyscenic.transform - WARNING - Less than 80% of the genes in H1FX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 31.4s\n",
      "2022-10-06 23:05:14,003 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF165 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 31.6s\n",
      "2022-10-06 23:05:14,146 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 31.7s\n",
      "2022-10-06 23:05:14,228 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF207 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:14,309 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF22 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 31.8s\n",
      "2022-10-06 23:05:14,379 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF224 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:14,397 - pyscenic.transform - WARNING - Less than 80% of the genes in HBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 31.9s\n",
      "2022-10-06 23:05:14,451 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF226 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[###############################         ] | 78% Completed |  9min 32.0s\n",
      "2022-10-06 23:05:14,538 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF24 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 32.1s\n",
      "2022-10-06 23:05:14,701 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF267 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 32.2s\n",
      "2022-10-06 23:05:14,734 - pyscenic.transform - WARNING - Less than 80% of the genes in HDAC1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:14,819 - pyscenic.transform - WARNING - Less than 80% of the genes in NFKB2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 32.3s\n",
      "2022-10-06 23:05:14,834 - pyscenic.transform - WARNING - Less than 80% of the genes in HDAC2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:14,856 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF280D could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:14,863 - pyscenic.transform - WARNING - Less than 80% of the genes in RCOR1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:14,917 - pyscenic.transform - WARNING - Less than 80% of the genes in HDAC3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 32.5s\n",
      "2022-10-06 23:05:15,079 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF302 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:15,103 - pyscenic.transform - WARNING - Less than 80% of the genes in HES1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 32.7s\n",
      "2022-10-06 23:05:15,277 - pyscenic.transform - WARNING - Less than 80% of the genes in HES6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 32.9s\n",
      "2022-10-06 23:05:15,465 - pyscenic.transform - WARNING - Less than 80% of the genes in HHEX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 33.1s\n",
      "2022-10-06 23:05:15,669 - pyscenic.transform - WARNING - Less than 80% of the genes in HIF1A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 33.2s\n",
      "2022-10-06 23:05:15,826 - pyscenic.transform - WARNING - Less than 80% of the genes in HIST2H2BE could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 33.5s\n",
      "2022-10-06 23:05:16,087 - pyscenic.transform - WARNING - Less than 80% of the genes in HIVEP3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 33.6s\n",
      "2022-10-06 23:05:16,157 - pyscenic.transform - WARNING - Less than 80% of the genes in HLCS could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 33.7s\n",
      "2022-10-06 23:05:16,311 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF32 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 33.9s\n",
      "2022-10-06 23:05:16,475 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF331 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 34.0s\n",
      "2022-10-06 23:05:16,575 - pyscenic.transform - WARNING - Less than 80% of the genes in HMG20B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 34.2s\n",
      "2022-10-06 23:05:16,789 - pyscenic.transform - WARNING - Less than 80% of the genes in DHX36 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 34.3s\n",
      "2022-10-06 23:05:16,932 - pyscenic.transform - WARNING - Less than 80% of the genes in DMAP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 34.4s\n",
      "2022-10-06 23:05:17,025 - pyscenic.transform - WARNING - Less than 80% of the genes in DNAJC21 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 34.6s\n",
      "2022-10-06 23:05:17,182 - pyscenic.transform - WARNING - Less than 80% of the genes in BACH1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:17,224 - pyscenic.transform - WARNING - Less than 80% of the genes in DNTTIP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 34.9s\n",
      "2022-10-06 23:05:17,480 - pyscenic.transform - WARNING - Less than 80% of the genes in DUSP22 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 35.1s\n",
      "2022-10-06 23:05:17,738 - pyscenic.transform - WARNING - Less than 80% of the genes in MAFF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 35.3s\n",
      "2022-10-06 23:05:17,883 - pyscenic.transform - WARNING - Less than 80% of the genes in SPR could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 35.6s\n",
      "2022-10-06 23:05:18,187 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXN3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 35.7s\n",
      "2022-10-06 23:05:18,322 - pyscenic.transform - WARNING - Less than 80% of the genes in HMGN3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 35.8s\n",
      "2022-10-06 23:05:18,419 - pyscenic.transform - WARNING - Less than 80% of the genes in HNRNPA0 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 36.5s\n",
      "2022-10-06 23:05:19,056 - pyscenic.transform - WARNING - Less than 80% of the genes in MAFG could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:19,132 - pyscenic.transform - WARNING - Less than 80% of the genes in MAGEF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 36.6s\n",
      "2022-10-06 23:05:19,211 - pyscenic.transform - WARNING - Less than 80% of the genes in MAGOH could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:19,217 - pyscenic.transform - WARNING - Less than 80% of the genes in SRF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 36.7s\n",
      "2022-10-06 23:05:19,272 - pyscenic.transform - WARNING - Less than 80% of the genes in HNRNPH3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:19,304 - pyscenic.transform - WARNING - Less than 80% of the genes in SRP9 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[###############################         ] | 78% Completed |  9min 36.8s\n",
      "2022-10-06 23:05:19,435 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXO1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 36.9s\n",
      "2022-10-06 23:05:19,520 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXO3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 37.0s\n",
      "2022-10-06 23:05:19,628 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF385A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 37.1s\n",
      "2022-10-06 23:05:19,697 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF394 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 37.5s\n",
      "2022-10-06 23:05:20,111 - pyscenic.transform - WARNING - Less than 80% of the genes in RELA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 37.7s\n",
      "2022-10-06 23:05:20,254 - pyscenic.transform - WARNING - Less than 80% of the genes in BAD could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 37.8s\n",
      "2022-10-06 23:05:20,358 - pyscenic.transform - WARNING - Less than 80% of the genes in NR3C1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 37.9s\n",
      "2022-10-06 23:05:20,466 - pyscenic.transform - WARNING - Less than 80% of the genes in NR4A1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 38.0s\n",
      "2022-10-06 23:05:20,615 - pyscenic.transform - WARNING - Less than 80% of the genes in E2F3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:20,626 - pyscenic.transform - WARNING - Less than 80% of the genes in MAX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 38.1s\n",
      "2022-10-06 23:05:20,655 - pyscenic.transform - WARNING - Less than 80% of the genes in BAX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:20,696 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 38.2s\n",
      "2022-10-06 23:05:20,759 - pyscenic.transform - WARNING - Less than 80% of the genes in HOMEZ could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:20,760 - pyscenic.transform - WARNING - Less than 80% of the genes in BBX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:20,777 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:20,793 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:20,852 - pyscenic.transform - WARNING - Less than 80% of the genes in BCL11A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 38.3s\n",
      "2022-10-06 23:05:20,867 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:20,904 - pyscenic.transform - WARNING - Less than 80% of the genes in NUP107 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:20,933 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF429 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:20,942 - pyscenic.transform - WARNING - Less than 80% of the genes in MBNL2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 38.4s\n",
      "2022-10-06 23:05:21,036 - pyscenic.transform - WARNING - Less than 80% of the genes in FUBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:21,053 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF43 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 38.5s\n",
      "2022-10-06 23:05:21,117 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF430 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed |  9min 38.6s\n",
      "2022-10-06 23:05:21,192 - pyscenic.transform - WARNING - Less than 80% of the genes in HP1BP3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 38.8s\n",
      "2022-10-06 23:05:21,382 - pyscenic.transform - WARNING - Less than 80% of the genes in HSPA5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 39.0s\n",
      "2022-10-06 23:05:21,495 - pyscenic.transform - WARNING - Less than 80% of the genes in NONO could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 39.4s\n",
      "2022-10-06 23:05:21,985 - pyscenic.transform - WARNING - Less than 80% of the genes in E2F4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 39.6s\n",
      "2022-10-06 23:05:22,221 - pyscenic.transform - WARNING - Less than 80% of the genes in BCL3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 39.8s\n",
      "2022-10-06 23:05:22,419 - pyscenic.transform - WARNING - Less than 80% of the genes in BCLAF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 40.0s\n",
      "2022-10-06 23:05:22,616 - pyscenic.transform - WARNING - Less than 80% of the genes in BHLHE40 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 40.1s\n",
      "2022-10-06 23:05:22,740 - pyscenic.transform - WARNING - Less than 80% of the genes in BHLHE41 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 40.2s\n",
      "2022-10-06 23:05:22,843 - pyscenic.transform - WARNING - Less than 80% of the genes in BPTF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 41.2s\n",
      "2022-10-06 23:05:23,773 - pyscenic.transform - WARNING - Less than 80% of the genes in EGR1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 41.6s\n",
      "2022-10-06 23:05:24,243 - pyscenic.transform - WARNING - Less than 80% of the genes in MECP2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,258 - pyscenic.transform - WARNING - Less than 80% of the genes in KDM5A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 41.7s\n",
      "2022-10-06 23:05:24,265 - pyscenic.transform - WARNING - Less than 80% of the genes in GADD45A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-10-06 23:05:24,298 - pyscenic.transform - WARNING - Less than 80% of the genes in PBX2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,324 - pyscenic.transform - WARNING - Less than 80% of the genes in MED30 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 41.8s\n",
      "2022-10-06 23:05:24,381 - pyscenic.transform - WARNING - Less than 80% of the genes in PBX3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,406 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,425 - pyscenic.transform - WARNING - Less than 80% of the genes in KDM7A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,449 - pyscenic.transform - WARNING - Less than 80% of the genes in PCK2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 41.9s\n",
      "2022-10-06 23:05:24,484 - pyscenic.transform - WARNING - Less than 80% of the genes in GIT2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,501 - pyscenic.transform - WARNING - Less than 80% of the genes in KIAA0907 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,525 - pyscenic.transform - WARNING - Less than 80% of the genes in PDLIM5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 42.1s\n",
      "2022-10-06 23:05:24,684 - pyscenic.transform - WARNING - Less than 80% of the genes in PHLDA2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,696 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF10 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,735 - pyscenic.transform - WARNING - Less than 80% of the genes in GRHPR could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,755 - pyscenic.transform - WARNING - Less than 80% of the genes in PHTF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 42.3s\n",
      "2022-10-06 23:05:24,788 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF13 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:24,826 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2A2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 42.4s\n",
      "2022-10-06 23:05:24,903 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 42.5s\n",
      "2022-10-06 23:05:25,041 - pyscenic.transform - WARNING - Less than 80% of the genes in EGR2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:25,047 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 42.6s\n",
      "2022-10-06 23:05:25,115 - pyscenic.transform - WARNING - Less than 80% of the genes in ILF2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:25,167 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF2I could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 42.8s\n",
      "2022-10-06 23:05:25,273 - pyscenic.transform - WARNING - Less than 80% of the genes in ING4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:25,356 - pyscenic.transform - WARNING - Less than 80% of the genes in GTF3A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 42.9s\n",
      "2022-10-06 23:05:25,464 - pyscenic.transform - WARNING - Less than 80% of the genes in PLAGL1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 43.0s\n",
      "2022-10-06 23:05:25,548 - pyscenic.transform - WARNING - Less than 80% of the genes in PML could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 43.1s\n",
      "2022-10-06 23:05:25,572 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:25,619 - pyscenic.transform - WARNING - Less than 80% of the genes in POLD2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed |  9min 43.2s\n",
      "2022-10-06 23:05:25,693 - pyscenic.transform - WARNING - Less than 80% of the genes in POLE3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:25,708 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2C could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:25,766 - pyscenic.transform - WARNING - Less than 80% of the genes in POLE4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 80% Completed |  9min 44.4s\n",
      "2022-10-06 23:05:26,980 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF8 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:27,006 - pyscenic.transform - WARNING - Less than 80% of the genes in RFXANK could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:27,018 - pyscenic.transform - WARNING - Less than 80% of the genes in POLR2A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:27,048 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2D could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:27,052 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB11 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 80% Completed |  9min 44.5s\n",
      "2022-10-06 23:05:27,054 - pyscenic.transform - WARNING - Less than 80% of the genes in KLF9 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:27,093 - pyscenic.transform - WARNING - Less than 80% of the genes in POLR3G could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 80% Completed |  9min 44.6s\n",
      "2022-10-06 23:05:27,166 - pyscenic.transform - WARNING - Less than 80% of the genes in KMT2A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 80% Completed |  9min 44.8s\n",
      "2022-10-06 23:05:27,345 - pyscenic.transform - WARNING - Less than 80% of the genes in RNASEH2C could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:27,352 - pyscenic.transform - WARNING - Less than 80% of the genes in LARP4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 80% Completed |  9min 45.1s\n",
      "2022-10-06 23:05:27,654 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB33 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:27,682 - pyscenic.transform - WARNING - Less than 80% of the genes in LMO2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[################################        ] | 81% Completed |  9min 45.2s\n",
      "2022-10-06 23:05:27,814 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB38 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 45.3s\n",
      "2022-10-06 23:05:27,911 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:27,916 - pyscenic.transform - WARNING - Less than 80% of the genes in LSM6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 45.4s\n",
      "2022-10-06 23:05:28,001 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB41 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:28,020 - pyscenic.transform - WARNING - Less than 80% of the genes in LUZP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 45.5s\n",
      "2022-10-06 23:05:28,116 - pyscenic.transform - WARNING - Less than 80% of the genes in LYL1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 45.6s\n",
      "2022-10-06 23:05:28,188 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB43 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 45.7s\n",
      "2022-10-06 23:05:28,308 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB44 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:28,328 - pyscenic.transform - WARNING - Less than 80% of the genes in POU2AF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 45.8s\n",
      "2022-10-06 23:05:28,395 - pyscenic.transform - WARNING - Less than 80% of the genes in POU2F2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 45.9s\n",
      "2022-10-06 23:05:28,459 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB48 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:28,522 - pyscenic.transform - WARNING - Less than 80% of the genes in EIF5A2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 46.0s\n",
      "2022-10-06 23:05:28,591 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:28,667 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT5A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 46.3s\n",
      "2022-10-06 23:05:28,811 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 46.4s\n",
      "2022-10-06 23:05:28,958 - pyscenic.transform - WARNING - Less than 80% of the genes in STUB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 46.5s\n",
      "2022-10-06 23:05:29,114 - pyscenic.transform - WARNING - Less than 80% of the genes in SUCLG1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 46.7s\n",
      "2022-10-06 23:05:29,272 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF429 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:29,283 - pyscenic.transform - WARNING - Less than 80% of the genes in SUZ12 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 46.8s\n",
      "2022-10-06 23:05:29,346 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF43 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:29,425 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF430 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 46.9s\n",
      "2022-10-06 23:05:29,448 - pyscenic.transform - WARNING - Less than 80% of the genes in TAF9 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 47.0s\n",
      "2022-10-06 23:05:29,576 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF438 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:29,643 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF439 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:29,645 - pyscenic.transform - WARNING - Less than 80% of the genes in TBL1XR1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 47.3s\n",
      "2022-10-06 23:05:29,809 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF451 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 47.9s\n",
      "2022-10-06 23:05:30,438 - pyscenic.transform - WARNING - Less than 80% of the genes in ZSCAN25 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:30,444 - pyscenic.transform - WARNING - Less than 80% of the genes in MAFF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:30,513 - pyscenic.transform - WARNING - Less than 80% of the genes in ZXDC could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 48.0s\n",
      "2022-10-06 23:05:30,593 - pyscenic.transform - WARNING - Less than 80% of the genes in ZZZ3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 48.8s\n",
      "2022-10-06 23:05:31,398 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF467 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 48.9s\n",
      "2022-10-06 23:05:31,481 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 49.1s\n",
      "2022-10-06 23:05:31,634 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF493 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:31,642 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT5B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:31,720 - pyscenic.transform - WARNING - Less than 80% of the genes in STAT6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:31,721 - pyscenic.transform - WARNING - Less than 80% of the genes in MAFG could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 49.2s\n",
      "2022-10-06 23:05:31,724 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF503 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed |  9min 49.3s\n",
      "2022-10-06 23:05:31,852 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF506 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-10-06 23:05:31,885 - pyscenic.transform - WARNING - Less than 80% of the genes in STUB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 49.4s\n",
      "2022-10-06 23:05:31,980 - pyscenic.transform - WARNING - Less than 80% of the genes in SUCLG1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:32,012 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF510 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 49.5s\n",
      "2022-10-06 23:05:32,081 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF513 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:32,166 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF516 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 49.7s\n",
      "2022-10-06 23:05:32,175 - pyscenic.transform - WARNING - Less than 80% of the genes in SUZ12 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:32,256 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF524 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 49.9s\n",
      "2022-10-06 23:05:32,427 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF529 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 50.0s\n",
      "2022-10-06 23:05:32,606 - pyscenic.transform - WARNING - Less than 80% of the genes in NR3C1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 50.1s\n",
      "2022-10-06 23:05:32,627 - pyscenic.transform - WARNING - Less than 80% of the genes in PPARG could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:32,697 - pyscenic.transform - WARNING - Less than 80% of the genes in PPP1R10 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 50.2s\n",
      "2022-10-06 23:05:32,756 - pyscenic.transform - WARNING - Less than 80% of the genes in CAT could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:32,766 - pyscenic.transform - WARNING - Less than 80% of the genes in PPP2R3B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 50.3s\n",
      "2022-10-06 23:05:32,839 - pyscenic.transform - WARNING - Less than 80% of the genes in PQBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:32,930 - pyscenic.transform - WARNING - Less than 80% of the genes in PRDM1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 50.4s\n",
      "2022-10-06 23:05:33,024 - pyscenic.transform - WARNING - Less than 80% of the genes in PRDX5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 50.8s\n",
      "2022-10-06 23:05:33,295 - pyscenic.transform - WARNING - Less than 80% of the genes in PRNP could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:33,375 - pyscenic.transform - WARNING - Less than 80% of the genes in PSMA6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 50.9s\n",
      "2022-10-06 23:05:33,415 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB7B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:33,440 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF580 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:33,453 - pyscenic.transform - WARNING - Less than 80% of the genes in PSMC2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 51.1s\n",
      "2022-10-06 23:05:33,584 - pyscenic.transform - WARNING - Less than 80% of the genes in ODC1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:33,593 - pyscenic.transform - WARNING - Less than 80% of the genes in ZC3H7A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:33,619 - pyscenic.transform - WARNING - Less than 80% of the genes in PTPMT1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 51.2s\n",
      "2022-10-06 23:05:33,688 - pyscenic.transform - WARNING - Less than 80% of the genes in ZCCHC17 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:33,771 - pyscenic.transform - WARNING - Less than 80% of the genes in ZDHHC24 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 51.4s\n",
      "2022-10-06 23:05:33,939 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF611 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed |  9min 51.5s\n",
      "2022-10-06 23:05:34,029 - pyscenic.transform - WARNING - Less than 80% of the genes in CBFB could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 82% Completed |  9min 51.6s\n",
      "2022-10-06 23:05:34,154 - pyscenic.transform - WARNING - Less than 80% of the genes in TCF7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:34,173 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF622 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 51.7s\n",
      "2022-10-06 23:05:34,237 - pyscenic.transform - WARNING - Less than 80% of the genes in TCF7L2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 51.9s\n",
      "2022-10-06 23:05:34,425 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF626 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:34,492 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF639 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 52.0s\n",
      "2022-10-06 23:05:34,572 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF641 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 52.1s\n",
      "2022-10-06 23:05:34,637 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF649 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:34,727 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF652 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 52.3s\n",
      "2022-10-06 23:05:34,901 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF655 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 52.4s\n",
      "2022-10-06 23:05:35,025 - pyscenic.transform - WARNING - Less than 80% of the genes in MAGEF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[#################################       ] | 83% Completed |  9min 52.6s\n",
      "2022-10-06 23:05:35,122 - pyscenic.transform - WARNING - Less than 80% of the genes in MAGOH could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 52.7s\n",
      "2022-10-06 23:05:35,221 - pyscenic.transform - WARNING - Less than 80% of the genes in MAPK1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 52.8s\n",
      "2022-10-06 23:05:35,364 - pyscenic.transform - WARNING - Less than 80% of the genes in AGMAT could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 53.2s\n",
      "2022-10-06 23:05:35,728 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF682 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 53.3s\n",
      "2022-10-06 23:05:35,886 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF687 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 53.8s\n",
      "2022-10-06 23:05:36,349 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF708 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 54.1s\n",
      "2022-10-06 23:05:36,620 - pyscenic.transform - WARNING - Less than 80% of the genes in MAX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:36,637 - pyscenic.transform - WARNING - Less than 80% of the genes in ELK3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 54.2s\n",
      "2022-10-06 23:05:36,724 - pyscenic.transform - WARNING - Less than 80% of the genes in ELK4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:36,752 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF721 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 54.3s\n",
      "2022-10-06 23:05:36,889 - pyscenic.transform - WARNING - Less than 80% of the genes in ZEB2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 54.7s\n",
      "2022-10-06 23:05:37,257 - pyscenic.transform - WARNING - Less than 80% of the genes in PBX2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:37,347 - pyscenic.transform - WARNING - Less than 80% of the genes in PBX3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 54.8s\n",
      "2022-10-06 23:05:37,422 - pyscenic.transform - WARNING - Less than 80% of the genes in PCK2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 54.9s\n",
      "2022-10-06 23:05:37,508 - pyscenic.transform - WARNING - Less than 80% of the genes in PDLIM5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:37,522 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF791 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 55.0s\n",
      "2022-10-06 23:05:37,594 - pyscenic.transform - WARNING - Less than 80% of the genes in PDS5A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed |  9min 55.2s\n",
      "2022-10-06 23:05:37,872 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF813 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed |  9min 55.5s\n",
      "2022-10-06 23:05:38,026 - pyscenic.transform - WARNING - Less than 80% of the genes in RREB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:38,032 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF821 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed |  9min 55.6s\n",
      "2022-10-06 23:05:38,137 - pyscenic.transform - WARNING - Less than 80% of the genes in MAZ could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:38,140 - pyscenic.transform - WARNING - Less than 80% of the genes in RUFY3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:38,191 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF844 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed |  9min 55.7s\n",
      "2022-10-06 23:05:38,217 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:38,234 - pyscenic.transform - WARNING - Less than 80% of the genes in RUNX1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:38,257 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF879 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed |  9min 55.8s\n",
      "2022-10-06 23:05:38,318 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:38,350 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF880 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:38,421 - pyscenic.transform - WARNING - Less than 80% of the genes in MBD4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed |  9min 56.0s\n",
      "2022-10-06 23:05:38,589 - pyscenic.transform - WARNING - Less than 80% of the genes in RUVBL1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed |  9min 56.1s\n",
      "2022-10-06 23:05:38,654 - pyscenic.transform - WARNING - Less than 80% of the genes in ZRSR2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 56.3s\n",
      "2022-10-06 23:05:38,887 - pyscenic.transform - WARNING - Less than 80% of the genes in TCF3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:39,057 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 57.7s\n",
      "2022-10-06 23:05:40,355 - pyscenic.transform - WARNING - Less than 80% of the genes in TCF7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:40,368 - pyscenic.transform - WARNING - Less than 80% of the genes in ZFP14 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 57.9s\n",
      "2022-10-06 23:05:40,418 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2C could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 58.0s\n",
      "2022-10-06 23:05:40,520 - pyscenic.transform - WARNING - Less than 80% of the genes in MEF2D could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 58.2s\n",
      "2022-10-06 23:05:40,717 - pyscenic.transform - WARNING - Less than 80% of the genes in PHLDA2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:40,792 - pyscenic.transform - WARNING - Less than 80% of the genes in ZFP91 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 58.3s\n",
      "2022-10-06 23:05:40,866 - pyscenic.transform - WARNING - Less than 80% of the genes in ZFPM1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 58.4s\n",
      "2022-10-06 23:05:40,953 - pyscenic.transform - WARNING - Less than 80% of the genes in ZFX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:41,018 - pyscenic.transform - WARNING - Less than 80% of the genes in MIOS could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:41,030 - pyscenic.transform - WARNING - Less than 80% of the genes in ZHX1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 58.6s\n",
      "2022-10-06 23:05:41,198 - pyscenic.transform - WARNING - Less than 80% of the genes in ZKSCAN1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:41,242 - pyscenic.transform - WARNING - Less than 80% of the genes in MLX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 58.7s\n",
      "2022-10-06 23:05:41,268 - pyscenic.transform - WARNING - Less than 80% of the genes in ZKSCAN3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:41,301 - pyscenic.transform - WARNING - Less than 80% of the genes in EP300 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:41,331 - pyscenic.transform - WARNING - Less than 80% of the genes in MLXIP could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:41,338 - pyscenic.transform - WARNING - Less than 80% of the genes in ZKSCAN5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 58.9s\n",
      "2022-10-06 23:05:41,412 - pyscenic.transform - WARNING - Less than 80% of the genes in MLXIPL could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:41,429 - pyscenic.transform - WARNING - Less than 80% of the genes in ZMAT2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 59.0s\n",
      "2022-10-06 23:05:41,485 - pyscenic.transform - WARNING - Less than 80% of the genes in ERF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:41,491 - pyscenic.transform - WARNING - Less than 80% of the genes in MNT could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:41,582 - pyscenic.transform - WARNING - Less than 80% of the genes in MRPL1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 59.1s\n",
      "2022-10-06 23:05:41,676 - pyscenic.transform - WARNING - Less than 80% of the genes in MRPL2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed |  9min 59.8s\n",
      "2022-10-06 23:05:42,368 - pyscenic.transform - WARNING - Less than 80% of the genes in PKNOX1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  0.1s\n",
      "2022-10-06 23:05:42,706 - pyscenic.transform - WARNING - Less than 80% of the genes in POLD2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  0.2s\n",
      "2022-10-06 23:05:42,793 - pyscenic.transform - WARNING - Less than 80% of the genes in ZMIZ1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  0.3s\n",
      "2022-10-06 23:05:42,875 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF100 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  0.4s\n",
      "2022-10-06 23:05:42,949 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF12 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  1.4s\n",
      "2022-10-06 23:05:43,955 - pyscenic.transform - WARNING - Less than 80% of the genes in POLE3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  2.1s\n",
      "2022-10-06 23:05:44,642 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF143 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  2.4s\n",
      "2022-10-06 23:05:44,941 - pyscenic.transform - WARNING - Less than 80% of the genes in SAP30 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:44,947 - pyscenic.transform - WARNING - Less than 80% of the genes in ETFB could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:44,991 - pyscenic.transform - WARNING - Less than 80% of the genes in MSRA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  2.7s\n",
      "2022-10-06 23:05:45,297 - pyscenic.transform - WARNING - Less than 80% of the genes in POLR2A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  4.1s\n",
      "2022-10-06 23:05:46,672 - pyscenic.transform - WARNING - Less than 80% of the genes in ETS2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:46,695 - pyscenic.transform - WARNING - Less than 80% of the genes in TERF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  4.3s\n",
      "2022-10-06 23:05:46,860 - pyscenic.transform - WARNING - Less than 80% of the genes in TFAM could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  4.6s\n",
      "2022-10-06 23:05:47,201 - pyscenic.transform - WARNING - Less than 80% of the genes in NR2F6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  4.7s\n",
      "2022-10-06 23:05:47,331 - pyscenic.transform - WARNING - Less than 80% of the genes in HSF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  4.9s\n",
      "2022-10-06 23:05:47,505 - pyscenic.transform - WARNING - Less than 80% of the genes in HSF4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  5.7s\n",
      "2022-10-06 23:05:48,260 - pyscenic.transform - WARNING - Less than 80% of the genes in MTF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:48,309 - pyscenic.transform - WARNING - Less than 80% of the genes in SF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  5.8s\n",
      "2022-10-06 23:05:48,396 - pyscenic.transform - WARNING - Less than 80% of the genes in SF3B1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[##################################      ] | 85% Completed | 10min  5.9s\n",
      "2022-10-06 23:05:48,453 - pyscenic.transform - WARNING - Less than 80% of the genes in MXD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:48,482 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF207 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  6.0s\n",
      "2022-10-06 23:05:48,554 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF217 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  6.1s\n",
      "2022-10-06 23:05:48,731 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF22 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  6.4s\n",
      "2022-10-06 23:05:49,025 - pyscenic.transform - WARNING - Less than 80% of the genes in MYC could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  7.0s\n",
      "2022-10-06 23:05:49,577 - pyscenic.transform - WARNING - Less than 80% of the genes in POU2F1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  7.1s\n",
      "2022-10-06 23:05:49,661 - pyscenic.transform - WARNING - Less than 80% of the genes in POU2F2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:49,702 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  7.2s\n",
      "2022-10-06 23:05:49,806 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  7.3s\n",
      "2022-10-06 23:05:49,907 - pyscenic.transform - WARNING - Less than 80% of the genes in ETV6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  7.5s\n",
      "2022-10-06 23:05:50,103 - pyscenic.transform - WARNING - Less than 80% of the genes in TFCP2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  7.6s\n",
      "2022-10-06 23:05:50,183 - pyscenic.transform - WARNING - Less than 80% of the genes in TFCP2L1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  8.5s\n",
      "2022-10-06 23:05:51,155 - pyscenic.transform - WARNING - Less than 80% of the genes in OLIG1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  9.2s\n",
      "2022-10-06 23:05:51,763 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF224 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:51,839 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF226 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  9.4s\n",
      "2022-10-06 23:05:51,973 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  9.5s\n",
      "2022-10-06 23:05:52,146 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF24 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min  9.9s\n",
      "2022-10-06 23:05:52,521 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF263 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min 10.6s\n",
      "2022-10-06 23:05:53,117 - pyscenic.transform - WARNING - Less than 80% of the genes in EWSR1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min 10.7s\n",
      "2022-10-06 23:05:53,196 - pyscenic.transform - WARNING - Less than 80% of the genes in EXOSC3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min 10.8s\n",
      "2022-10-06 23:05:53,304 - pyscenic.transform - WARNING - Less than 80% of the genes in TFDP2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min 11.3s\n",
      "2022-10-06 23:05:53,863 - pyscenic.transform - WARNING - Less than 80% of the genes in PPARA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 85% Completed | 10min 11.5s\n",
      "2022-10-06 23:05:54,017 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF274 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:54,102 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF276 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 12.1s\n",
      "2022-10-06 23:05:54,738 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF324 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 12.4s\n",
      "2022-10-06 23:05:54,948 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF331 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 12.6s\n",
      "2022-10-06 23:05:55,103 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:55,120 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF33A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:55,180 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:55,194 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF34 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 12.8s\n",
      "2022-10-06 23:05:55,358 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAD7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 12.9s\n",
      "2022-10-06 23:05:55,456 - pyscenic.transform - WARNING - Less than 80% of the genes in SMAP2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 13.8s\n",
      "2022-10-06 23:05:56,402 - pyscenic.transform - WARNING - Less than 80% of the genes in TFEB could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 13.9s\n",
      "2022-10-06 23:05:56,438 - pyscenic.transform - WARNING - Less than 80% of the genes in EZR could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:56,518 - pyscenic.transform - WARNING - Less than 80% of the genes in FANK1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 14.1s\n",
      "2022-10-06 23:05:56,690 - pyscenic.transform - WARNING - Less than 80% of the genes in FHL2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[##################################      ] | 86% Completed | 10min 14.2s\n",
      "2022-10-06 23:05:56,776 - pyscenic.transform - WARNING - Less than 80% of the genes in FIP1L1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:56,780 - pyscenic.transform - WARNING - Less than 80% of the genes in TGIF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 14.3s\n",
      "2022-10-06 23:05:56,861 - pyscenic.transform - WARNING - Less than 80% of the genes in FLI1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 14.4s\n",
      "2022-10-06 23:05:56,992 - pyscenic.transform - WARNING - Less than 80% of the genes in SMARCB1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 14.5s\n",
      "2022-10-06 23:05:57,071 - pyscenic.transform - WARNING - Less than 80% of the genes in SMARCC1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 14.7s\n",
      "2022-10-06 23:05:57,238 - pyscenic.transform - WARNING - Less than 80% of the genes in SMUG1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 15.9s\n",
      "2022-10-06 23:05:58,478 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAI1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 16.0s\n",
      "2022-10-06 23:05:58,583 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF358 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 16.1s\n",
      "2022-10-06 23:05:58,710 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAPC4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 16.2s\n",
      "2022-10-06 23:05:58,784 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAPC5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 16.3s\n",
      "2022-10-06 23:05:58,869 - pyscenic.transform - WARNING - Less than 80% of the genes in SND1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 16.4s\n",
      "2022-10-06 23:05:58,962 - pyscenic.transform - WARNING - Less than 80% of the genes in SNRNP70 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 16.6s\n",
      "2022-10-06 23:05:59,160 - pyscenic.transform - WARNING - Less than 80% of the genes in SOCS4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:05:59,233 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF394 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 17.0s\n",
      "2022-10-06 23:05:59,585 - pyscenic.transform - WARNING - Less than 80% of the genes in SOX4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 17.4s\n",
      "2022-10-06 23:05:59,990 - pyscenic.transform - WARNING - Less than 80% of the genes in PPP1R10 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 17.7s\n",
      "2022-10-06 23:06:00,239 - pyscenic.transform - WARNING - Less than 80% of the genes in PQBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 17.8s\n",
      "2022-10-06 23:06:00,342 - pyscenic.transform - WARNING - Less than 80% of the genes in PRDM1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 17.9s\n",
      "2022-10-06 23:06:00,539 - pyscenic.transform - WARNING - Less than 80% of the genes in PRDX5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 18.4s\n",
      "2022-10-06 23:06:00,941 - pyscenic.transform - WARNING - Less than 80% of the genes in SP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:06:00,958 - pyscenic.transform - WARNING - Less than 80% of the genes in THAP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 18.5s\n",
      "2022-10-06 23:06:01,028 - pyscenic.transform - WARNING - Less than 80% of the genes in THAP11 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:06:01,105 - pyscenic.transform - WARNING - Less than 80% of the genes in THAP5 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 18.6s\n",
      "2022-10-06 23:06:01,194 - pyscenic.transform - WARNING - Less than 80% of the genes in THOC2 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 18.8s\n",
      "2022-10-06 23:06:01,365 - pyscenic.transform - WARNING - Less than 80% of the genes in TIA1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 18.9s\n",
      "2022-10-06 23:06:01,520 - pyscenic.transform - WARNING - Less than 80% of the genes in TIMM8A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 19.0s\n",
      "2022-10-06 23:06:01,607 - pyscenic.transform - WARNING - Less than 80% of the genes in FOXO4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:06:01,611 - pyscenic.transform - WARNING - Less than 80% of the genes in TMEM33 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 19.9s\n",
      "2022-10-06 23:06:02,438 - pyscenic.transform - WARNING - Less than 80% of the genes in SP3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:06:02,515 - pyscenic.transform - WARNING - Less than 80% of the genes in SP4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 20.0s\n",
      "2022-10-06 23:06:02,605 - pyscenic.transform - WARNING - Less than 80% of the genes in SPAG7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 20.3s\n",
      "2022-10-06 23:06:02,934 - pyscenic.transform - WARNING - Less than 80% of the genes in TOPORS could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 21.5s\n",
      "2022-10-06 23:06:04,129 - pyscenic.transform - WARNING - Less than 80% of the genes in PSMA6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 21.6s\n",
      "2022-10-06 23:06:04,178 - pyscenic.transform - WARNING - Less than 80% of the genes in TP53 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 21.8s\n",
      "2022-10-06 23:06:04,415 - pyscenic.transform - WARNING - Less than 80% of the genes in PTPMT1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 21.9s\n",
      "2022-10-06 23:06:04,501 - pyscenic.transform - WARNING - Less than 80% of the genes in PURA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 22.3s\n",
      "2022-10-06 23:06:04,875 - pyscenic.transform - WARNING - Less than 80% of the genes in RAB2A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 86% Completed | 10min 22.4s\n",
      "2022-10-06 23:06:04,981 - pyscenic.transform - WARNING - Less than 80% of the genes in RAB7A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[##################################      ] | 87% Completed | 10min 22.6s\n",
      "2022-10-06 23:06:05,186 - pyscenic.transform - WARNING - Less than 80% of the genes in FOSL1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 87% Completed | 10min 26.4s\n",
      "2022-10-06 23:06:08,945 - pyscenic.transform - WARNING - Less than 80% of the genes in TRAF4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 26.5s\n",
      "2022-10-06 23:06:09,022 - pyscenic.transform - WARNING - Less than 80% of the genes in TRIM21 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 26.7s\n",
      "2022-10-06 23:06:09,271 - pyscenic.transform - WARNING - Less than 80% of the genes in TRIM33 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 26.8s\n",
      "2022-10-06 23:06:09,361 - pyscenic.transform - WARNING - Less than 80% of the genes in TRIM69 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 26.9s\n",
      "2022-10-06 23:06:09,558 - pyscenic.transform - WARNING - Less than 80% of the genes in TRMT1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 27.1s\n",
      "2022-10-06 23:06:09,741 - pyscenic.transform - WARNING - Less than 80% of the genes in TRPS1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 27.2s\n",
      "2022-10-06 23:06:09,834 - pyscenic.transform - WARNING - Less than 80% of the genes in TSC22D4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 27.3s\n",
      "2022-10-06 23:06:09,922 - pyscenic.transform - WARNING - Less than 80% of the genes in TSN could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 27.4s\n",
      "2022-10-06 23:06:10,006 - pyscenic.transform - WARNING - Less than 80% of the genes in TSNAX could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 27.5s\n",
      "2022-10-06 23:06:10,110 - pyscenic.transform - WARNING - Less than 80% of the genes in U2AF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 28.7s\n",
      "2022-10-06 23:06:11,276 - pyscenic.transform - WARNING - Less than 80% of the genes in MYNN could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 29.3s\n",
      "2022-10-06 23:06:11,935 - pyscenic.transform - WARNING - Less than 80% of the genes in UBE2K could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 29.4s\n",
      "2022-10-06 23:06:12,006 - pyscenic.transform - WARNING - Less than 80% of the genes in UBE2V1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 29.6s\n",
      "2022-10-06 23:06:12,155 - pyscenic.transform - WARNING - Less than 80% of the genes in UBTF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 29.7s\n",
      "2022-10-06 23:06:12,291 - pyscenic.transform - WARNING - Less than 80% of the genes in UBXN1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 30.0s\n",
      "2022-10-06 23:06:12,579 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF143 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 30.2s\n",
      "2022-10-06 23:06:12,856 - pyscenic.transform - WARNING - Less than 80% of the genes in GLI4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 31.0s\n",
      "2022-10-06 23:06:13,599 - pyscenic.transform - WARNING - Less than 80% of the genes in USP39 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 31.1s\n",
      "2022-10-06 23:06:13,687 - pyscenic.transform - WARNING - Less than 80% of the genes in UTP18 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 31.2s\n",
      "2022-10-06 23:06:13,779 - pyscenic.transform - WARNING - Less than 80% of the genes in VAMP3 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 31.3s\n",
      "2022-10-06 23:06:13,867 - pyscenic.transform - WARNING - Less than 80% of the genes in VDR could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 31.4s\n",
      "2022-10-06 23:06:14,022 - pyscenic.transform - WARNING - Less than 80% of the genes in VEZF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 31.5s\n",
      "2022-10-06 23:06:14,106 - pyscenic.transform - WARNING - Less than 80% of the genes in VPS4B could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 31.6s\n",
      "2022-10-06 23:06:14,188 - pyscenic.transform - WARNING - Less than 80% of the genes in VPS72 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 31.8s\n",
      "2022-10-06 23:06:14,374 - pyscenic.transform - WARNING - Less than 80% of the genes in XBP1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:06:14,461 - pyscenic.transform - WARNING - Less than 80% of the genes in XPA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 33.0s\n",
      "2022-10-06 23:06:15,485 - pyscenic.transform - WARNING - Less than 80% of the genes in SRBD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 33.8s\n",
      "2022-10-06 23:06:16,277 - pyscenic.transform - WARNING - Less than 80% of the genes in YEATS4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:06:16,356 - pyscenic.transform - WARNING - Less than 80% of the genes in YOD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 35.2s\n",
      "2022-10-06 23:06:17,699 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF200 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[###################################     ] | 88% Completed | 10min 35.6s\n",
      "2022-10-06 23:06:18,120 - pyscenic.transform - WARNING - Less than 80% of the genes in YY1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 35.7s\n",
      "2022-10-06 23:06:18,194 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBED1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 10min 37.1s\n",
      "2022-10-06 23:06:19,664 - pyscenic.transform - WARNING - Less than 80% of the genes in RFX1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 89% Completed | 10min 40.2s\n",
      "2022-10-06 23:06:22,747 - pyscenic.transform - WARNING - Less than 80% of the genes in SRF could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[###################################     ] | 89% Completed | 10min 40.4s\n",
      "2022-10-06 23:06:22,934 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF248 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 42.6s\n",
      "2022-10-06 23:06:25,129 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF598 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 42.7s\n",
      "2022-10-06 23:06:25,219 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF599 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 43.0s\n",
      "2022-10-06 23:06:25,552 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF429 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 43.8s\n",
      "2022-10-06 23:06:26,334 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF254 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 46.5s\n",
      "2022-10-06 23:06:29,057 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF514 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 48.6s\n",
      "2022-10-06 23:06:31,215 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF562 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 48.8s\n",
      "2022-10-06 23:06:31,392 - pyscenic.transform - WARNING - Less than 80% of the genes in HES6 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 48.9s\n",
      "2022-10-06 23:06:31,528 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF317 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 49.5s\n",
      "2022-10-06 23:06:32,084 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF669 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 50.3s\n",
      "2022-10-06 23:06:32,898 - pyscenic.transform - WARNING - Less than 80% of the genes in CBX7 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "\n",
      "2022-10-06 23:06:32,943 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBED1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 50.7s\n",
      "2022-10-06 23:06:33,306 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB11 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 51.0s\n",
      "2022-10-06 23:06:33,621 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF35 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 51.4s\n",
      "2022-10-06 23:06:34,032 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF354C could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 51.7s\n",
      "2022-10-06 23:06:34,348 - pyscenic.transform - WARNING - Less than 80% of the genes in ZBTB26 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 52.6s\n",
      "2022-10-06 23:06:35,208 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF577 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 10min 52.8s\n",
      "2022-10-06 23:06:35,420 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF580 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 91% Completed | 10min 55.7s\n",
      "2022-10-06 23:06:38,191 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF846 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 92% Completed | 10min 56.5s\n",
      "2022-10-06 23:06:39,077 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF93 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 92% Completed | 10min 57.4s\n",
      "2022-10-06 23:06:39,916 - pyscenic.transform - WARNING - Less than 80% of the genes in ZSCAN29 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 92% Completed | 10min 57.6s\n",
      "2022-10-06 23:06:40,183 - pyscenic.transform - WARNING - Less than 80% of the genes in ZSCAN5A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################    ] | 92% Completed | 10min 57.7s\n",
      "2022-10-06 23:06:40,281 - pyscenic.transform - WARNING - Less than 80% of the genes in ZNF75A could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[#####################################   ] | 94% Completed | 11min  2.4s\n",
      "2022-10-06 23:06:44,969 - pyscenic.transform - WARNING - Less than 80% of the genes in PTCD1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################### ] | 97% Completed | 11min 22.5s\n",
      "2022-10-06 23:07:04,998 - pyscenic.transform - WARNING - Less than 80% of the genes in TBX18 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################### ] | 98% Completed | 11min 27.0s\n",
      "2022-10-06 23:07:09,585 - pyscenic.transform - WARNING - Less than 80% of the genes in SNAPC4 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################### ] | 98% Completed | 11min 36.1s\n",
      "2022-10-06 23:07:18,697 - pyscenic.transform - WARNING - Less than 80% of the genes in TERF1 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################### ] | 99% Completed | 11min 53.8s\n",
      "2022-10-06 23:07:36,372 - pyscenic.transform - WARNING - Less than 80% of the genes in THRA could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[####################################### ] | 99% Completed | 12min  1.6s\n",
      "2022-10-06 23:07:44,143 - pyscenic.transform - WARNING - Less than 80% of the genes in TRIM33 could be mapped to hg19-tss-centered-10kb-7species.mc9nr. Skipping this module.\n",
      "[########################################] | 100% Completed | 12min  8.9s\n",
      "\n",
      "2022-10-06 23:07:51,482 - pyscenic.cli.pyscenic - INFO - Writing results to file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!pyscenic ctx {f_out_adj_path} \\\n",
    "    {f_db_names} \\\n",
    "    --annotations_fname {f_motif_path} \\\n",
    "    --expression_mtx_fname {f_loom_path_scenic} \\\n",
    "    --output {f_out_reg_path} \\\n",
    "    --mask_dropouts \\\n",
    "    --num_workers 12"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 4: Cellular enrichment (aka AUCell) from CLI\n",
    "\n",
    "It is important to check that most cells have a substantial fraction of expressed/detected genes in the calculation of the AUC.\n",
    "The following histogram gives an idea of the distribution and allows selection of an appropriate threshold.\n",
    "In this plot, a few thresholds are highlighted, with the number of genes selected shown in red text and the corresponding percentile in parentheses).\n",
    "See [the relevant section in the R tutorial](https://bioconductor.org/packages/devel/bioc/vignettes/AUCell/inst/doc/AUCell.html#build-gene-expression-rankings-for-each-cell) for more information.\n",
    "\n",
    "By using the default setting for `--auc_threshold` of `0.05`, we see that **1192** genes are selected for the rankings based on the plot below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "f_loom_path_scenic = \"/node210data/gyeongdaekim/pyscenic/thyroid/new_M1_M2/m2_cd163_po.loom\"\n",
    "f_pyscenic_output = \"/node210data/gyeongdaekim/pyscenic/thyroid/new_M1_M2/m2_cd163_po_reg_output.loom\"\n",
    "f_out_reg_path = \"/node210data/gyeongdaekim/pyscenic/thyroid/new_M1_M2/m2_cd163_po_reg.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/gyeongdaekim/anaconda3/lib/python3.7/site-packages/dask/dataframe/utils.py:14: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n",
      "\n",
      "2022-10-06 23:08:02,224 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.\n",
      "\n",
      "2022-10-06 23:08:20,612 - pyscenic.cli.pyscenic - INFO - Loading gene signatures.\n",
      "Create regulons from a dataframe of enriched features.\n",
      "Additional columns saved: []\n",
      "\n",
      "2022-10-06 23:08:38,135 - pyscenic.cli.pyscenic - INFO - Calculating cellular enrichment.\n",
      "\n",
      "2022-10-06 23:09:45,109 - pyscenic.cli.pyscenic - INFO - Writing results to file.\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!pyscenic aucell \\\n",
    "    {f_loom_path_scenic} \\\n",
    "    {f_out_reg_path} \\\n",
    "    --output {f_pyscenic_output} \\\n",
    "    --num_workers 12"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualization of SCENIC's AUC matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
