{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pysam \n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_222102/3242569405.py:5: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  GencodeGFF = pd.read_csv(PATH_to_GencodeGFF, sep='\\t', skiprows=skiprows_n, header=None)\n"
     ]
    }
   ],
   "source": [
    "PATH_genome = '/node200data/18parkky/datasets/reference/gd/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa'\n",
    "PATH_gff    = '/node200data/18parkky/datasets/reference/gd/Arabidopsis_thaliana.TAIR10.56.gtf'\n",
    "\n",
    "def LoadGencodeGFF( PATH_to_GencodeGFF, featureOfInterest='exon', skiprows_n=5):\n",
    "    GencodeGFF = pd.read_csv(PATH_to_GencodeGFF, sep='\\t', skiprows=skiprows_n, header=None)\n",
    "    GencodeGFF.columns = ['sequence', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
    "\n",
    "    GencodeGFF.dropna(inplace=True)\n",
    "    if featureOfInterest == None:\n",
    "        pass \n",
    "    else:\n",
    "        GencodeGFF = GencodeGFF[(GencodeGFF['feature']==featureOfInterest)]\n",
    "        \n",
    "    col1, col2, col3, col4 = list(), list(), list(), list()\n",
    "\n",
    "    for tup in GencodeGFF.itertuples():\n",
    "        try:\n",
    "            attributes = tup.attributes.split(';')\n",
    "        except:\n",
    "            print(tup)\n",
    "            raise ValueError\n",
    "        \n",
    "        ENSG, ENST = None, None\n",
    "        GeneName, TranscriptName = None, None\n",
    "        \n",
    "        for attribute in attributes:\n",
    "            if 'gene_id' in attribute:\n",
    "                ENSG = attribute.strip().split(\" \")[1].split('\\\"')[1]\n",
    "                col1.append(ENSG)\n",
    "                \n",
    "            elif 'transcript_id' in attribute:\n",
    "                ENST = attribute.strip().split(\" \")[1].split('\\\"')[1]\n",
    "                col2.append(ENST)\n",
    "\n",
    "            elif 'gene_name' in attribute:\n",
    "                GeneName = attribute.strip().split(\" \")[1].split('\\\"')[1].upper()\n",
    "                col3.append(GeneName)\n",
    "\n",
    "            elif 'transcript_name' in attribute:\n",
    "                TranscriptName = attribute.strip().split(\" \")[1].split('\\\"')[1].upper()\n",
    "                col4.append(TranscriptName)\n",
    "\n",
    "        if ENSG == None:\n",
    "            col1.append('N/A')\n",
    "        if ENST == None:\n",
    "            col2.append('N/A')\n",
    "        if GeneName == None:\n",
    "            col3.append('N/A')\n",
    "        if TranscriptName == None:\n",
    "            col4.append('N/A')\n",
    "        \n",
    "    GencodeGFF['ENSG'] = col1\n",
    "    GencodeGFF['ENST'] = col2\n",
    "    GencodeGFF['GeneName'] = col3\n",
    "    GencodeGFF['TranscriptName'] = col4\n",
    "    \n",
    "    return GencodeGFF\n",
    "\n",
    "gff = LoadGencodeGFF( PATH_gff, featureOfInterest=None, )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome = pysam.FastaFile(PATH_genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'/node200data/18parkky/datasets/reference/gd/CustomTranscriptome.fa', 'w') as fa:\n",
    "    for ENST, edf in gff[gff['feature']=='exon'].groupby('ENST'):\n",
    "        ENSG_e              = edf.iloc[0].ENSG\n",
    "        GeneName_e          = edf.iloc[0].GeneName\n",
    "        TranscriptName_e    = edf.iloc[0].TranscriptName\n",
    "        EntryName = f'{ENST}|{ENSG_e}|{ENST}|{ENSG_e}|{TranscriptName_e}|{GeneName_e}|1|1'\n",
    "        \n",
    "        chrom = edf.iloc[0].sequence\n",
    "        \n",
    "        seq = str()\n",
    "        for tup in edf.sort_values('start').itertuples():\n",
    "            for nt in genome.fetch( str(tup.sequence), tup.start-1, tup.end ): seq += nt.upper()\n",
    "        \n",
    "        fa.write(f'>{EntryName}\\n')\n",
    "        fa.write(f'{seq}\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "def SummarizeAlignment( bamfile, References, GencodeGFF, DIR_out, filename, processNumber ):\n",
    "    \n",
    "    AlignmentSummary = list()\n",
    "\n",
    "    for reference in References:\n",
    "\n",
    "        for read in bamfile.fetch(reference, multiple_iterators=True):\n",
    "            \n",
    "            ReadName = read.query_name.split('|')[0]\n",
    "            CellType = read.query_name.split('|')[1]\n",
    "            CB, UMI  = read.query_name.split('|')[2], read.query_name.split('|')[3]\n",
    "            \n",
    "            ENSMUST = read.reference_name.split('|')[0]\n",
    "            ENSMUSG = read.reference_name.split('|')[1]\n",
    "            GeneName = read.reference_name.split('|')[5]\n",
    "            \n",
    "            GFF_oi = GencodeGFF[GencodeGFF['ENST']==ENSMUST.split('.')[0]].sort_values('start')\n",
    "            GFF_oi['length'] = GFF_oi['end'] - GFF_oi['start'] + 1\n",
    "            GFF_oi.reset_index(inplace=True, drop=True)\n",
    "\n",
    "            if len(GFF_oi)<=1: continue \n",
    "            \n",
    "            dict_ExonNumber_to_PERC     = dict()\n",
    "            AlignedReferencePositions   = set(read.get_reference_positions())\n",
    "            \n",
    "            for idx, tup in enumerate(GFF_oi.itertuples()):\n",
    "                if idx == 0:\n",
    "                    ExonRelativeStartPos    = 1\n",
    "                    ExonRelativeEndPos      = int(tup.length)\n",
    "                else:\n",
    "                    ExonRelativeStartPos    = int(ExonRelativeEndPos + 1)\n",
    "                    ExonRelativeEndPos      = int(ExonRelativeStartPos + tup.length)\n",
    "                    \n",
    "                uncoveredPosCount = 0\n",
    "                    \n",
    "                for pos in range(ExonRelativeStartPos, ExonRelativeEndPos):\n",
    "                    if pos in AlignedReferencePositions:\n",
    "                        pass \n",
    "                    else:\n",
    "                        uncoveredPosCount += 1\n",
    "                \n",
    "                PERC = 100*(1-uncoveredPosCount/tup.length)\n",
    "                dict_ExonNumber_to_PERC[idx] = PERC\n",
    "                \n",
    "\n",
    "            AlignmentSummary.append( [ReadName, CB, UMI, CellType, \n",
    "                                    GeneName, ENSMUSG, ENSMUST, \n",
    "                                    read.mapping_quality, read.flag, read.cigarstring,\n",
    "                                    sum(list(dict_ExonNumber_to_PERC.values())), np.mean(list(dict_ExonNumber_to_PERC.values()))] )\n",
    "        \n",
    "    AlignmentSummary = pd.DataFrame(AlignmentSummary, columns=['ReadName', 'CB', 'UMI', 'CellType', \n",
    "                                                               'GeneName', 'ENSMUSG', 'ENSMUST', \n",
    "                                                               'MAPQ', 'flag', 'cigarstring',\n",
    "                                                               'SumOfPERC', 'AvgOfPERC', ])\n",
    "    \n",
    "    return AlignmentSummary\n",
    "    # AlignmentSummary.to_csv(f'{DIR_out}/{filename}.SummarizeAlignment.temp{processNumber}.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[W::hts_idx_load3] The index file is older than the data file: /node200data/18parkky/datasets/reference/gd/arabidopsis_run1_2_realigned_filtered.bam.bai\n"
     ]
    }
   ],
   "source": [
    "PATH_BAM    = '/node200data/18parkky/datasets/reference/gd/arabidopsis_run1_2_realigned_filtered.bam'\n",
    "bamfile     = pysam.AlignmentFile(PATH_BAM, 'rb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import math \n",
    "threads = 72\n",
    "\n",
    "ChunkSize = math.ceil( len(bamfile.references)/threads )\n",
    "ReferenceIsoforms_chunks = [ bamfile.references[i:i+ChunkSize] for i in range(0, len(bamfile.references), ChunkSize) ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AT1G01010.1\n"
     ]
    }
   ],
   "source": [
    "import numpy as np \n",
    "\n",
    "AlignmentSummary = list()\n",
    "\n",
    "for reference in ReferenceIsoforms_chunks[0]:\n",
    "\n",
    "    for read in bamfile.fetch(reference, multiple_iterators=True):\n",
    "        \n",
    "        ReadName = read.query_name.split('|')[0]\n",
    "        CellType = read.query_name.split('|')[1]\n",
    "        CB, UMI  = read.query_name.split('|')[2], read.query_name.split('|')[3]\n",
    "        \n",
    "        ENSMUST = read.reference_name.split('|')[0]\n",
    "        ENSMUSG = read.reference_name.split('|')[1]\n",
    "        GeneName = read.reference_name.split('|')[5]\n",
    "        \n",
    "        print(ENSMUST)\n",
    "        \n",
    "        \n",
    "        GFF_oi = gff[gff['ENST']==ENSMUST].sort_values('start')\n",
    "        GFF_oi['length'] = GFF_oi['end'] - GFF_oi['start'] + 1\n",
    "        GFF_oi.reset_index(inplace=True, drop=True)\n",
    "\n",
    "        if len(GFF_oi)<=1: \n",
    "            continue \n",
    "        \n",
    "        dict_ExonNumber_to_PERC     = dict()\n",
    "        AlignedReferencePositions   = set(read.get_reference_positions())\n",
    "        \n",
    "        for idx, tup in enumerate(GFF_oi.itertuples()):\n",
    "            if idx == 0:\n",
    "                ExonRelativeStartPos    = 1\n",
    "                ExonRelativeEndPos      = int(tup.length)\n",
    "            else:\n",
    "                ExonRelativeStartPos    = int(ExonRelativeEndPos + 1)\n",
    "                ExonRelativeEndPos      = int(ExonRelativeStartPos + tup.length)\n",
    "                \n",
    "            uncoveredPosCount = 0\n",
    "                \n",
    "            for pos in range(ExonRelativeStartPos, ExonRelativeEndPos):\n",
    "                if pos in AlignedReferencePositions:\n",
    "                    pass \n",
    "                else:\n",
    "                    uncoveredPosCount += 1\n",
    "            \n",
    "            PERC = 100*(1-uncoveredPosCount/tup.length)\n",
    "            dict_ExonNumber_to_PERC[idx] = PERC\n",
    "            \n",
    "\n",
    "        AlignmentSummary.append( [ReadName, CB, UMI, CellType, \n",
    "                                GeneName, ENSMUSG, ENSMUST, \n",
    "                                read.mapping_quality, read.flag, read.cigarstring,\n",
    "                                sum(list(dict_ExonNumber_to_PERC.values())), np.mean(list(dict_ExonNumber_to_PERC.values()))] )\n",
    "        \n",
    "    break\n",
    "AlignmentSummary = pd.DataFrame(AlignmentSummary, columns=['ReadName', 'CB', 'UMI', 'CellType', \n",
    "                                                            'GeneName', 'ENSMUSG', 'ENSMUST', \n",
    "                                                            'MAPQ', 'flag', 'cigarstring',\n",
    "                                                            'SumOfPERC', 'AvgOfPERC', ])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ReadName</th>\n",
       "      <th>CB</th>\n",
       "      <th>UMI</th>\n",
       "      <th>CellType</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>ENSMUSG</th>\n",
       "      <th>ENSMUST</th>\n",
       "      <th>MAPQ</th>\n",
       "      <th>flag</th>\n",
       "      <th>cigarstring</th>\n",
       "      <th>SumOfPERC</th>\n",
       "      <th>AvgOfPERC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>05fb7733-3e93-4669-91c9-294162dcce05_R_0</td>\n",
       "      <td>AGTACCACAAGTATCC</td>\n",
       "      <td>AAGTCCGTACAT</td>\n",
       "      <td>2</td>\n",
       "      <td>NAC001</td>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>AT1G01010.1</td>\n",
       "      <td>60</td>\n",
       "      <td>0</td>\n",
       "      <td>3S1476M2D72M24S</td>\n",
       "      <td>68.267959</td>\n",
       "      <td>4.015762</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   ReadName                CB           UMI  \\\n",
       "0  05fb7733-3e93-4669-91c9-294162dcce05_R_0  AGTACCACAAGTATCC  AAGTCCGTACAT   \n",
       "\n",
       "  CellType GeneName    ENSMUSG      ENSMUST  MAPQ  flag      cigarstring  \\\n",
       "0        2   NAC001  AT1G01010  AT1G01010.1    60     0  3S1476M2D72M24S   \n",
       "\n",
       "   SumOfPERC  AvgOfPERC  \n",
       "0  68.267959   4.015762  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AlignmentSummary"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "sc",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
