{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os \n",
    "import glob\n",
    "import pysam\n",
    "import subprocess\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ~ 1 minute\n",
    "alleleTable = pd.read_csv('/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/SG-BULK/AlleleTable.concat.tsv', sep='\\t')\n",
    "alleleTable['diff'] = alleleTable['read_STR_length'] - alleleTable['reference_STR_allele']\n",
    "\n",
    "crude_STR_profile = list()\n",
    "for patient_id, edf in alleleTable.groupby(\"patient_id\"):\n",
    "    crude_STR_profile.append( [patient_id, len(edf), np.mean( edf['diff'] )] )\n",
    "crude_STR_profile = pd.DataFrame(crude_STR_profile, columns=['sample_id', 'n_reads', 'avg_diff'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DIR_FASTQ = '/node200data/18parkky/datasets/data/public/FASTQ/SG-BULK'\n",
    "DIR_FASTQ_MSI = f'{DIR_FASTQ}/MSI'\n",
    "DIR_FASTQ_MSS = f'{DIR_FASTQ}/MSS'\n",
    "\n",
    "list_DIR_FASTQ_MSI_by_sample = [ d for d in glob.glob( f'{DIR_FASTQ_MSI}/*' ) if os.path.isdir(d)==True and len(d.split(\"_\")) == 1 and 'EGAF' not in d.split(\"/\")[-1] ]\n",
    "list_DIR_FASTQ_MSS_by_sample = [ d for d in glob.glob( f'{DIR_FASTQ_MSS}/*' ) if os.path.isdir(d)==True and len(d.split(\"_\")) == 1 and 'EGAF' not in d.split(\"/\")[-1] ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "({'1054', '1236', '1253', '1627', '1657', '609', '612', '649'}, 8)"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DIR_BAM_out = '/node200data/18parkky/datasets/data/public/BAM/Joanito_et_al/SG-BULK/nonhost-reads'\n",
    "DIR_processed_BAM_MSI_out = f'{DIR_BAM_out}/MSI'\n",
    "DIR_processed_BAM_MSS_out = f'{DIR_BAM_out}/MSS'\n",
    "\n",
    "processed_MSI_samples = { d.split(\"/\")[-1].split(\".\")[0] for d in glob.glob(f'{DIR_processed_BAM_MSI_out}/*') if os.path.isfile(d)==True and d.split(\"/\")[-1].split(\".\")[-1] == 'bam'}\n",
    "processed_MSS_samples = { d.split(\"/\")[-1].split(\".\")[0] for d in glob.glob(f'{DIR_processed_BAM_MSS_out}/*') if os.path.isfile(d)==True and d.split(\"/\")[-1].split(\".\")[-1] == 'bam'}\n",
    "\n",
    "processed_MSI_samples, len(processed_MSI_samples)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run bowtie2 to human genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DIR_BAM_MSI_out = '/node200data/18parkky/datasets/data/public/BAM/Joanito_et_al/SG-BULK/MSI/raw'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "# (1) Run bowtie2 to human reference genome\n",
    "#! Bowtie 2 parameters\n",
    "num_threads = 12\n",
    "DIR_bowtie2_index = '/node200data/18parkky/datasets/reference/z_program_references/bowtie2/hg38/hg38'\n",
    "\n",
    "\n",
    "with open( '/home/18parkky/git/CRC_MSI_intensity_analysis/scripts/Joanito/bulkRNAseq/microbial_read_analysis/temp.sh', 'w' ) as bash_script:\n",
    "    # MSI\n",
    "    for DIR_FASTQ_MSI in list_DIR_FASTQ_MSI_by_sample:\n",
    "        sample_id = DIR_FASTQ_MSI.split(\"/\")[-1]\n",
    "        \n",
    "        if sample_id in processed_MSI_samples:\n",
    "            continue\n",
    "        \n",
    "        else:        \n",
    "            # print(f'Running pipeline for {sample_id}', flush=True)\n",
    "            \n",
    "            PATH_R1, PATH_R2 = f'{DIR_FASTQ_MSI}/{sample_id}_R1.fastq.gz', f'{DIR_FASTQ_MSI}/{sample_id}_R2.fastq.gz'\n",
    "            PATH_BAM_out = f'{DIR_BAM_MSI_out}/{sample_id}.sorted.bam'\n",
    "            cmd = f'bowtie2 -p {num_threads} -x {DIR_bowtie2_index} -1 {PATH_R1} -2 {PATH_R2} | samtools view -Sb | samtools sort -@ {num_threads} > {PATH_BAM_out}'\n",
    "            bash_script.write(f'{cmd}\\n')\n",
    "            cmd = f'samtools index {PATH_BAM_out} -@ {num_threads}'\n",
    "            bash_script.write(f'{cmd}\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# (2) Extract unmapped reads\n",
    "with open( '/home/18parkky/git/CRC_MSI_intensity_analysis/scripts/Joanito/bulkRNAseq/microbial_read_analysis/temp.sh', 'w' ) as bash_script:\n",
    "    # MSI\n",
    "    for DIR_FASTQ_MSI in list_DIR_FASTQ_MSI_by_sample:\n",
    "        sample_id = DIR_FASTQ_MSI.split(\"/\")[-1]\n",
    "        \n",
    "        if sample_id in processed_MSI_samples:\n",
    "            continue\n",
    "        \n",
    "        else:       \n",
    "            \n",
    "            \n",
    "            PATH_unmapped_BAM_out = f'{DIR_BAM_MSI_out}/{sample_id}.unmapped.bam'\n",
    "            \n",
    "            bamfile = pysam.AlignmentFile( PATH_BAM_out, 'rb' )\n",
    "            unmapped_read_bamfile  = pysam.AlignmentFile( PATH_unmapped_BAM_out, 'wb', template=bamfile )\n",
    "\n",
    "            for read in bamfile.fetch():\n",
    "                if read.is_unmapped==True:\n",
    "                    unmapped_read_bamfile.write( read )\n",
    "\n",
    "            unmapped_read_bamfile.close()\n",
    "            cmd = f'samtools index {PATH_unmapped_BAM_out} -@ {num_threads}'\n",
    "            bash_script.write(f'{cmd}\\n')\n",
    "            \n",
    "            # (4) Convert unmapped_paired_read_bamfile into FASTQ\n",
    "            DIR_nonhost_converted_FASTQ = f'/node200data/18parkky/datasets/data/public/FASTQ/SG-BULK/nonhost_reads/MSI'\n",
    "\n",
    "            # Unpaired reads\n",
    "            converted_fastq = f'{DIR_nonhost_converted_FASTQ}/{sample_id}.nonhost.fq'\n",
    "            cmd = f'bedtools bamtofastq -i {PATH_unmapped_BAM_out} -fq {converted_fastq}'\n",
    "            bash_script.write(f'{cmd}\\n')\n",
    "            cmd = f'gzip {converted_fastq}'\n",
    "            bash_script.write(f'{cmd}\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Align nonhost reads to human microbial genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "DIR_MSI_nonhost_FASTQ = '/node200data/18parkky/datasets/data/public/FASTQ/SG-BULK/nonhost_reads/MSI'\n",
    "list_PATH_MSI_nonhost_FASTQ = glob.glob(f'{DIR_MSI_nonhost_FASTQ}/*.fq.gz')\n",
    "DIR_bowtie2_index = '/node200data/18parkky/datasets/reference/z_program_references/bowtie2/HRGM/HRGM'\n",
    "num_threads = 12\n",
    "\n",
    "DIR_HRGM_alignment_out = '/node200data/18parkky/datasets/data/public/BAM/Joanito_et_al/SG-BULK/nonhost-reads/MSI'\n",
    "\n",
    "bash_script = '/node200data/18parkky/datasets/data/public/BAM/Joanito_et_al/SG-BULK/nonhost-reads/MSI/run_bowtie2.HRGM.sh'\n",
    "with open(bash_script, 'w') as sh:\n",
    "    for PATH_MSI_nonhost_FASTQ in list_PATH_MSI_nonhost_FASTQ:\n",
    "        sample_id = PATH_MSI_nonhost_FASTQ.split(\"/\")[-1].split(\".\")[0]\n",
    "        cmd = f'bowtie2 -p {num_threads} -x {DIR_bowtie2_index} {PATH_MSI_nonhost_FASTQ} | samtools view -Sb | samtools sort -@ {num_threads} > {DIR_HRGM_alignment_out}/{sample_id}.HRGM.sorted.bam'\n",
    "        sh.write(f'{cmd}\\n')\n",
    "        cmd = f'samtools index {DIR_HRGM_alignment_out}/{sample_id}.HRGM.sorted.bam -@ {num_threads}'\n",
    "        sh.write(f'{cmd}\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "list_PATH_MSI_nonhost_HRGM_BAM = glob.glob(f'{DIR_HRGM_alignment_out}/*HRGM.sorted.bam')\n",
    "\n",
    "for PATH_MSI_nonhost_HRGM_BAM in list_PATH_MSI_nonhost_HRGM_BAM:\n",
    "    sample_id = PATH_MSI_nonhost_HRGM_BAM.split(\"/\")[-1].split(\".\")[0]\n",
    "    PATH_MSI_nonhost_HRGM_mapped_BAM = f'{DIR_HRGM_alignment_out}/{sample_id}.HRGM.sorted.mapped.bam'\n",
    "    if os.path.exists(PATH_MSI_nonhost_HRGM_mapped_BAM) == True:    continue\n",
    "    \n",
    "    origBam = pysam.AlignmentFile(PATH_MSI_nonhost_HRGM_BAM, 'rb')\n",
    "    procBam = pysam.AlignmentFile(PATH_MSI_nonhost_HRGM_mapped_BAM, 'wb', template=origBam)\n",
    "    for read in origBam.fetch():\n",
    "        if read.is_mapped==True: procBam.write(read)\n",
    "    origBam.close()\n",
    "    procBam.close()\n",
    "    \n",
    "    os.remove( PATH_MSI_nonhost_HRGM_BAM )\n",
    "    pysam.index(PATH_MSI_nonhost_HRGM_mapped_BAM, '-@', str(num_threads))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analyze results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "HRGM_metadata = pd.read_csv('/node200data/18parkky/datasets/reference/genome/microbiome/gut/HRGM_Genomes_metadata.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "dict_HRGMname_to_metadata = { list(tup)[1] : [ list(tup)[3].split(\";\")[0].split(\"__\")[1],  # domain\n",
    "                                               list(tup)[3].split(\";\")[1].split(\"__\")[1],  # phylum\n",
    "                                               list(tup)[3].split(\";\")[2].split(\"__\")[1],  # class\n",
    "                                               list(tup)[3].split(\";\")[3].split(\"__\")[1],  # order\n",
    "                                               list(tup)[3].split(\";\")[4].split(\"__\")[1],  # family\n",
    "                                               list(tup)[3].split(\";\")[5].split(\"__\")[1],  # genus\n",
    "                                               list(tup)[3].split(\";\")[6].split(\"__\")[1],  # species\n",
    "\n",
    "                                              list(tup)[4],     # Member count \n",
    "                                              list(tup)[5],     # continents\n",
    "                                              list(tup)[6],     # country\n",
    "                                              ] for tup in HRGM_metadata.itertuples() }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Bacteria composition inference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "list_PATH_MSI_nonhost_HRGM_BAM = glob.glob(f'/node200data/18parkky/datasets/data/public/BAM/Joanito_et_al/SG-BULK/nonhost-reads/MSI/*HRGM.sorted.mapped.bam')\n",
    "\n",
    "list_composition_summary = list()\n",
    "\n",
    "for PATH_MSI_nonhost_HRGM_BAM in list_PATH_MSI_nonhost_HRGM_BAM:\n",
    "    sample_id = PATH_MSI_nonhost_HRGM_BAM.split(\"/\")[-1].split(\".\")[0]\n",
    "    HRGM_BAM = pysam.AlignmentFile( PATH_MSI_nonhost_HRGM_BAM, 'rb' )\n",
    "    dict_HRGMname_to_count = dict()\n",
    "    \n",
    "    for read in HRGM_BAM.fetch():\n",
    "        HRGM_name = \"_\".join(read.reference_name.split(\"_\")[:-1])\n",
    "        # HRGM_name_member = read.reference_name\n",
    "        \n",
    "        try: dict_HRGMname_to_count[HRGM_name] += 1\n",
    "        except KeyError: dict_HRGMname_to_count[HRGM_name] = 1\n",
    "\n",
    "    total_count = sum(dict_HRGMname_to_count.values())\n",
    "    for HRGMname, count_e in dict_HRGMname_to_count.items():\n",
    "        list_composition_summary.append( [sample_id, HRGMname, count_e, 100*count_e/total_count] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>HRGM_name</th>\n",
       "      <th>count</th>\n",
       "      <th>perc</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>member_count</th>\n",
       "      <th>continents</th>\n",
       "      <th>country</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1236</td>\n",
       "      <td>HRGM_Genome_0025</td>\n",
       "      <td>3</td>\n",
       "      <td>0.021274</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Enterobacter_A</td>\n",
       "      <td>Enterobacter_A timonensis</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1236</td>\n",
       "      <td>HRGM_Genome_0026</td>\n",
       "      <td>1</td>\n",
       "      <td>0.007091</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Citrobacter</td>\n",
       "      <td>Citrobacter portucalensis_A</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_id         HRGM_name  count      perc    domain          phylum  \\\n",
       "0      1236  HRGM_Genome_0025      3  0.021274  Bacteria  Proteobacteria   \n",
       "1      1236  HRGM_Genome_0026      1  0.007091  Bacteria  Proteobacteria   \n",
       "\n",
       "                 class             order              family           genus  \\\n",
       "0  Gammaproteobacteria  Enterobacterales  Enterobacteriaceae  Enterobacter_A   \n",
       "1  Gammaproteobacteria  Enterobacterales  Enterobacteriaceae     Citrobacter   \n",
       "\n",
       "                       species  member_count continents country  \n",
       "0    Enterobacter_A timonensis             1        NaN     NaN  \n",
       "1  Citrobacter portucalensis_A             2        NaN     NaN  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "composition_summary = pd.DataFrame(list_composition_summary, columns=['sample_id', 'HRGM_name', 'count', 'perc'])\n",
    "composition_summary['domain']   = [ dict_HRGMname_to_metadata[HRGM_name][0] for HRGM_name in composition_summary['HRGM_name'] ]\n",
    "composition_summary['phylum']   = [ dict_HRGMname_to_metadata[HRGM_name][1] for HRGM_name in composition_summary['HRGM_name'] ]\n",
    "composition_summary['class']    = [ dict_HRGMname_to_metadata[HRGM_name][2] for HRGM_name in composition_summary['HRGM_name'] ]\n",
    "composition_summary['order']    = [ dict_HRGMname_to_metadata[HRGM_name][3] for HRGM_name in composition_summary['HRGM_name'] ]\n",
    "composition_summary['family']   = [ dict_HRGMname_to_metadata[HRGM_name][4] for HRGM_name in composition_summary['HRGM_name'] ]\n",
    "composition_summary['genus']    = [ dict_HRGMname_to_metadata[HRGM_name][5] for HRGM_name in composition_summary['HRGM_name'] ]\n",
    "composition_summary['species']  = [ dict_HRGMname_to_metadata[HRGM_name][6] for HRGM_name in composition_summary['HRGM_name'] ]\n",
    "\n",
    "composition_summary['member_count']     = [ dict_HRGMname_to_metadata[HRGM_name][7] for HRGM_name in composition_summary['HRGM_name'] ]\n",
    "composition_summary['continents']       = [ dict_HRGMname_to_metadata[HRGM_name][8] for HRGM_name in composition_summary['HRGM_name'] ]\n",
    "composition_summary['country']          = [ dict_HRGMname_to_metadata[HRGM_name][9] for HRGM_name in composition_summary['HRGM_name'] ]\n",
    "\n",
    "composition_summary.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>HRGM_name</th>\n",
       "      <th>count</th>\n",
       "      <th>perc</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>member_count</th>\n",
       "      <th>continents</th>\n",
       "      <th>country</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>932</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_0388</td>\n",
       "      <td>317</td>\n",
       "      <td>0.911522</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Fusobacteriota</td>\n",
       "      <td>Fusobacteriia</td>\n",
       "      <td>Fusobacteriales</td>\n",
       "      <td>Fusobacteriaceae</td>\n",
       "      <td>Fusobacterium</td>\n",
       "      <td>Fusobacterium nucleatum</td>\n",
       "      <td>2</td>\n",
       "      <td>NA;Asia</td>\n",
       "      <td>Japan;NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>941</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_0397</td>\n",
       "      <td>7</td>\n",
       "      <td>0.020128</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Fusobacteriota</td>\n",
       "      <td>Fusobacteriia</td>\n",
       "      <td>Fusobacteriales</td>\n",
       "      <td>Fusobacteriaceae</td>\n",
       "      <td>Fusobacterium</td>\n",
       "      <td>Fusobacterium nucleatum_E</td>\n",
       "      <td>2</td>\n",
       "      <td>Asia</td>\n",
       "      <td>China</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1557</th>\n",
       "      <td>649</td>\n",
       "      <td>HRGM_Genome_0388</td>\n",
       "      <td>125</td>\n",
       "      <td>0.443782</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Fusobacteriota</td>\n",
       "      <td>Fusobacteriia</td>\n",
       "      <td>Fusobacteriales</td>\n",
       "      <td>Fusobacteriaceae</td>\n",
       "      <td>Fusobacterium</td>\n",
       "      <td>Fusobacterium nucleatum</td>\n",
       "      <td>2</td>\n",
       "      <td>NA;Asia</td>\n",
       "      <td>Japan;NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1564</th>\n",
       "      <td>649</td>\n",
       "      <td>HRGM_Genome_0397</td>\n",
       "      <td>1</td>\n",
       "      <td>0.003550</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Fusobacteriota</td>\n",
       "      <td>Fusobacteriia</td>\n",
       "      <td>Fusobacteriales</td>\n",
       "      <td>Fusobacteriaceae</td>\n",
       "      <td>Fusobacterium</td>\n",
       "      <td>Fusobacterium nucleatum_E</td>\n",
       "      <td>2</td>\n",
       "      <td>Asia</td>\n",
       "      <td>China</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2312</th>\n",
       "      <td>612</td>\n",
       "      <td>HRGM_Genome_0388</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007951</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Fusobacteriota</td>\n",
       "      <td>Fusobacteriia</td>\n",
       "      <td>Fusobacteriales</td>\n",
       "      <td>Fusobacteriaceae</td>\n",
       "      <td>Fusobacterium</td>\n",
       "      <td>Fusobacterium nucleatum</td>\n",
       "      <td>2</td>\n",
       "      <td>NA;Asia</td>\n",
       "      <td>Japan;NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2847</th>\n",
       "      <td>1657</td>\n",
       "      <td>HRGM_Genome_0388</td>\n",
       "      <td>114</td>\n",
       "      <td>0.588904</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Fusobacteriota</td>\n",
       "      <td>Fusobacteriia</td>\n",
       "      <td>Fusobacteriales</td>\n",
       "      <td>Fusobacteriaceae</td>\n",
       "      <td>Fusobacterium</td>\n",
       "      <td>Fusobacterium nucleatum</td>\n",
       "      <td>2</td>\n",
       "      <td>NA;Asia</td>\n",
       "      <td>Japan;NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2855</th>\n",
       "      <td>1657</td>\n",
       "      <td>HRGM_Genome_0397</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010332</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Fusobacteriota</td>\n",
       "      <td>Fusobacteriia</td>\n",
       "      <td>Fusobacteriales</td>\n",
       "      <td>Fusobacteriaceae</td>\n",
       "      <td>Fusobacterium</td>\n",
       "      <td>Fusobacterium nucleatum_E</td>\n",
       "      <td>2</td>\n",
       "      <td>Asia</td>\n",
       "      <td>China</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     sample_id         HRGM_name  count      perc    domain          phylum  \\\n",
       "932       1054  HRGM_Genome_0388    317  0.911522  Bacteria  Fusobacteriota   \n",
       "941       1054  HRGM_Genome_0397      7  0.020128  Bacteria  Fusobacteriota   \n",
       "1557       649  HRGM_Genome_0388    125  0.443782  Bacteria  Fusobacteriota   \n",
       "1564       649  HRGM_Genome_0397      1  0.003550  Bacteria  Fusobacteriota   \n",
       "2312       612  HRGM_Genome_0388      2  0.007951  Bacteria  Fusobacteriota   \n",
       "2847      1657  HRGM_Genome_0388    114  0.588904  Bacteria  Fusobacteriota   \n",
       "2855      1657  HRGM_Genome_0397      2  0.010332  Bacteria  Fusobacteriota   \n",
       "\n",
       "              class            order            family          genus  \\\n",
       "932   Fusobacteriia  Fusobacteriales  Fusobacteriaceae  Fusobacterium   \n",
       "941   Fusobacteriia  Fusobacteriales  Fusobacteriaceae  Fusobacterium   \n",
       "1557  Fusobacteriia  Fusobacteriales  Fusobacteriaceae  Fusobacterium   \n",
       "1564  Fusobacteriia  Fusobacteriales  Fusobacteriaceae  Fusobacterium   \n",
       "2312  Fusobacteriia  Fusobacteriales  Fusobacteriaceae  Fusobacterium   \n",
       "2847  Fusobacteriia  Fusobacteriales  Fusobacteriaceae  Fusobacterium   \n",
       "2855  Fusobacteriia  Fusobacteriales  Fusobacteriaceae  Fusobacterium   \n",
       "\n",
       "                        species  member_count continents   country  \n",
       "932     Fusobacterium nucleatum             2    NA;Asia  Japan;NA  \n",
       "941   Fusobacterium nucleatum_E             2       Asia     China  \n",
       "1557    Fusobacterium nucleatum             2    NA;Asia  Japan;NA  \n",
       "1564  Fusobacterium nucleatum_E             2       Asia     China  \n",
       "2312    Fusobacterium nucleatum             2    NA;Asia  Japan;NA  \n",
       "2847    Fusobacterium nucleatum             2    NA;Asia  Japan;NA  \n",
       "2855  Fusobacterium nucleatum_E             2       Asia     China  "
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "composition_summary[composition_summary['species'].isin(['Fusobacterium nucleatum_E', 'Fusobacterium nucleatum'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1054 34777\n",
      "1236 14102\n",
      "1253 18097\n",
      "1627 24376\n",
      "1657 19358\n",
      "609 15732\n",
      "612 25154\n",
      "649 28167\n"
     ]
    }
   ],
   "source": [
    "for sample_id, edf in composition_summary.groupby(\"sample_id\"):\n",
    "    print(sample_id, sum(edf['count']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>n_reads</th>\n",
       "      <th>avg_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1054</td>\n",
       "      <td>1011050</td>\n",
       "      <td>-1.242936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1253</td>\n",
       "      <td>1414755</td>\n",
       "      <td>-1.016385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1657</td>\n",
       "      <td>1952344</td>\n",
       "      <td>-0.981056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>612</td>\n",
       "      <td>2057903</td>\n",
       "      <td>-0.920462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>649</td>\n",
       "      <td>1242088</td>\n",
       "      <td>-0.888940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1627</td>\n",
       "      <td>1708005</td>\n",
       "      <td>-0.872526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1236</td>\n",
       "      <td>1343181</td>\n",
       "      <td>-0.635272</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_id  n_reads  avg_diff\n",
       "7        1054  1011050 -1.242936\n",
       "10       1253  1414755 -1.016385\n",
       "15       1657  1952344 -0.981056\n",
       "1         612  2057903 -0.920462\n",
       "2         649  1242088 -0.888940\n",
       "14       1627  1708005 -0.872526\n",
       "9        1236  1343181 -0.635272"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crude_STR_profile[crude_STR_profile['sample_id'].isin( [ int(s) for s in composition_summary['sample_id'] ] )]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import pearsonr\n",
    "\n",
    "pearsonr(d['bacterial_alignments'], d['STR_profile'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Axes: xlabel='bacterial_alignments', ylabel='STR_profile'>"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAz4AAAIiCAYAAAAJqB2MAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8hTgPZAAAACXBIWXMAABcSAAAXEgFnn9JSAAA+60lEQVR4nO3dfZwdZX3//9cn2SxJgA2CBgwKJBFxDUUQlShK0MQbQCoqflt/RqSWCvr9ive2P6x+qa14U29qtTVYtWipeA+oUK1YE2+RKgI1BJQgdwaCCGaBbFg2+Xz/mFk4HM7Zu3N2T3bO6/l4zGP2XHPNNdfZM5ns+8zMNZGZSJIkSVKVzep0ByRJkiRpqhl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFVeT6c7oImLiNuA+cDNne6LJEmS1CaPBbZm5j5T0Xhk5lS0qykUEQO77LLL7kuXLu10VyRJkqS22LhxI/fdd9/dmdk3Fe17xmdmunnp0qVPXL9+faf7IUmSJLXFsmXLuPrqq6fsiibv8ZEkSZJUeQYfSZIkSZVn8JEkSZJUeQYfSZIkSZVn8JEkSZJUeQYfSZIkSZVn8JEkSZJUeQYfSZIkSZVn8JEkSZJUeQYfSZIkSZVn8JEkSZJUeQYfSZIkSZVn8JEkSZJUeT2d7oBmhuHtOxjavoOBwWG2DA6xYF4vffN66J09i57Z5mdJkiTt3Aw+GtPWoWEu2bCZs9ddz/pNAw+UL1vUx6krlrCqf2/m97orSZIkaeflV/Ua1dahYc66eAOnn3fFQ0IPwPpNA5x+3hWcdfE1bB0a7lAPJUmSpLEZfNTU8PYdXLJhM+deetOo9c699Ea+u+F2hrfvmKaeSZIkSRNj8FFTQ9t3sGbt9eOqu2bdRoYMPpIkSdpJGXzU1MDgMFffOjB2RYrL3gYGvdxNkiRJOyeDj5raMjg0ofoD2+6fop5IkiRJrTH4qKkF83onVL9v7pwp6okkSZLUGoOPmuqb18OyRX3jqrtsUR998xzSWpIkSTsng4+a6p09i1NXLBlX3dNWLKXXB5lKkiRpJ+VfqmqqZ/YsVvXvzerl+41ab/Xy/VnZv5Aeg48kSZJ2Ul6bpFHN7+3hjGP7OWLxXqxZt/EhDzFdtqiP01YsZWX/Qub3uitJkiRp5+VfqxrT/N4ejjl4H1b2L2RgcJiBbffTN3cOffN66J09yzM9kiRJ2ukZfDQuPWXAmd/bwz4L5na6O5IkSdKE+FW9JEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+AARcWREXBwRd0bEPRFxWUScNIl2boiIHGO6firegyRJkqTmejrdgU6LiJcCX6QIgd8H7gBWAp+NiEMy860TaO4rwCObLFsBHAD8YPK9lSRJkjQZXR18ImJP4DPAbOClmfm1snxv4IfAWyLim5m5djztNQtJETELuKV8+W+t9luSJEnSxHT7pW6nAH3AhSOhByAzNwNvL1++pQ3bWQk8Gvgt8F9taE+SJEnSBHR78DmunH+lwbKLgG3AqoiY2+J2Vpfzz2fmjhbbkiRJkjRB3R58nlTOL69fkJlDwC+BucDjJ7uBiJgHvLh8ee5k25EkSZI0eV17j09E9AELype3NKl2C/AUYH/gqklu6gRgd+CqzJxQGxGxvsmipZPsiyRJktSVuvmMz241P29tUufecr57C9t5ZTl3UANJkiSpQ2b0GZ+IOB/on+BqJ2XmZVPRn3oRsRB4LrAD+PxE18/MZU3aXQ88sbXeSZIkSd1jRgcfYDFw0ATXmV/O76krG2hQd9dyfvcEtzHiTyl+x9/JzE2TbEOSJElSi2Z08MnMQ1tYdyAitlDc5/MY4OoG1R5Tzm+c5GZGRnNzUANJkiSpg7r5Hh+AK8v5k+sXRMQc4GCKIa1/NdGGI+LxwFMp7h/62hjVJUmSJE2hbg8+F5XzExsseyHFUNaXZOa2SbQ9crbn/My8Z9SakiRJkqZUtwefT1Hc2/OiiHjJSGE5KMEHypcfql8pIq4pp31HafsV5dzR3CRJkqQOm9H3+LQqM++MiFcDXwK+EhFrgd8Dq4A9gA9n5toGq44MqDCnUbsR8QxgCXAbcEl7ey1JkiRporo6+ABk5lcj4ijgr4HlQC/FQAcfz8zPTrLZkcvczsvM7W3opiRJkqQWdH3wAcjMHwHHTKB+jLH8dcDrWu2XJEmSpPbo9nt8JEmSJHUBg48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8gw8QEUdGxMURcWdE3BMRl0XESZNsa0FEnBUR6yNia0Rsi4hrI+IjEbGw3X2XJEmSNLaeTneg0yLipcAXKULg94E7gJXAZyPikMx86wTaeiTwY+BA4DbgO+WipwFvBP4kIp6emTe27x1IkiRJGktXn/GJiD2BzwCzgRMz8+jMPBF4AnAd8JaIOHoCTZ5BEXq+DizOzBdl5ouAxcD5wKOBd7fvHUiSJEkaj64OPsApQB9wYWZ+baQwMzcDby9fvmUC7R1Vzt+bmdtq2tsG/G358qmT764kSZKkyej24HNcOf9Kg2UXAduAVRExd5zt3TeOOr8fZ1uSJEmS2qTbg8+Tyvnl9Qsycwj4JTAXePw42/vPcv5XtWGp/Pmd5ctPT66rkiRJkiarawc3iIg+YEH58pYm1W4BngLsD1w1jmY/CKwAXgT8JiJ+WpYfQRGg3pqZ50ygj+ubLFo63jYkSZIkdXHwAXar+Xlrkzr3lvPdx9NgZt4bEccBnwRWUwSgEd8DfjjRTkqSJElq3YwOPhFxPtA/wdVOyszLpqg/+1HcG/Ro4CTgW+WiY4CPAGsj4nmZ+YPxtJeZy5psZz3wxNZ7LEmSJHWHGR18KIaJPmiC68wv5/fUlQ00qLtrOb97nG1/FjgYeHFmXlBT/rmIuAf4KsXlcEeMu7eSJEmSWjajg09mHtrCugMRsYXiPp/HAFc3qPaYcj7mA0cj4rHA0RQju32jQZULgSHgqRExt3a4a0mSJElTq9tHdbuynD+5fkFEzKE4e7MN+NU42hoJSfdm5vb6hWXZvUAAe0yms5IkSZImp9uDz0Xl/MQGy15IMRLbJeM8O3NbOd8zIhbXL4yIpcAjKMLPHZPoqyRJkqRJ6vbg8ymKe3teFBEvGSmMiIXAB8qXH6pfKSKuKad9R8oy8zc8OOT12RGxoKb+HsDZ5csLMnO4re9CkiRJ0qhm9D0+rcrMOyPi1cCXgK9ExFrg98AqisvRPpyZaxusOjKgwpy68tcAlwDPBa6reY7PcmAv4Abg7e17B5IkSZLGo9vP+JCZXwWOAr4NHAYcC1wHnJyZb5lgWz8FDgX+heJM0irgOcBm4H3A4Zm5qW2dlyRJkjQuXX3GZ0Rm/ojiWTvjrR+jLNtIceZHkiRJ0k6i68/4SJIkSao+g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkypuS4BMRz4uI8yPitxFxX0R8umbZ8yPiwxGxaCq2LUmSJEn12h58IuKjwH8ALwJ2B+YAUVPlVuCNwJ+0e9uSJEmS1Ehbg09EnAS8Hvg58OTM7Kuvk5lXATcDx7dz25IkSZLUTE+b23st8AfguMz83Sj1rgL+qM3bliSpJcPbdzC0fQcDg8NsGRxiwbxe+ub10Dt7Fj2zvS1Wkmaydgefg4F1Y4QegC3A3m3etiRJk7Z1aJhLNmzm7HXXs37TwAPlyxb1ceqKJazq35v5ve3+b1OSNF2m4uurHEedRcDgFGxbkqQJ2zo0zFkXb+D08654SOgBWL9pgNPPu4KzLr6GrUPDHeqhJKlV7Q4+vwaeHBFzmlWIiN2BQ4H1bd62JEkTNrx9B5ds2My5l940ar1zL72R7264neHtO6apZ5Kkdmp38Pky8GjgfaPUeS+wAPhCm7ctSdKEDW3fwZq114+r7pp1Gxky+EjSjNTui5X/AfhT4I0R8QzgwrJ8aUS8CXgx8EzgcuBf2rxtSZImbGBwmKtvHRi7IsVlbwODw97rI0kzUFuP3Jk5GBGrgHOAY4CnlYueVU4A3wFWZ+ZQO7ctSdJkbBmc2H9HA9vuZ58Fc6eoN5KkqdL2r6zKEd2Oi4gnAc8DDqC4pO4W4DuZeVm7tylJ0mQtmNc7ofp9c5vexipJ2olN2bn6zLwSuHKq2pckqR365vWwbFHfw0Zza2TZoj765nmZmyTNRD6NTZLU1Xpnz+LUFUvGVfe0FUvp9UGmkjQjtfS1VUQc1cr6mfn9VtaXJKlVPbNnsap/b1Yv32/UIa1XL9+flf0L6TH4SNKM1Or5+rWM74GlzcxucfuSJLVsfm8PZxzbzxGL92LNuo0Puext2aI+TluxlJX9Cx3NbQYa3r6Doe07GBgcZsvgEAvm9dI3r4fe2bMMsVKXafUI/jlaCz6SJO0U5vf2cMzB+7CyfyEDg8MMbLufvrlz/CN5Bts6NMwlGzZz9rrrHxZmT12xhFX9extmpS7S0r/2zDy5Tf2QJKnjesqAM7+3xyGrZ7itQ8OcdfGGhpcvrt80wOnnXcHq5ftzxrFPMPxIXcKvryRJUqUMb9/BJRs2j3rPFsC5l97IdzfczvD2HdPUM0mdZPCRJEmVMrR9B2vWXj+uumvWbWTI4CN1hVZHdTup/PH8zLy75vW4ZObnWtm+JElSvYHBYa6+deznMkFx2dvA4LCXu0ldoNV/5edQDG5wKXB3zeuxRFnP4CNJktpqy+DQhOoPbLvfe7qkLtBq8Hk3RYC5o+71jBIRRwLvAJYDvcDVwMcnc0YqIuYDbwZeDiwBtgKXAe/PzLXt6rMkSWpswbzeCdXvmztninoiaWfS6qhuZ472eiaIiJcCX6S43+n7FCFuJfDZiDgkM986gbZ2A74HPAW4E7gEWFC29/yIOCUzP9PmtyBJkmr0zeth2aK+hwxh3cyyRX30zfMyN6kbtDS4QURcHxHvr3l9UkQ8o/VuTY+I2BP4DMWDVE/MzKMz80TgCcB1wFsi4ugJNPleitDzc+AJmXl8Zh4FHA3cC3wiIvZv3zuQJEn1emfP4tQVS8ZV97QVS+n1GU1SV2j1X/oBwKNqXp8DnNJim9PpFKAPuDAzvzZSmJmbgbeXL98ynoYiohd4dfny9Mz8XU17Pwb+keIyuje23m1JktRMz+xZrOrfm9XL9xu13url+7Oyf6EPp5W6RKv/0geBPdrQj045rpx/pcGyi4BtwKqIGM8dj/3AfOA+4CcNln+vnL9oop2UJEkTM7+3hzOO7edjLz+MZYv6HrJs2aI+Pvbyw3x4qdRlWv3Xfg3w3Ih4FfCbsmyfiDhqPCtn5vdb3H6rnlTOL69fkJlDEfFLikvXHg9cNUZbu5bzLZnZaICH35fzxRHRl5njG2dTkiRNyvzeHo45eB9W9i9kYHCYgW330zd3Dn3zeuidPcszPVKXaTX4fBD4N4r7ZEY8v5zGY3aL25+0iOijGHgA4JYm1W6hCD77M3bwGbm07VERMS8zB+uWL675eT/gl+Po4/omi5aOta4kSSoue+uZPYv5vT0OWS11uVZHdTsvIn4NHA88FjgZ2Aj8qPWuTbndan7e2qTOveV893G0dx1wK/Bo4FXAmrrlr675eTztSZIkSWqTli9szcyfAT8DiIiTgR9m5qtHXalNIuJ8intrJuKkzLys3X3JzIyI9wEfBf4+Iu4DLqQYPOFNFPcTDVP8zneMs81ljcrLM0FPbEe/JUmSpG7Q7jv6/ozizMd0WQwcNMF15pfze+rKGt1zM3Lfzt3jbPtjwOOA1/PQy/8S+GvgDRSj4N013s5KkiRJal1bg09mfrad7Y1je4e2sO5ARGyhuM/nMcDVDao9ppzfOM42Ezg9Ij5DMXrbvsDtwFfL9v+GYiS86yfbb0mSJEkTNyVjOEbEIcD/Bp5F8cc/wG+B7wP/nJljDRQwXa4EjgKeTF3wiYg5wMEUQ1r/aiKNZuYVwBV17R1FMZjDjzJzeNI9liRJkjRhbR/HMSLeQHHPzynAEyhu5N+9/Pk1wM/KOjuDi8r5iQ2WvRCYC1ySmdvasK3Xl/NPtqEtSZIkSRPQ1uATEc8FPgIMlfPDgEdQPOT0UOBDFA/4/HBErGzntifpUxT39rwoIl4yUhgRC4EPlC8/VL9SRFxTTvvWlS+MiP3qynoi4m8owtX3MvPL7X4TkiRJkkbX7kvd3kwxctnzMvPHdcuuAt4WEV+juOTtLcB327z9CcnMOyPi1cCXgK9ExFqKB42uoghrH87MtQ1WHRlQYU5d+ROB/4qIX1A80DWAp1MMcf0LGp9ZkiRJkjTF2h18ngasaxB6HpCZPykDxhFt3vakZOZXy/tv/hpYDvRS3O/z8UkM1rAR+CxwJHAMxbDV11I86PXjmTnUto5Lkmac4e07GNq+g4HBYbYMDrFgXi9983roLR+yKUmaOu0OPvOB342j3u94cFjpjsvMH1EElfHWjyblN1MM6S1J0kNsHRrmkg2bOXvd9azf9OATFJYt6uPUFUtY1b8383unZMwhSRLtH9zgZuDpEdH0yF0ue3pZV5Kkyts6NMxZF2/g9POueEjoAVi/aYDTz7uCsy6+hq1DDvopSVOl3cHnQmB/4DMRsUf9wojoA/4F2A+4oM3bliRppzO8fQeXbNjMuZfeNGq9cy+9ke9uuJ3h7TumqWeS1F3afU79vcBLgFdQjJT2LeCGctn+wAuAPooHeL63zduWJGmnM7R9B2vWju+51WvWbWRl/0Lv95GkKdDW4FOOknYUsAY4DnhZg2oXAadm5l3t3LYkSTujgcFhrr51YOyKFJe9DQwOe6+PJE2Bth9ZM/O3wPERsRh4JrCoXLQJ+GFm/qbd25QkaWe1ZXBiA3oObLuffRbMnaLeSFL3amvwiYjLgY2Z+bIy4BhyJEldbcG83gnV75tb/4g4SVI7tPsi4oOA+9vcpiRJM1bfvB6WLeobV91li/rom+dlbpI0FdodfH4N7NXmNiVJmrF6Z8/i1BVLxlX3tBVL6XVgA0maEu0+un4aWBERT2hzu5IkzUg9s2exqn9vVi/fb9R6q5fv74hukjSF2j2q28ciYhmwLiLeB3wDuCkzJ3ZnpyRJFTK/t4czju3niMV7sWbdxoc8xHTZoj5OW7GUlf0LHc1NkqZQuwc32D7yI/DBciIiGlXPzPQIL0nqCvN7ezjm4H1Y2b+QgcFhBrbdT9/cOfTN66F39izP9EjSFGt38LgZyDa3KUlSJfSUAWd+b49DVkvSNGv3pW4HtLM9SZIkSWoHz6tLkiRJqrwpv8cmIh5R/viHzPQyOEmSJEnTbkrO+ETEH0fEf0bEPcAd5XR3WfaiqdimJEmSJDXT1uAThc8A5wOrgPnAlnKaX5Z9LSLOiSZDvUmSJElSu7X7jM8bgJOBW4HXAntk5p6ZuSewADitXPbKsq4kSZIkTbl2B5/XAFuBZ2Xm2Zn5wBPaMvPuzPwk8CxgsKwrSZIkSVOu3cFnMfDdzPxNswrlsu+WdSVJkiRpyrU7+PwOGBpHvfspBjyQJEmSpCnX7uBzPvCcmiGsHyYi9gSeA1zQ5m1LkiRJUkPtDj5/DVwP/FdEPKd+YUQ8G/gOsBE4o83bliRJkqSG2v0A0wspLnU7HPhORNwJ3Fgu2w/Yq/z5UuDCuhGtMzNXtrk/kiRJktT24HN0zc9BEXT2alDv6Q3Kss19kSRJkiSg/cHHkdokSZIk7XTaGnwy88axazUWEXtGxH6ZeVM7+yRJkiRJ7R7coBUfohgYQZIkSZLaamcKPlDcFyRJkiRJbbWzBR9JkiRJajuDjyRJkqTKM/hIkiRJqjyDjyRJkqTKM/hIkiRJqjyDjyRJkqTKM/hIkiRJqjyDjyRJkqTK25mCzzXA9zvdCUmSJEnV05HgExGHRMQXassy8/2Z+exO9EeSJElStfVM58YiYjnwDuDYsuhPp3P7kma24e07GNq+g4HBYbYMDrFgXi9983ronT2Lntk70wlsSZK0s2k5+ETErsAbgOcDC4Hbgf8A/jEzt5Z1ngq8F3g2EMAg8IlWty2pe2wdGuaSDZs5e931rN808ED5skV9nLpiCav692Z+77R+lyNJkmaQlv5KKEPPj4A/ogg0AAcBzwSOj4ijKM7wvBOYDWwD1gDvz8zNrWxbUvfYOjTMWRdv4NxLb3rYsvWbBjj9vCtYvXx/zjj2CYYfSZLUUKt/IbwFOITiLM+HgfXA7sAxwGrgwvJngE8CZ2bmbS1uU1IXGd6+g0s2bG4Yemqde+mNHLF4T445eB8ve5MkSQ/TavA5geKytSMzc2NN+Rci4jfAu4AE/jQzv9zitiR1oaHtO1iz9vpx1V2zbiMr+xcafCRJ0sO0+tfB44Cf1IWeEZ8u55cbeiRN1sDgMFffOjB2RYrL3gYGh6e4R5IkaSZqNfjsBtzcaEFmjpRf2+I2JHWxLYNDE6o/sO3+KeqJJEmaydpxPUiOsXxif7VIUo0F83onVL9v7pwp6okkSZrJ2jH80W4Rsd9klmfm6HcrS+p6ffN6WLao7yFDWDezbFEfffMc1U2SJD1cO874vBT4TZMpR1k+vruVJXW13tmzOHXFknHVPW3FUnod2ECSJDXQ6lejNzH2pW6SNGk9s2exqn9vVi/fb9QhrVcv398R3SRJUlMtBZ/MPKBN/ZCkpub39nDGsf0csXgv1qzb+JDL3pYt6uO0FUtZ2b/Qh5dKkqSmWvorobx3557MvLNN/ZlWEbEr8BLgaeV0KNAL/E1mntlCu8cDbwUOK4suB/4+My9qpb9SN5vf28MxB+/Dyv6FDAwOM7DtfvrmzqFvXg+9s2d5pkeSJI2q1a9HfwOcA/x5613piAOBz7WzwYh4I/ARYBi4BLgPeB7wzYh4fWZ+vJ3bk7pJTxlw5vf2sM+CuZ3ujiRJmkFa/Yo0ymmmupviQaunAYcD72qlsYg4CPggRdg5KjOPycwTKM4k/R74SEQ8rpVtSJIkSZq4rr42JDM3ZuYpmXl2Zl4OtPrkwzcAs4E1mfmTmu38CngPxRm2N7S4DUmSJEkT1NXBZwocV86/0mDZSNnx09QXSZIkSSWDT5tExB7AyINaf1G/PDNvBu4A9o+IvmnsmiRJktT12jH26wsi4r8msV5m5so2bH9nMRJ67srMe5vUuQV4JLA/8D9jNRgR65ssWjrx7kmSJEndqx3BZ29gn0msV7UHn+5WzreOUmckEO0+xX2RJEmSVKMdwedHFCOjTbuIOB/on+BqJ2XmZVPRn3bLzGWNysszQU+c5u5IkiRJM1Y7gs91mfnZNrQzGYuBgya4zvyp6Ahwzzja37Wc3z1FfZAkSZLUQDuCT8dk5qGd7kONm8r5IyJi1yb3+TymnN84TX2SJEmShKO6tU1m/oEHw89h9csj4rEUAxvcmJkD09g1SZIkqesZfNrronJ+YoNlI2XfmKa+SJMyvH0HW4eGuW3LNq69bYDbtmxj69Aww9t3dLprkiRJk9bqpW7rgGvGqhQRc4E9gDsyc7jFbXZcRIy855WZ+duaRR8FXgOcFhFfyMxLy/oHAu8Ahss60k5p69Awl2zYzNnrrmf9pgdPTC5b1MepK5awqn9v5vfO6CtkJUlSl2r1jM+LgLUR8fhGCyPiwIj4NjAA/Ba4OyK+EBGPbnG7bRMR50fEpRFxKfC6sviUkbJy5Lh6B5XTnNrCzLwWeBuwC/CDiLg4Ii4ArgT2At6cmddN1XuRWrF1aJizLt7A6edd8ZDQA7B+0wCnn3cFZ118DVuHZvx3F5IkqQu1GnxeCfwEeGH9gojYB/gBsIrizFJQBIKXAd+LiHktbrtdDgOOKKfHlmX71pQ97H6d0WTmR4A/pvi9PAtYCfwMOD4zP9amPkttNbx9B5ds2My5l940ar1zL72R72643cveJEnSjNNq8FkB7ADObbDsncBC4C6KM0O7A0+mCAEHAq9tcdttkZkHZGaMMh3QYJ2RZTc0afMbmXlUZu5eTkdl5jen+r1IkzW0fQdr1l4/rrpr1m1kyOAjSZJmmFaDzyHA/2Tm7bWFETELeDmQwBllELg3M68AXkJxr8sJLW5bUpsMDA5z9a3jG2xw/aYBBga93E2SJM0srQafhcC1DcoPoRjMYBj4Qu2CcjCAnwL9LW5bUptsGRyaUP2BbfdPUU8kSZKmRqvBZ1fqbvAvHV7Or2ryzJpbgL4Wty2pTRbM651Q/b65jf7ZS5Ik7bxaDT6/A57QoPyZFJe5XdZkvbnAlha3LalN+ub1sGzR+L6LWLaoj755DmktSZJmllaDz0+B/og4fqQgIh5FcR8PwH82We+JwKYWty2pTXpnz+LUFUvGVfe0FUvpne2zjyVJ0szS6l8vH6cYpvrLEfG5iPgQ8N8UI7jdDFxUv0JELAEeT/FsG0k7gZ7Zs1jVvzerl+83ar3Vy/dnZf9Cegw+kiRphmnpepXM/F5EnAn8X2A1xeVtAQwCf5aZjYZ+Oq2cf7uVbUtqr/m9PZxxbD9HLN6LNes2PuQhpssW9XHaiqWs7F/I/F4vc5MkSTNPy3/BZOa7I+KbFJe3PYriTM+/Z+ZvmqxyH/BR4FutbltSe83v7eGYg/dhZf9CBgaHGdh2P31z59A3r4fe2bM80yNJkmastnx1m5mXA5ePs+4727FNSVOjpww483t72GfB3E53R5IkqS38+laSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFVeVwefiNg1Il4ZER+LiJ9GxH0RkRFx5iTbe2RE/HlEfDIiroiI4bK9k9vbc0mSJEkT0dPpDnTYgcDn2tjeM4FPtbE9SZIkSW3Q1Wd8gLuBTwOnAYcD72qxvc3APwOvBv4I+JcW25MkSZLUBl19xiczNwKnjLyOiOe12N5PgJ/UtLejlfYkSZIktUe3n/GRJEmS1AUMPpIkSZIqr6svddvZRcT6JouWTmtHJEmSpBnOMz6SJEmSKm9Gn/GJiPOB/gmudlJmXjYV/Wm3zFzWqLw8E/TEae6OJEmSNGPN6OADLAYOmuA686eiI5IkSZJ2XjM6+GTmoZ3ugyRJkqSdn/f4SJIkSao8g48kSZKkyjP4TEJEXFNO+3a6L5IkSZLGNqPv8WmHcmS4R5cvF5XzUyLiBeXPt2bmi+tWGxlQYU6D9i6tebm4nL8zIk4rf748M1/XYrclSZIkTUDXBx/gMGD/urJ9ywngxgm2d0SDsiXlBLBtgu1JkiRJalHXB5/MPGAS68RklkmSJEnqDO/xkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5Bh9JkiRJlWfwkSRJklR5XR18ImLXiHhlRHwsIn4aEfdFREbEmZNs7/CIODMifhwRf4iIoYi4OSLOjYhD2tx9SZIkSePU0+kOdNiBwOfa0VBE9AA/K1/eCfwYuBc4DHgF8LKIeEVmfqUd25MkSZI0fl19xge4G/g0cBpwOPCuFtv7b+AEYGFmHpuZLwMeD7wH6AU+ExGPbHEbkiRJkiaoq8/4ZOZG4JSR1xHxvBbaGgae1qB8R0S8EzgROAg4DvjsZLcjSZIkaeK6/YzPtMjMBK4qXy7qZF8kSZKkbmTwmT5LyvltHe2FJEmS1IW6+lK36RIRz6S4h2gI+NYE1lvfZNHSdvRLkiRJ6hae8ZliEdEHfKZ8+ZHMvLWT/ZEkSZK60Yw+4xMR5wP9E1ztpMy8bCr6Uy8iZgP/TjFs9mVMcNS4zFzWpN31wBNb7qAkSZLUJWZ08AEWU4yUNhHzp6IjTXwCeCFwLXBcZg5N47YlSZIklWZ08MnMQzvdh2Yi4n3AXwA3A8/NzDs63CVJkiSpa3mPzxSIiLcDfwncThF6bu5wlyRJkqSuZvBps4j4C+D9wB+A52fmtZ3tkSRJkiSDzyRExDXltG9d+YnAGuAe4NjMvKIT/ZMkSZL0UDP6Hp92KEeGe3T5clE5PyUiXlD+fGtmvrhutZEBFebUtLOQYgS3WcBvgFMj4tQGm7wgMy9oR98lSZIkjU/XBx/gMGD/urJ9ywngxnG2Mx/oLX/+o3Jq5AbggvF3T5IkSVKruj74ZOYBk1gnGpTdADysXJIkSVLneY+PJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMoz+EiSJEmqPIOPJEmSpMrr6XQHJEmSJHXO8PYdDG3fwcDgMFsGh1gwr5e+eT30zp5Fz+zqnCcx+EiSJEldauvQMJds2MzZ665n/aaBB8qXLerj1BVLWNW/N/N7qxEZqhPhJEmSJI3b1qFhzrp4A6efd8VDQg/A+k0DnH7eFZx18TVsHRruUA/by+AjSZIkdZnh7Tu4ZMNmzr30plHrnXvpjXx3w+0Mb98xTT2bOgYfSZIkqcsMbd/BmrXXj6vumnUbGTL4SJIkSZppBgaHufrWgbErUlz2NjA48y93M/hIkiRJXWbL4NCE6g9su3+KejJ9DD6SJElSl1kwr3dC9fvmzpminkwfg48kSZLUZfrm9bBsUd+46i5b1EffvJk/pLXBR5IkSeoyvbNnceqKJeOqe9qKpfRW4EGmM/8dSJIkSZqQntmzWNW/N6uX7zdqvdXL92dl/0J6KhB8Zv45K0mSJEkTNr+3hzOO7eeIxXuxZt3GhzzEdNmiPk5bsZSV/QuZ31uNyFCNdyFJkiRpwub39nDMwfuwsn8hA4PDDGy7n765c+ib10Pv7FmVONMzwuAjSZIkdbGeMuDM7+1hnwVzO92dKVOdCCdJkiRJTRh8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFWewUeSJElS5Rl8JEmSJFVeZGan+6AJioiBXXbZZfelS5d2uiuSJElSW2zcuJH77rvv7szsm4r2DT4zUETcBswHbu50X6bBSLrb2NFeqMrcxzQd3M801dzHNNWmYx97LLA1M/eZisYNPtqpRcR6gMxc1um+qJrcxzQd3M801dzHNNWqsI95j48kSZKkyjP4SJIkSao8g48kSZKkyjP4SJIkSao8g48kSZKkynNUN0mSJEmV5xkfSZIkSZVn8JEkSZJUeQYfSZIkSZVn8JEkSZJUeQYfSZIkSZVn8JEkSZJUeQYfSZIkSZVn8JEkSZJUeQYfjUtEHB4RfxURX4uIWyIiI6Lp028j4syROk2m942y7pERcXFE3BkR90TEZRFx0hj9e0xE/GtEbIqIbRHxq4j4m4iYO8o68yLi3WXdbeW6n4mIfcf3W1E7RcT8iDghIj4dEdeWn8m9EXFlRLwrInYbZd2Ty/3knnK/uTginjHG9tzPusxk9jGPZZqoiHhz+X/lryNiS0TcFxE3RsTnIuKPRlnP45jGbaL7mceyUmY6OY05ARcAWT+NUv/Mss4PgXMaTC9rst5LgWFgB7AW+ApwV9nWB5us8zjgd2Wd/wG+CGys2f4uDdaZC/ykrLOpXOen5evbgSWd/p132wScUrNvXQ18CfgWMFCWbQAWNljvH8rlW8v99FvA/eV+dIL7mVMr+5jHMqdJ7Gd3AIPl5/C1crq2/EyGgBc2WMfjmNOU7mcey8rtdPqDc5oZE/CXwLuB44F9gG2ML/icPIFt7AlsKdd7SU353sCvy/KjG6z3w3LZR2vKesqDQAJnNljn78plPwZ2qyl/c1m+ttO/826bgFcBZwP9deWPBi4vP5fP1y1bVZbfARxYU/504L7y4LyH+5lTC/uYxzKnie5nRwJzG5S/rvxMbgN6aso9jjlNx37msSwNPk6TnJia4PP2cp0LGix7cbnsG3XlTyvLN1P3DUL5D3MIuLPuH38v8IdyvcMabOvKctnhnf49Oz3wmTy9/Ey2Ab015ReX5W9ssM5Hy2VvcT9zamEf81jm1M797LryMzmkpszjmNN07GceyzK9x0c7lePK+VcaLLuI4g+SVXXXh46s843MvK92hczcDPwAeATwzJpFRwILgI2Z+YsG2xrZ/vET676m0JXlfBdgLyiuBQaeU5Y32meafY7uZ2rkYftYC9zH1Mz95XwIPI5pyjxkP2tB5fYzg4+m2nMi4h8iYk1E/HVEHD5K3SeV88vrF2TmEPBLimtAHz+ederKD2lxHXXWknJ+P8U3RQAHUfyR+rvMvKXBOs0+R/czNdJoH6vlsUwtiYhXUhy3fl1O4HFMbdZkP6vV1ceynlYbkMbwyrrXfxsRX6U41XrPSGFE9FGkfYBGB/+R8qcA+wNXlWX7jWMdynVGTGYdddYbyvm3ar5BGvVzzMx7I+IPwCMiYvfMvNv9TKNotI/V8limCYmItwHLgF2B/vLnTcDLM3N7Wc3jmFoyzv2sVlcfyzzjo6lyHfBWin+AuwGPBV4B/JZihJB/q6tfO4zs1iZt3lvOd2+w3lSvow6JiGOBP6f4Jv6dNYvG+hzh4Z+l+5keZpR9DDyWafKeTzGgxokU+8+NFH+M/rymjscxtWo8+xl4LAMMPpoimXluZn4oM6/OzHsz85bM/DzwVOD3wAkRsbzD3dROLiKeAJwLBPC2zLxyjFWkCRlrH/NYpsnKzFWZGRT3MxxFcdnRuoh4R2d7pioZ737msaxg8NG0ysxbgX8tX76gZtE9NT/Pb7L6ruX87gbrTfU6mmblA8u+RXEw/3BmfrSuylifIzz8s3Q/0wPGsY815bFM45WZf8jMHwDHAj+nuLToqeVij2NqizH2s9HW66pjmcFHnTBys92jRwoyc4BirHiAxzRZb6T8xpqym6ZpHU2jiNgT+E+K63n/leL0fL1RP8eI2BXYA7grM+8G9zM9aJz72Fg8lmncMvN+iocyBg+OTuVxTG3VZD8bS9ccyww+6oRHlPN768pHLjF5cv0KETEHOJhi6MRfjWeduvKrasoms46mSUTsBvwH8ESKh539RZYD+de5luLhfo8qv7mv1+xzdD/rchPYx8bisUwTdUc5f1Q59zimqVC/n42la45lBh9Nq4gIiodewcOHLbyonJ/YYNUXUgyZeElmbmuwzvERsUvdtvYGnkXx1Osf1Sz6EcW3GEsj4tAG2xrZ/jeavxNNhfIzvJDiAWjfpvmoNGTmIPBf5cuXNajS7HN0P+tiE9nHxmjHY5kmY0U53wgexzRlHrKfjabrjmWtPgHVqTsnioSfTZY9CvjfwO515bsBayievnsrML9u+Z7ljp/AS2rKF1Kchk3g6Abb+2G57B9qynqAr5blZzZY5+/KZT8Cdq0pf3NZvrbTv+Num4DZFN++J/D9+v2jyTqryvp3AAfWlD+93EfvAvZwP3OazD7mscxpEvvYkRT3ScyqK58DvB7YTjFy1WNrlnkcc5rS/cxjWc12Ov3hOc2MieJJvJfWTDvKHbG27Liy7gHlsrspvsn6d4pr6e8oy+8CjmyynZeW/2B3lOt+uayfwIearHNgTdtXAV+g+JZj5B/QLg3WmVv2OSnGu/9izevbgSWd/p1320TxHJUsp68B5zSZHlm33j+U69wLXABcTDEs8TBwgvuZ02T3MY9lTpPYx04uf/e/oxg4498pzixuKssHgf/VYD2PY05Ttp95LKvZTqc/PKeZMdX8IxttOrmsuzvwPmAtxUOntpUH818CHwT2HWNbR1Jcf39Xud5/A68aY53HUtygfCvF9dK/Bt4NzB1lnXllnevKdUZGNnlMp3/f3TgBZ45jH0vggCb758/K/eWucv95hvuZUyv7mMcyp0nsY4uB91B8470JGKIYseqXwD8CjxtlXY9jTlOyn3kse3CKckOSJEmSVFkObiBJkiSp8gw+kiRJkirP4CNJkiSp8gw+kiRJkirP4CNJkiSp8gw+kiRJkirP4CNJkiSp8gw+kiRJkirP4CNJkiSp8gw+kiRJkirP4CNJkiSp8gw+ktQBEXFARGRErO10XzotIm6IiGxDO0eXv9Nz2tCt8W6z4efYib5IkkZn8JEkARAR55R/rB/d6b6oe7UrCEtSvZ5Od0CS1PVWAnM63Yk2uwzoB7Z0uiOSpILBR5LUUZm5sdN9aLfM3Apc0+l+SJIe5KVuktRhEdEXER+NiJsjYltEbIiIN0XErLp6h0bEByLi5xHxu4i4LyKuj4h/johFo7T/2Ij4x4j4VUQMRsSdEfGziPi/EdFX1kngVeUq3ysveRuZDqhr7wURcVFdHz4cEXs12PYDl89FxPMj4nsR8YeybI+yTsNLmyLiuIj4TPn7GIiIeyPiyog4IyJ2mejveTwi4lkR8fGIuCoi7ip/X9dExPtG+jvOdpre4xMRu5bt3VB+3tdFxDsjYk6j30VtWxGxZ0R8IiJuLX/3v4yIVzfYxgP3HpXb+3C5fw1GxOURcXxN3ZdFxE/L3+/mcl+Z1+R9zY+I/z8ifhER95TTpRHxqib1s3xPsyPiL8t98L6yL++v/RxH3iewf826I9MNNfV2K/twZURsKfuwMSK+HBHPH/PDkdS1POMjSZ21C/BfwNJy3ktx6deHgScBJ9fU/SvgpcBVwA/LskOB1wInRMRTMnNTbeMR8Szg68AewA3AN4B5wBOAM4ELgSuAzwLPLPvxbeC2mmbuqWnvfcBfAkPAfwO3lv18E/DHEXFkZm5u8D7/P+AU4GfAf5TbGes+jk+Xff1l+Z4XAE8D3gOsjIjnZeb2MdqYqL+neD9XAd8F5gJPpnjPL4yI5Zl5zyjrj6r8Q/8SYDlwJ/BNin3gr8rtjGYP4CfAbsAPgEcCRwGfjohZmfmpBuv0lu9jMfD9mnXOj4gXAH8EfABYR/G5HwW8HtgLeEVd3xcC3wEOodg/1gEBPAM4p9z/Xt+k758HjgXWAtcCzwLeDuwLrC7r3EaxH54I7Fr+POKOsg+zKX5/R5Rla4FtwGPK9u8t34ckPVxmOjk5OTlN8wQcQPGHfwJXAo+sWbYU+G257ISa8mcDe9e1Mwt4V1n3M3XL9gRuL5e9FZhVt/zpwMKa1+eUdY9u0ueXlcv/B3hcTXkAf1Mu+0LdOiNtJvAnTdq9ofjv6GHlLwLm1ZXtThHeEjipbtnRZfk5LXwuxwAL6sp2Ac4u235Xk89x7Xj6Avx1Wf5TYI+6dm4a+V01aSuB84BdapadUJbfOMr+9V1g15plJ5flv6YIX0+pWbYI2FwuX1LX5kVl+T/U9WFvihCcwAvq1hnpw9XAPjXli4G7ymVLx7M/1PwbSIp7qObWLesDDp/Kf7dOTk4ze/JSN0nqvLdm5h0jL7K45+Vvy5f/p6b8e1l3NiUzd2TmuymC0h/XtXsK8CjgW5n5wczcUbfuTzLz9gn08x3l/OWZeV1NO0lx9ugK4MSIeGSDdS/KzC9OYFtk5oWZOVhXdjfF2SUoglFbZeZ/ZOaWurL7gDcCw23Y5mnl/C2Z+YeabdwAvHuMdQeA/1P2Z2S9CyjOiO1Xf0liaQfw2sy8t6bscxRnSx4H/FNm/qymvU3Av5cvjxopj4hDKc6o/Dfw5ro+bAZeU758bZO+n56ZD5xFzMzfAOeWL5/VZJ1GHlXOf5SZ22oXZOZAZv58Am1J6jJe6iZJnXVnZn6nQfl5wCeAZ5SXMe0AKO+j+WPgYIpLn2aX9ecAe0XEnpl5Z1m2qpyf3Wony8ucngT8OjN/Wb88MzMifkRx6d3hPPxyo69PcrsHUvzB/TiKy59mUZxhAjhwMm2OY5v7AsdTXA7Yx4P3ww61ss2I2J/i0q7bMvOHDap8EfiXUZr4eWb+vkH5ryj2h0dTnC2pdUNm/qq2IDN3RMSNFJe9/WeD9q4v54+uKXteOb+gPkCXbf4iIu6huBSx3v3A95r0u347Y7mCIsz9WURcDXytye9Ekh7G4CNJnXVjo8LM3BIRf6AIN48Afh8RLwc+SXGPRzO7U1y+BPDYct6OUdMOKOcHNhqIoE6jMz43TWRjERHABynO7kSTartPpM1xbvfNwPuYmuG1R/7Av7nRwsy8u+Yzb+SWJuV3l/NGAz78tsk694yyfGRZbXsHlPP3RMR7mrQJxT1R9W7LxvdijdbvhjLzVxHxduC9FP8W1kTELyku5zsnM68ab1uSuo/BR5JmgPJswTnlyzdS3G/x25FLwSLixxT37DQLCa0aOetxG2PfPN4ozG1rUDaaPwHeTBES3kRxU//vMvP+iOgF7qPN7zUilgMfonj2zhsobpy/beSyrojYxMTOTrTbw860tGGd8bY58vn/kIkH6cn0u6nM/FBEfIni/qbnUlwq9ybgjRHxpsz8aDu3J6k6DD6S1Fn7NSqMYpjpPYBB4A/A/6IYoeuDTf6wW9Kg7GaKy7WWUgxI0IqRsw13ZObJLbY1Hi8u56/NzIvqljV6r+3c5jsys3ZEMcrhnfdpsf1by/ljGy2MiN1pfran00Y+/wsy80Md7QmQmTcDHwM+FhE9wJ8C/wp8ICI+l5l3dbSDknZKDm4gSZ21V0SsbFD+p+X8J+VlQo8oXz/scqeIOIpiZK16l5Tz1zRY1shQOX/Yl2KZeQvFAzmfGBGPH2d7rWj6filC4HRv82W0eIYpM2+kuLRsn4h4RpNt7KxG7kN78ai12mMIoAw0Y8rM4cw8l2LghV6m6N4vSTOfwUeSOu+DUfPwz4hYTDFENcA/lfORG8FXR8SuNXX3BdY0afdTFKN3HRMRbyzvm3lARCwvBy0YMfIMoIOatPe3FP9vfLUc5eshImKviPiLJutO1Mj7fU1tv8vnEr2tTdtots0/j4gH7vGJiCcC72/TNkY+qw9FxIKabezPg5/5Ticzf0oRfo6MiH8qz0g+REQ8qXw2UKua7ocR8eyIWBUPf7jvYqCfYqjrZvdCSepyBh9J6qxLKe6BuC4ivhoRX6cYnnhf4NzM/FpZ7+vAeuApZd2vRMQ3Kf5Yvwv4cX3D5ehuL6O4ifwjwMaI+GJEfD0ifk1x38yimlVGno/zwYi4ICI+VU57le19HjiLYgSxn0fEzyPiSxHx5Yi4nOL5L3/fpt/LP1I8jPJ1wC8j4ryI+D7FQzObBb1W/SvFPUzHA9eWv6vvUIwk9gOaDEQxQX9P8Zkvp/g8vlx+5uspnud0E8UoaDuj1cAvKD6TGyPiexHx7xHxzYi4ieL31I7gMzIC4HfLz/1T5YNzoRhZ8DvAbRHxHxFxbkR8G9hAcZngx7PuIb6SNMLgI0mddR/wHIon2y8Hnk9xb85bKR40CUBmDlHcxP0JioECXkjxDffHKG7wbvjHcmaupfhjcQ3FpVonAEdS3MD/LmpuVC+fgbKa4mGTzwP+vJx2r6nzDmAF8FWKe15OoHio5Oyyb/XPEpqUcgjmp1CEsUeW7e4GnJqZU3LGpxwW+akUn0Vvuc19gXcCL2/TNu6j+Lw+QBHs/hhYRjGowp9QXLK4Uw7PXD7z6RnA6RT7yGHAicAhFENgv41iJL5W/SPwdxSjy72UYh8cufTzm+WyX1Hs1y+j+P39sKz7hjZsX1JFRfHcOUmS1EnlqHI/oXjg7DGd7o8kVY1nfCRJmkYRcViDe1SW8OCDZs+d/l5JUvV5xkeSpGkUEdcACyiGGP89xZDmh1M8yPPrwAnpf86S1HYGH0lSJUXEKcAzx1n9fZl5zVT2Z0RE/G+KIbkPAvakuGfraoozPWsyc3g6+iFJ3cbgI0mqpIg4B3jVOKs/uxwIQpJUUQYfSZIkSZXn4AaSJEmSKs/gI0mSJKnyDD6SJEmSKs/gI0mSJKnyDD6SJEmSKs/gI0mSJKnyDD6SJEmSKs/gI0mSJKnyDD6SJEmSKs/gI0mSJKnyDD6SJEmSKs/gI0mSJKnyDD6SJEmSKu//AbwZZtTXTA3gAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 900x600 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "d = [   \n",
    "        [ 1054, 34777, 1011050, -1.24 ],\n",
    "        [ 649, 28167, 1242088, -0.888940 ],\n",
    "        [ 612, 25154, 2057903, -0.920462 ],\n",
    "        [ 1627, 24376, 1708005, -0.872526 ],\n",
    "        [ 1657, 19358, 1952344, -0.981056 ],\n",
    "        [ 1253 , 18097, 1414755, -1.016385 ],\n",
    "    # [ 609, 15732, 0 ]\n",
    "        [1236, 14102, 1343181, -0.635272],\n",
    " ]\n",
    "\n",
    "d = pd.DataFrame(d, columns=['sample_id', 'bacterial_alignments', 'total_STR_reads', 'STR_profile'])\n",
    "\n",
    "plt.figure(dpi=150)\n",
    "sns.scatterplot(data=d, x='bacterial_alignments', y='STR_profile')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>HRGM_name</th>\n",
       "      <th>count</th>\n",
       "      <th>perc</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>member_count</th>\n",
       "      <th>continents</th>\n",
       "      <th>country</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>901</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_0235</td>\n",
       "      <td>10652</td>\n",
       "      <td>30.629439</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroides cellulosilyticus</td>\n",
       "      <td>483</td>\n",
       "      <td>Oceania;NA;North America;Asia;Europe</td>\n",
       "      <td>Mongolia;Austria;Denmark;South Korea;Netherlan...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>936</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_0392</td>\n",
       "      <td>5084</td>\n",
       "      <td>14.618857</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Fusobacteriota</td>\n",
       "      <td>Fusobacteriia</td>\n",
       "      <td>Fusobacteriales</td>\n",
       "      <td>Fusobacteriaceae</td>\n",
       "      <td>Fusobacterium</td>\n",
       "      <td>Fusobacterium animalis</td>\n",
       "      <td>36</td>\n",
       "      <td>Europe;NA;Asia;North America</td>\n",
       "      <td>China;Spain;United States;NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1021</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_1222</td>\n",
       "      <td>3367</td>\n",
       "      <td>9.681686</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Xanthomonadales</td>\n",
       "      <td>Xanthomonadaceae</td>\n",
       "      <td>Stenotrophomonas</td>\n",
       "      <td>Stenotrophomonas maltophilia_F</td>\n",
       "      <td>2</td>\n",
       "      <td>North America</td>\n",
       "      <td>United States</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1474</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_5206</td>\n",
       "      <td>1492</td>\n",
       "      <td>4.290192</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Mycoplasmatales</td>\n",
       "      <td>Mycoplasmoidaceae</td>\n",
       "      <td>Ureaplasma</td>\n",
       "      <td>Ureaplasma parvum</td>\n",
       "      <td>3</td>\n",
       "      <td>North America</td>\n",
       "      <td>United States</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1178</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_1823</td>\n",
       "      <td>1190</td>\n",
       "      <td>3.421802</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Campylobacterota</td>\n",
       "      <td>Campylobacteria</td>\n",
       "      <td>Campylobacterales</td>\n",
       "      <td>Campylobacteraceae</td>\n",
       "      <td>Campylobacter_B</td>\n",
       "      <td>Campylobacter_B ureolyticus</td>\n",
       "      <td>16</td>\n",
       "      <td>Europe;North America</td>\n",
       "      <td>Canada;United States;Ireland</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>881</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_0169</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002875</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Tissierellales</td>\n",
       "      <td>Helcococcaceae</td>\n",
       "      <td>Parvimonas</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Europe</td>\n",
       "      <td>Austria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>882</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_0182</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002875</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Actinobacteriota</td>\n",
       "      <td>Coriobacteriia</td>\n",
       "      <td>Coriobacteriales</td>\n",
       "      <td>Coriobacteriaceae</td>\n",
       "      <td>Collinsella</td>\n",
       "      <td>Collinsella ihuae</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>884</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_0187</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002875</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Tissierellales</td>\n",
       "      <td>Helcococcaceae</td>\n",
       "      <td>Anaerococcus</td>\n",
       "      <td>Anaerococcus senegalensis</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>888</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_0192</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002875</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Tissierellales</td>\n",
       "      <td>Helcococcaceae</td>\n",
       "      <td>Anaerococcus</td>\n",
       "      <td>Anaerococcus obesiensis</td>\n",
       "      <td>29</td>\n",
       "      <td>Europe;Asia;North America</td>\n",
       "      <td>Canada;Spain;Bangladesh;France;United States;U...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891</th>\n",
       "      <td>1054</td>\n",
       "      <td>HRGM_Genome_0203</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002875</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Rikenellaceae</td>\n",
       "      <td>Alistipes</td>\n",
       "      <td>Alistipes putredinis</td>\n",
       "      <td>5377</td>\n",
       "      <td>South America;Oceania;NA;Asia;North America;Eu...</td>\n",
       "      <td>Mongolia;Austria;Denmark;South Korea;Netherlan...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>637 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     sample_id         HRGM_name  count       perc    domain  \\\n",
       "901       1054  HRGM_Genome_0235  10652  30.629439  Bacteria   \n",
       "936       1054  HRGM_Genome_0392   5084  14.618857  Bacteria   \n",
       "1021      1054  HRGM_Genome_1222   3367   9.681686  Bacteria   \n",
       "1474      1054  HRGM_Genome_5206   1492   4.290192  Bacteria   \n",
       "1178      1054  HRGM_Genome_1823   1190   3.421802  Bacteria   \n",
       "...        ...               ...    ...        ...       ...   \n",
       "881       1054  HRGM_Genome_0169      1   0.002875  Bacteria   \n",
       "882       1054  HRGM_Genome_0182      1   0.002875  Bacteria   \n",
       "884       1054  HRGM_Genome_0187      1   0.002875  Bacteria   \n",
       "888       1054  HRGM_Genome_0192      1   0.002875  Bacteria   \n",
       "891       1054  HRGM_Genome_0203      1   0.002875  Bacteria   \n",
       "\n",
       "                phylum                class              order  \\\n",
       "901       Bacteroidota          Bacteroidia      Bacteroidales   \n",
       "936     Fusobacteriota        Fusobacteriia    Fusobacteriales   \n",
       "1021    Proteobacteria  Gammaproteobacteria    Xanthomonadales   \n",
       "1474        Firmicutes              Bacilli    Mycoplasmatales   \n",
       "1178  Campylobacterota      Campylobacteria  Campylobacterales   \n",
       "...                ...                  ...                ...   \n",
       "881       Firmicutes_A           Clostridia     Tissierellales   \n",
       "882   Actinobacteriota       Coriobacteriia   Coriobacteriales   \n",
       "884       Firmicutes_A           Clostridia     Tissierellales   \n",
       "888       Firmicutes_A           Clostridia     Tissierellales   \n",
       "891       Bacteroidota          Bacteroidia      Bacteroidales   \n",
       "\n",
       "                  family             genus                         species  \\\n",
       "901       Bacteroidaceae       Bacteroides    Bacteroides cellulosilyticus   \n",
       "936     Fusobacteriaceae     Fusobacterium          Fusobacterium animalis   \n",
       "1021    Xanthomonadaceae  Stenotrophomonas  Stenotrophomonas maltophilia_F   \n",
       "1474   Mycoplasmoidaceae        Ureaplasma               Ureaplasma parvum   \n",
       "1178  Campylobacteraceae   Campylobacter_B     Campylobacter_B ureolyticus   \n",
       "...                  ...               ...                             ...   \n",
       "881       Helcococcaceae        Parvimonas                                   \n",
       "882    Coriobacteriaceae       Collinsella               Collinsella ihuae   \n",
       "884       Helcococcaceae      Anaerococcus       Anaerococcus senegalensis   \n",
       "888       Helcococcaceae      Anaerococcus         Anaerococcus obesiensis   \n",
       "891        Rikenellaceae         Alistipes            Alistipes putredinis   \n",
       "\n",
       "      member_count                                         continents  \\\n",
       "901            483               Oceania;NA;North America;Asia;Europe   \n",
       "936             36                       Europe;NA;Asia;North America   \n",
       "1021             2                                      North America   \n",
       "1474             3                                      North America   \n",
       "1178            16                               Europe;North America   \n",
       "...            ...                                                ...   \n",
       "881              1                                             Europe   \n",
       "882              1                                                NaN   \n",
       "884              1                                                NaN   \n",
       "888             29                          Europe;Asia;North America   \n",
       "891           5377  South America;Oceania;NA;Asia;North America;Eu...   \n",
       "\n",
       "                                                country  \n",
       "901   Mongolia;Austria;Denmark;South Korea;Netherlan...  \n",
       "936                        China;Spain;United States;NA  \n",
       "1021                                      United States  \n",
       "1474                                      United States  \n",
       "1178                       Canada;United States;Ireland  \n",
       "...                                                 ...  \n",
       "881                                             Austria  \n",
       "882                                                 NaN  \n",
       "884                                                 NaN  \n",
       "888   Canada;Spain;Bangladesh;France;United States;U...  \n",
       "891   Mongolia;Austria;Denmark;South Korea;Netherlan...  \n",
       "\n",
       "[637 rows x 14 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "composition_summary[composition_summary['sample_id']=='1054'].sort_values('perc', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "'Fusobacterium animalis'"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "sc",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
