{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import glob \n",
    "import numpy as np\n",
    "import pandas as pd \n",
    "import scanpy as sc\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Collect Cellranger outputs for both datasets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Kinker et al"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_155434/1346728635.py:1: DtypeWarning: Columns (4,9,10,11,12,13,14,15,16,17,18,19,20) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  kinker_metadata = pd.read_csv('/node200data/18parkky/datasets/data/public/processed_data/Kinker_et_al/Metadata.txt', sep='\\t')\n",
      "/home/18parkky/anaconda3-2021.11/envs/sc/lib/python3.12/site-packages/anndata/_core/aligned_df.py:67: ImplicitModificationWarning: Transforming to str index.\n",
      "  warnings.warn(\"Transforming to str index.\", ImplicitModificationWarning)\n",
      "/home/18parkky/anaconda3-2021.11/envs/sc/lib/python3.12/site-packages/anndata/_core/aligned_df.py:67: ImplicitModificationWarning: Transforming to str index.\n",
      "  warnings.warn(\"Transforming to str index.\", ImplicitModificationWarning)\n"
     ]
    }
   ],
   "source": [
    "kinker_metadata = pd.read_csv('/node200data/18parkky/datasets/data/public/processed_data/Kinker_et_al/Metadata.txt', sep='\\t')\n",
    "adata = sc.read_h5ad('/node200data/18parkky/datasets/data/public/processed_data/Kinker_et_al/CPM_data.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "dict_CB_to_MetaData = { tup.NAME : [tup.Cell_line, tup.Pool_ID, tup.Cancer_type] for tup in kinker_metadata.itertuples() }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "for CB in adata.obs['CB']:\n",
    "    try: dict_CB_to_MetaData[CB]\n",
    "    except KeyError: dict_CB_to_MetaData[CB] = [None, None, None]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.obs['CellLine'] = [ dict_CB_to_MetaData[CB][0] for CB in adata.obs['CB'] ]\n",
    "adata.obs['PoolID'] = [ dict_CB_to_MetaData[CB][1] for CB in adata.obs['CB'] ]\n",
    "adata.obs['CancerType'] = [ dict_CB_to_MetaData[CB][2] for CB in adata.obs['CB'] ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.write('/node200data/18parkky/datasets/data/public/processed_data/Kinker_et_al/CPM_data.metalabeled.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Joanito et al (MSI)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 MSI-H patients from:\tCRC-SG1 cohort\n",
      "2 MSI-H patients from:\tKUL cohort\n"
     ]
    }
   ],
   "source": [
    "joanito_metadata = pd.read_csv('/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/CRC-SG1/synapse-metadata/patient_clinical_information.csv', encoding = \"ISO-8859-1\")\n",
    "joanito_metadata = joanito_metadata[joanito_metadata['dataset'].isin(['CRC-SG1', 'KUL5', 'KUL3'])].copy()\n",
    "joanito_metadata = joanito_metadata[joanito_metadata['MSS_MSI']=='MSI-H'].copy()\n",
    "\n",
    "PATH_SG1_metadata = \"/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/CRC-SG1/EGAD00001008555-metadata/organized_metadata.tsv\"\n",
    "PATH_KUL_metadata = \"/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/KUL/organized_metadata.tsv\"\n",
    "\n",
    "SG1_metadata = pd.read_csv(PATH_SG1_metadata, sep='\\t')\n",
    "KUL_metadata = pd.read_csv(PATH_KUL_metadata, sep='\\t')\n",
    "\n",
    "SG1_metadata = SG1_metadata[SG1_metadata['patient.ID'].isin( joanito_metadata['patient.ID'] )].copy()\n",
    "KUL_metadata = KUL_metadata[KUL_metadata['patient.ID'].isin( joanito_metadata['patient.ID'] )].copy()\n",
    "\n",
    "print(f'{len(set(SG1_metadata['patient.ID']))} MSI-H patients from:\\tCRC-SG1 cohort')\n",
    "print(f'{len(set(KUL_metadata['patient.ID']))} MSI-H patients from:\\tKUL cohort')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "DIR_CRC_SG1_cellranger_out = '/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/CRC-SG1/cellranger'       #\n",
    "DIR_KUL_MSI_cellranger_out = '/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/KUL/MSI_H/cellranger'     # iCMS3 MSI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### CRC-SG1 cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CRC2783\n",
      "CRC2817\n"
     ]
    }
   ],
   "source": [
    "adata = list()\n",
    "\n",
    "for patientID, edf in SG1_metadata.groupby(\"patient.ID\"):\n",
    "    print(patientID, flush=True)\n",
    "    \n",
    "    for tup in edf.itertuples():\n",
    "        PATH_cellranger_h5ad = f'{DIR_CRC_SG1_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_cellranger_scrublet_out = f'{DIR_CRC_SG1_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "        \n",
    "        if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "            if os.path.exists( PATH_cellranger_scrublet_out ) == True:  # Add to list\n",
    "                adata_temp = sc.read_h5ad( PATH_cellranger_scrublet_out )\n",
    "                \n",
    "                adata_temp.obs['SampleID']  = tup.file_ID\n",
    "                adata_temp.obs['PatientID'] = list(tup)[2]\n",
    "                adata_temp.obs['BiopsySite'] = list(tup)[1].split(\"_\")[1]\n",
    "    \n",
    "                adata.append( adata_temp ) \n",
    "                \n",
    "            else:   # Run scrublet, save to disk then add to list\n",
    "                adata_temp = sc.read_10x_h5( PATH_cellranger_h5ad )\n",
    "                adata_temp.obs_names_make_unique()\n",
    "                adata_temp.var_names_make_unique()\n",
    "                sc.pp.scrublet(adata_temp)\n",
    "                adata_temp.write(PATH_cellranger_scrublet_out)\n",
    "                adata.append( adata_temp )\n",
    "        else:   # Report\n",
    "            print(f'{patientID}-{tup.file_ID} has not been processed by CellRanger!')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/18parkky/anaconda3-2021.11/envs/sc/lib/python3.12/site-packages/anndata/_core/anndata.py:1818: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"obs\")\n"
     ]
    }
   ],
   "source": [
    "adata = sc.concat( adata  )\n",
    "adata.obs_names_make_unique()\n",
    "\n",
    "# mitochondrial genes, \"MT-\" for human, \"Mt-\" for mouse\n",
    "adata.var[\"mt\"] = adata.var_names.str.startswith(\"MT-\")\n",
    "# # ribosomal genes\n",
    "# adata.var[\"ribo\"] = adata.var_names.str.startswith((\"RPS\", \"RPL\"))\n",
    "# # hemoglobin genes\n",
    "# adata.var[\"hb\"] = adata.var_names.str.contains(\"^HB[^(P)]\")\n",
    "\n",
    "sc.pp.calculate_qc_metrics(\n",
    "    adata, qc_vars=[\"mt\",], inplace=True, log1p=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "PATH_adata_save = '/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/1_raw_h5ad/Joanito.CRC-SG1.MSI.raw.h5ad'\n",
    "adata.write(PATH_adata_save)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# adata = sc.read_h5ad(PATH_adata_save)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### KUL cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_alias</th>\n",
       "      <th>patient.ID</th>\n",
       "      <th>file_ID</th>\n",
       "      <th>Gender</th>\n",
       "      <th>MSS_MSI</th>\n",
       "      <th>Site</th>\n",
       "      <th>Sidedness</th>\n",
       "      <th>Group_Stage</th>\n",
       "      <th>Stage_TNM</th>\n",
       "      <th>iCMS_transcriptomic</th>\n",
       "      <th>iCMS_inferCNV</th>\n",
       "      <th>KRAS</th>\n",
       "      <th>BRAF</th>\n",
       "      <th>TP53</th>\n",
       "      <th>APC</th>\n",
       "      <th>PIK3CA</th>\n",
       "      <th>LymphNode</th>\n",
       "      <th>Normal</th>\n",
       "      <th>Tumor</th>\n",
       "      <th>CMS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SC035_tumor1</td>\n",
       "      <td>SC035</td>\n",
       "      <td>EXT048</td>\n",
       "      <td>M</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CMS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SC035_tumor2</td>\n",
       "      <td>SC035</td>\n",
       "      <td>EXT049</td>\n",
       "      <td>M</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CMS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SC035_normal</td>\n",
       "      <td>SC035</td>\n",
       "      <td>EXT050</td>\n",
       "      <td>M</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CMS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SC035_lymphnode</td>\n",
       "      <td>SC035</td>\n",
       "      <td>EXT051</td>\n",
       "      <td>M</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CMS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SC044_tumor1</td>\n",
       "      <td>SC044</td>\n",
       "      <td>EXT124</td>\n",
       "      <td>F</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>Caecum</td>\n",
       "      <td>R</td>\n",
       "      <td>III</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CMS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SC044_tumor2</td>\n",
       "      <td>SC044</td>\n",
       "      <td>EXT125</td>\n",
       "      <td>F</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>Caecum</td>\n",
       "      <td>R</td>\n",
       "      <td>III</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CMS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SC044_normal1</td>\n",
       "      <td>SC044</td>\n",
       "      <td>EXT126</td>\n",
       "      <td>F</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>Caecum</td>\n",
       "      <td>R</td>\n",
       "      <td>III</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CMS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SC044_tumor3</td>\n",
       "      <td>SC044</td>\n",
       "      <td>EXT127</td>\n",
       "      <td>F</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>Caecum</td>\n",
       "      <td>R</td>\n",
       "      <td>III</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CMS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SC044_tumor4</td>\n",
       "      <td>SC044</td>\n",
       "      <td>EXT128</td>\n",
       "      <td>F</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>Caecum</td>\n",
       "      <td>R</td>\n",
       "      <td>III</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CMS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SC044_normal2</td>\n",
       "      <td>SC044</td>\n",
       "      <td>EXT129</td>\n",
       "      <td>F</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>Caecum</td>\n",
       "      <td>R</td>\n",
       "      <td>III</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CMS1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       sample_alias patient.ID file_ID Gender MSS_MSI             Site  \\\n",
       "0      SC035_tumor1      SC035  EXT048      M   MSI-H  Ascending colon   \n",
       "1      SC035_tumor2      SC035  EXT049      M   MSI-H  Ascending colon   \n",
       "2      SC035_normal      SC035  EXT050      M   MSI-H  Ascending colon   \n",
       "3   SC035_lymphnode      SC035  EXT051      M   MSI-H  Ascending colon   \n",
       "20     SC044_tumor1      SC044  EXT124      F   MSI-H           Caecum   \n",
       "21     SC044_tumor2      SC044  EXT125      F   MSI-H           Caecum   \n",
       "22    SC044_normal1      SC044  EXT126      F   MSI-H           Caecum   \n",
       "23     SC044_tumor3      SC044  EXT127      F   MSI-H           Caecum   \n",
       "24     SC044_tumor4      SC044  EXT128      F   MSI-H           Caecum   \n",
       "25    SC044_normal2      SC044  EXT129      F   MSI-H           Caecum   \n",
       "\n",
       "   Sidedness Group_Stage Stage_TNM iCMS_transcriptomic iCMS_inferCNV KRAS  \\\n",
       "0          R          II    T3N0M0               iCMS3         iCMS3   wt   \n",
       "1          R          II    T3N0M0               iCMS3         iCMS3   wt   \n",
       "2          R          II    T3N0M0               iCMS3         iCMS3   wt   \n",
       "3          R          II    T3N0M0               iCMS3         iCMS3   wt   \n",
       "20         R         III    T3N1M0               iCMS3         iCMS3   wt   \n",
       "21         R         III    T3N1M0               iCMS3         iCMS3   wt   \n",
       "22         R         III    T3N1M0               iCMS3         iCMS3   wt   \n",
       "23         R         III    T3N1M0               iCMS3         iCMS3   wt   \n",
       "24         R         III    T3N1M0               iCMS3         iCMS3   wt   \n",
       "25         R         III    T3N1M0               iCMS3         iCMS3   wt   \n",
       "\n",
       "   BRAF TP53  APC  PIK3CA  LymphNode  Normal  Tumor   CMS  \n",
       "0    wt   wt  mut     NaN        1.0     1.0    2.0  CMS1  \n",
       "1    wt   wt  mut     NaN        1.0     1.0    2.0  CMS1  \n",
       "2    wt   wt  mut     NaN        1.0     1.0    2.0  CMS1  \n",
       "3    wt   wt  mut     NaN        1.0     1.0    2.0  CMS1  \n",
       "20   wt   wt   wt     NaN        0.0     2.0    3.0  CMS1  \n",
       "21   wt   wt   wt     NaN        0.0     2.0    3.0  CMS1  \n",
       "22   wt   wt   wt     NaN        0.0     2.0    3.0  CMS1  \n",
       "23   wt   wt   wt     NaN        0.0     2.0    3.0  CMS1  \n",
       "24   wt   wt   wt     NaN        0.0     2.0    3.0  CMS1  \n",
       "25   wt   wt   wt     NaN        0.0     2.0    3.0  CMS1  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "KUL_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SC035\n",
      "SC044\n"
     ]
    }
   ],
   "source": [
    "adata = list()\n",
    "\n",
    "for patientID, edf in KUL_metadata.groupby(\"patient.ID\"):\n",
    "    print(patientID, flush=True)\n",
    "    \n",
    "    for tup in edf.itertuples():\n",
    "        PATH_cellranger_h5ad = f'{DIR_KUL_MSI_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_cellranger_scrublet_out = f'{DIR_KUL_MSI_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "        \n",
    "        if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "            if os.path.exists( PATH_cellranger_scrublet_out ) == True:  # Add to list\n",
    "                adata_temp = sc.read_h5ad( PATH_cellranger_scrublet_out )\n",
    "                \n",
    "                adata_temp.obs['SampleID']  = tup.file_ID\n",
    "                adata_temp.obs['PatientID'] = list(tup)[2]\n",
    "                adata_temp.obs['BiopsySite'] = list(tup)[1].split(\"_\")[1]\n",
    "    \n",
    "                adata.append( adata_temp ) \n",
    "                \n",
    "            else:   # Run scrublet, save to disk then add to list\n",
    "                print(f'{patientID}-{tup.file_ID} has not been processed by Scrublet! Running Scrublet now..')\n",
    "                adata_temp = sc.read_10x_h5( PATH_cellranger_h5ad )\n",
    "                adata_temp.obs_names_make_unique()\n",
    "                adata_temp.var_names_make_unique()\n",
    "                sc.pp.scrublet(adata_temp)\n",
    "                adata_temp.write(PATH_cellranger_scrublet_out)\n",
    "                adata.append( adata_temp )\n",
    "        else:   # Report\n",
    "            print(f'{patientID}-{tup.file_ID} has not been processed by CellRanger!')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/18parkky/anaconda3-2021.11/envs/sc/lib/python3.12/site-packages/anndata/_core/anndata.py:1818: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"obs\")\n"
     ]
    }
   ],
   "source": [
    "adata = sc.concat( adata  )\n",
    "adata.obs_names_make_unique()\n",
    "\n",
    "# mitochondrial genes, \"MT-\" for human, \"Mt-\" for mouse\n",
    "adata.var[\"mt\"] = adata.var_names.str.startswith(\"MT-\")\n",
    "# # ribosomal genes\n",
    "# adata.var[\"ribo\"] = adata.var_names.str.startswith((\"RPS\", \"RPL\"))\n",
    "# # hemoglobin genes\n",
    "# adata.var[\"hb\"] = adata.var_names.str.contains(\"^HB[^(P)]\")\n",
    "\n",
    "sc.pp.calculate_qc_metrics(\n",
    "    adata, qc_vars=[\"mt\",], inplace=True, log1p=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.obs.rename({'Sample' : \"Accession\"}, axis=1, inplace=True)\n",
    "adata.obs['BiopsySite'] = adata.obs['BiopsySite'].map( {\n",
    "                                                        'lymphnode' : 'LymphNode', \n",
    "                                                        'normal' : 'Adjacent normal tissue', \n",
    "                                                        'normal1' : 'Adjacent normal tissue',\n",
    "                                                        'normal2' : 'Adjacent normal tissue',\n",
    "                                                        'tumor1' : 'Tumor',\n",
    "                                                        'tumor2' : 'Tumor',\n",
    "                                                        'tumor3' : 'Tumor',\n",
    "                                                        'tumor4' : 'Tumor',\n",
    "                                                        },)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "PATH_adata_save = '/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/1_raw_h5ad/Joanito.KUL.MSI.raw.h5ad'\n",
    "adata.write(PATH_adata_save)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# adata = sc.read_h5ad(PATH_adata_save)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Joanito et al (MSS; 6 patients)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CRC-SG1 cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# iCMS2: CRC2801\n",
    "# iCMS3: CRC2786, CRC2787, CRC2821"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 MSS patients from:\tCRC-SG1 cohort\n",
      "3 MSS patients from:\tKUL cohort\n"
     ]
    }
   ],
   "source": [
    "joanito_metadata = pd.read_csv('/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/CRC-SG1/synapse-metadata/patient_clinical_information.csv', encoding = \"ISO-8859-1\")\n",
    "joanito_metadata = joanito_metadata[joanito_metadata['dataset'].isin(['CRC-SG1', 'KUL5', 'KUL3'])].copy()\n",
    "joanito_metadata = joanito_metadata[joanito_metadata['MSS_MSI']!='MSI-H'].copy()\n",
    "\n",
    "PATH_SG1_metadata = \"/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/CRC-SG1/EGAD00001008555-metadata/organized_metadata.tsv\"\n",
    "PATH_KUL_metadata = \"/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/KUL/organized_metadata.tsv\"\n",
    "\n",
    "SG1_metadata = pd.read_csv(PATH_SG1_metadata, sep='\\t')\n",
    "KUL_metadata = pd.read_csv(PATH_KUL_metadata, sep='\\t')\n",
    "\n",
    "SG1_metadata = SG1_metadata[SG1_metadata['patient.ID'].isin( joanito_metadata['patient.ID'] )].copy()\n",
    "KUL_metadata = KUL_metadata[KUL_metadata['patient.ID'].isin( joanito_metadata['patient.ID'] )].copy()\n",
    "\n",
    "print(f'{len(set(SG1_metadata['patient.ID']))} MSS patients from:\\tCRC-SG1 cohort')\n",
    "print(f'{len(set(KUL_metadata['patient.ID']))} MSS patients from:\\tKUL cohort')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "DIR_CRC_SG1_cellranger_out = '/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/CRC-SG1/cellranger'       #\n",
    "DIR_KUL_MSS_cellranger_out = '/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/KUL/iCMS3_MSS'            # iCMS3 MSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CRC2786 True iCMS3\n",
      "CRC2787 True iCMS3\n",
      "CRC2801 True iCMS2\n",
      "CRC2821 True iCMS3\n"
     ]
    }
   ],
   "source": [
    "MSS_patients = list()\n",
    "\n",
    "for patientID, edf in SG1_metadata.groupby(\"patient.ID\"):\n",
    "    boolFullyProcessed = True\n",
    "    \n",
    "    for tup in edf.itertuples():\n",
    "        PATH_cellranger_h5ad = f'{DIR_CRC_SG1_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_cellranger_scrublet_out = f'{DIR_CRC_SG1_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "        \n",
    "        if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "            pass \n",
    "\n",
    "        else:   # Report\n",
    "            boolFullyProcessed = False \n",
    "            break \n",
    "    \n",
    "    if boolFullyProcessed==True:\n",
    "        print(patientID, boolFullyProcessed, edf.iloc[0].iCMS_transcriptomic, flush=True)\n",
    "        MSS_patients.append( patientID )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CRC2786 iCMS3\n",
      "CRC2787 iCMS3\n",
      "CRC2801 iCMS2\n",
      "CRC2821 iCMS3\n"
     ]
    }
   ],
   "source": [
    "adata = list()\n",
    "\n",
    "for patientID, edf in SG1_metadata[SG1_metadata['patient.ID'].isin(MSS_patients)].groupby(\"patient.ID\"):\n",
    "    print(patientID, edf.iloc[0].iCMS_transcriptomic, flush=True)\n",
    "    \n",
    "    for tup in edf.itertuples():\n",
    "        PATH_cellranger_h5ad = f'{DIR_CRC_SG1_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_cellranger_scrublet_out = f'{DIR_CRC_SG1_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "        \n",
    "        if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "            if os.path.exists( PATH_cellranger_scrublet_out ) == True:  # Add to list\n",
    "                adata_temp = sc.read_h5ad( PATH_cellranger_scrublet_out )\n",
    "                \n",
    "                adata_temp.obs['SampleID']  = tup.file_ID\n",
    "                adata_temp.obs['PatientID'] = list(tup)[2]\n",
    "                adata_temp.obs['BiopsySite'] = list(tup)[1].split(\"_\")[1]\n",
    "                adata.append( adata_temp ) \n",
    "                \n",
    "            else:   # Run scrublet, save to disk then add to list\n",
    "                adata_temp = sc.read_10x_h5( PATH_cellranger_h5ad )\n",
    "                adata_temp.obs_names_make_unique()\n",
    "                adata_temp.var_names_make_unique()\n",
    "                sc.pp.scrublet(adata_temp)\n",
    "                adata_temp.write(PATH_cellranger_scrublet_out)\n",
    "                adata.append( adata_temp )\n",
    "        else:   # Report\n",
    "            print(f'{patientID}-{tup.file_ID} has not been processed by CellRanger!')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/18parkky/anaconda3-2021.11/envs/sc/lib/python3.12/site-packages/anndata/_core/anndata.py:1818: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"obs\")\n"
     ]
    }
   ],
   "source": [
    "adata = sc.concat( adata  )\n",
    "adata.obs_names_make_unique()\n",
    "\n",
    "# mitochondrial genes, \"MT-\" for human, \"Mt-\" for mouse\n",
    "adata.var[\"mt\"] = adata.var_names.str.startswith(\"MT-\")\n",
    "# # ribosomal genes\n",
    "# adata.var[\"ribo\"] = adata.var_names.str.startswith((\"RPS\", \"RPL\"))\n",
    "# # hemoglobin genes\n",
    "# adata.var[\"hb\"] = adata.var_names.str.contains(\"^HB[^(P)]\")\n",
    "\n",
    "sc.pp.calculate_qc_metrics(\n",
    "    adata, qc_vars=[\"mt\", ], inplace=True, log1p=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "PATH_adata_save = '/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/1_raw_h5ad/Joanito.CRC-SG1.MSS.raw.h5ad'\n",
    "adata.write(PATH_adata_save)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### KUL cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# iCMS3: SC040, SC041"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_alias</th>\n",
       "      <th>patient.ID</th>\n",
       "      <th>file_ID</th>\n",
       "      <th>Gender</th>\n",
       "      <th>MSS_MSI</th>\n",
       "      <th>Site</th>\n",
       "      <th>Sidedness</th>\n",
       "      <th>Group_Stage</th>\n",
       "      <th>Stage_TNM</th>\n",
       "      <th>iCMS_transcriptomic</th>\n",
       "      <th>iCMS_inferCNV</th>\n",
       "      <th>KRAS</th>\n",
       "      <th>BRAF</th>\n",
       "      <th>TP53</th>\n",
       "      <th>APC</th>\n",
       "      <th>PIK3CA</th>\n",
       "      <th>LymphNode</th>\n",
       "      <th>Normal</th>\n",
       "      <th>Tumor</th>\n",
       "      <th>CMS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SC040_tumor1</td>\n",
       "      <td>SC040</td>\n",
       "      <td>EXT093</td>\n",
       "      <td>F</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SC040_tumor2</td>\n",
       "      <td>SC040</td>\n",
       "      <td>EXT094</td>\n",
       "      <td>F</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SC040_normal1</td>\n",
       "      <td>SC040</td>\n",
       "      <td>EXT095</td>\n",
       "      <td>F</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SC040_lymphnode</td>\n",
       "      <td>SC040</td>\n",
       "      <td>EXT096</td>\n",
       "      <td>F</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SC040_tumor3</td>\n",
       "      <td>SC040</td>\n",
       "      <td>EXT097</td>\n",
       "      <td>F</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SC040_tumor4</td>\n",
       "      <td>SC040</td>\n",
       "      <td>EXT098</td>\n",
       "      <td>F</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SC040_normal2</td>\n",
       "      <td>SC040</td>\n",
       "      <td>EXT099</td>\n",
       "      <td>F</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>R</td>\n",
       "      <td>II</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SC041_tumor1</td>\n",
       "      <td>SC041</td>\n",
       "      <td>EXT113</td>\n",
       "      <td>M</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Rectosigmoid</td>\n",
       "      <td>L</td>\n",
       "      <td>III</td>\n",
       "      <td>T4N1M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>mut</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CMS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SC041_tumor2</td>\n",
       "      <td>SC041</td>\n",
       "      <td>EXT114</td>\n",
       "      <td>M</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Rectosigmoid</td>\n",
       "      <td>L</td>\n",
       "      <td>III</td>\n",
       "      <td>T4N1M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>mut</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CMS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SC041_normal</td>\n",
       "      <td>SC041</td>\n",
       "      <td>EXT115</td>\n",
       "      <td>M</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Rectosigmoid</td>\n",
       "      <td>L</td>\n",
       "      <td>III</td>\n",
       "      <td>T4N1M0</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>iCMS3</td>\n",
       "      <td>mut</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CMS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SC043_tumor1</td>\n",
       "      <td>SC043</td>\n",
       "      <td>EXT121</td>\n",
       "      <td>F</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Sigmoid colon</td>\n",
       "      <td>L</td>\n",
       "      <td>III</td>\n",
       "      <td>T4N1M0</td>\n",
       "      <td>iCMS2</td>\n",
       "      <td>iCMS2</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CMS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SC043_tumor2</td>\n",
       "      <td>SC043</td>\n",
       "      <td>EXT122</td>\n",
       "      <td>F</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Sigmoid colon</td>\n",
       "      <td>L</td>\n",
       "      <td>III</td>\n",
       "      <td>T4N1M0</td>\n",
       "      <td>iCMS2</td>\n",
       "      <td>iCMS2</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CMS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SC043_normal</td>\n",
       "      <td>SC043</td>\n",
       "      <td>EXT123</td>\n",
       "      <td>F</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Sigmoid colon</td>\n",
       "      <td>L</td>\n",
       "      <td>III</td>\n",
       "      <td>T4N1M0</td>\n",
       "      <td>iCMS2</td>\n",
       "      <td>iCMS2</td>\n",
       "      <td>mut</td>\n",
       "      <td>wt</td>\n",
       "      <td>mut</td>\n",
       "      <td>mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CMS2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       sample_alias patient.ID file_ID Gender MSS_MSI             Site  \\\n",
       "4      SC040_tumor1      SC040  EXT093      F     MSS  Ascending colon   \n",
       "5      SC040_tumor2      SC040  EXT094      F     MSS  Ascending colon   \n",
       "6     SC040_normal1      SC040  EXT095      F     MSS  Ascending colon   \n",
       "7   SC040_lymphnode      SC040  EXT096      F     MSS  Ascending colon   \n",
       "8      SC040_tumor3      SC040  EXT097      F     MSS  Ascending colon   \n",
       "9      SC040_tumor4      SC040  EXT098      F     MSS  Ascending colon   \n",
       "10    SC040_normal2      SC040  EXT099      F     MSS  Ascending colon   \n",
       "11     SC041_tumor1      SC041  EXT113      M     MSS     Rectosigmoid   \n",
       "12     SC041_tumor2      SC041  EXT114      M     MSS     Rectosigmoid   \n",
       "13     SC041_normal      SC041  EXT115      M     MSS     Rectosigmoid   \n",
       "17     SC043_tumor1      SC043  EXT121      F     MSS    Sigmoid colon   \n",
       "18     SC043_tumor2      SC043  EXT122      F     MSS    Sigmoid colon   \n",
       "19     SC043_normal      SC043  EXT123      F     MSS    Sigmoid colon   \n",
       "\n",
       "   Sidedness Group_Stage Stage_TNM iCMS_transcriptomic iCMS_inferCNV KRAS  \\\n",
       "4          R          II    T3N0M0               iCMS3         iCMS3  mut   \n",
       "5          R          II    T3N0M0               iCMS3         iCMS3  mut   \n",
       "6          R          II    T3N0M0               iCMS3         iCMS3  mut   \n",
       "7          R          II    T3N0M0               iCMS3         iCMS3  mut   \n",
       "8          R          II    T3N0M0               iCMS3         iCMS3  mut   \n",
       "9          R          II    T3N0M0               iCMS3         iCMS3  mut   \n",
       "10         R          II    T3N0M0               iCMS3         iCMS3  mut   \n",
       "11         L         III    T4N1M0               iCMS3         iCMS3  mut   \n",
       "12         L         III    T4N1M0               iCMS3         iCMS3  mut   \n",
       "13         L         III    T4N1M0               iCMS3         iCMS3  mut   \n",
       "17         L         III    T4N1M0               iCMS2         iCMS2  mut   \n",
       "18         L         III    T4N1M0               iCMS2         iCMS2  mut   \n",
       "19         L         III    T4N1M0               iCMS2         iCMS2  mut   \n",
       "\n",
       "   BRAF TP53  APC  PIK3CA  LymphNode  Normal  Tumor   CMS  \n",
       "4    wt   wt  mut     NaN        1.0     2.0    4.0   NaN  \n",
       "5    wt   wt  mut     NaN        1.0     2.0    4.0   NaN  \n",
       "6    wt   wt  mut     NaN        1.0     2.0    4.0   NaN  \n",
       "7    wt   wt  mut     NaN        1.0     2.0    4.0   NaN  \n",
       "8    wt   wt  mut     NaN        1.0     2.0    4.0   NaN  \n",
       "9    wt   wt  mut     NaN        1.0     2.0    4.0   NaN  \n",
       "10   wt   wt  mut     NaN        1.0     2.0    4.0   NaN  \n",
       "11  mut   wt  mut     NaN        0.0     1.0    2.0  CMS3  \n",
       "12  mut   wt  mut     NaN        0.0     1.0    2.0  CMS3  \n",
       "13  mut   wt  mut     NaN        0.0     1.0    2.0  CMS3  \n",
       "17   wt  mut  mut     NaN        0.0     1.0    2.0  CMS2  \n",
       "18   wt  mut  mut     NaN        0.0     1.0    2.0  CMS2  \n",
       "19   wt  mut  mut     NaN        0.0     1.0    2.0  CMS2  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "KUL_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/KUL/iCMS3_MSS'"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DIR_KUL_MSI_cellranger_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SC040 True iCMS3\n",
      "SC041 True iCMS3\n"
     ]
    }
   ],
   "source": [
    "MSS_patients = list()\n",
    "\n",
    "for patientID, edf in KUL_metadata.groupby(\"patient.ID\"):\n",
    "    boolFullyProcessed = True\n",
    "    \n",
    "    for tup in edf.itertuples():\n",
    "        PATH_cellranger_h5ad = f'{DIR_KUL_MSS_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_cellranger_scrublet_out = f'{DIR_KUL_MSS_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "        \n",
    "        if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "            pass \n",
    "\n",
    "        else:   # Report\n",
    "            boolFullyProcessed = False \n",
    "            break \n",
    "    \n",
    "    if boolFullyProcessed==True:\n",
    "        print(patientID, boolFullyProcessed, edf.iloc[0].iCMS_transcriptomic, flush=True)\n",
    "        MSS_patients.append( patientID )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SC040\n",
      "SC041\n"
     ]
    }
   ],
   "source": [
    "adata = list()\n",
    "\n",
    "for patientID, edf in KUL_metadata[KUL_metadata['patient.ID'].isin(MSS_patients)].groupby(\"patient.ID\"):\n",
    "    print(patientID, flush=True)\n",
    "    \n",
    "    for tup in edf.itertuples():\n",
    "        PATH_cellranger_h5ad = f'{DIR_KUL_MSS_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_cellranger_scrublet_out = f'{DIR_KUL_MSS_cellranger_out}/{tup.file_ID}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "        \n",
    "        if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "            if os.path.exists( PATH_cellranger_scrublet_out ) == True:  # Add to list\n",
    "                adata_temp = sc.read_h5ad( PATH_cellranger_scrublet_out )\n",
    "                \n",
    "                adata_temp.obs['SampleID']  = tup.file_ID\n",
    "                adata_temp.obs['PatientID'] = list(tup)[2]\n",
    "                adata_temp.obs['BiopsySite'] = list(tup)[1].split(\"_\")[1]\n",
    "    \n",
    "                adata.append( adata_temp ) \n",
    "                \n",
    "            else:   # Run scrublet, save to disk then add to list\n",
    "                print(f'{patientID}-{tup.file_ID} has not been processed by Scrublet! Running Scrublet now..')\n",
    "                adata_temp = sc.read_10x_h5( PATH_cellranger_h5ad )\n",
    "                adata_temp.obs_names_make_unique()\n",
    "                adata_temp.var_names_make_unique()\n",
    "                sc.pp.scrublet(adata_temp)\n",
    "                adata_temp.write(PATH_cellranger_scrublet_out)\n",
    "                adata.append( adata_temp )\n",
    "        else:   # Report\n",
    "            print(f'{patientID}-{tup.file_ID} has not been processed by CellRanger!')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/18parkky/anaconda3-2021.11/envs/sc/lib/python3.12/site-packages/anndata/_core/anndata.py:1818: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"obs\")\n"
     ]
    }
   ],
   "source": [
    "adata = sc.concat( adata  )\n",
    "adata.obs_names_make_unique()\n",
    "\n",
    "# mitochondrial genes, \"MT-\" for human, \"Mt-\" for mouse\n",
    "adata.var[\"mt\"] = adata.var_names.str.startswith(\"MT-\")\n",
    "# # ribosomal genes\n",
    "# adata.var[\"ribo\"] = adata.var_names.str.startswith((\"RPS\", \"RPL\"))\n",
    "# # hemoglobin genes\n",
    "# adata.var[\"hb\"] = adata.var_names.str.contains(\"^HB[^(P)]\")\n",
    "\n",
    "sc.pp.calculate_qc_metrics(\n",
    "    adata, qc_vars=[\"mt\", ], inplace=True, log1p=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.obs.rename({'Sample' : \"Accession\"}, axis=1, inplace=True)\n",
    "adata.obs['BiopsySite'] = adata.obs['BiopsySite'].map( {\n",
    "                                                        'lymphnode' : 'LymphNode', \n",
    "                                                        'normal' : 'Adjacent normal tissue', \n",
    "                                                        'normal1' : 'Adjacent normal tissue',\n",
    "                                                        'normal2' : 'Adjacent normal tissue',\n",
    "                                                        'tumor1' : 'Tumor',\n",
    "                                                        'tumor2' : 'Tumor',\n",
    "                                                        'tumor3' : 'Tumor',\n",
    "                                                        'tumor4' : 'Tumor',\n",
    "                                                        },)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "PATH_adata_save = '/node200data/18parkky/datasets/data/public/processed_data/Joanito_et_al/1_raw_h5ad/Joanito.KUL.MSS.raw.h5ad'\n",
    "adata.write(PATH_adata_save)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Chen et al"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 MSI-H patients from:\tChen et al. cohort\n"
     ]
    }
   ],
   "source": [
    "chen_metadata = pd.read_csv('/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/metadata/Chen_et_al.MSI.run_meta.organized.tsv', sep='\\t')\n",
    "chen_metadata = chen_metadata[['Accession', 'SampleID', 'PatientID', 'TreatmentStage', 'TreatmentPoint', 'BiopsySite', 'MMR', 'MSI', 'TMB', 'Response', 'LibraryType']].copy()\n",
    "chen_metadata = chen_metadata[(chen_metadata['LibraryType']=='scRNA-seq')].copy()\n",
    "chen_metadata = chen_metadata[(chen_metadata['BiopsySite'].isin( ['Adjacent normal tissue', 'Tumor'] ))].copy()\n",
    "chen_metadata['BiopsySite'] = chen_metadata['BiopsySite'].map( { \n",
    "                                                                'Adjacent normal tissue' : 'Normal', \n",
    "                                                                'Tumor' : 'Tumor', \n",
    "                                                                })\n",
    "\n",
    "chen_patient_metadata = pd.read_csv('/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/metadata/Chen_et_al.patient_meta.csv',)\n",
    "dict_PatientID_to_TumorLoc   = { list(tup)[1] : list(tup)[5] for tup in chen_patient_metadata.itertuples() }\n",
    "dict_PatientID_to_TumorLoc['P15'] = 'Sigmoid colon'\n",
    "dict_PatientID_to_CancerType = { list(tup)[1] : list(tup)[4] for tup in chen_patient_metadata.itertuples() }\n",
    "\n",
    "chen_metadata['TumorLocation'] = [ dict_PatientID_to_TumorLoc[PID] for PID in chen_metadata['PatientID'] ]\n",
    "chen_metadata['CancerType'] = [ dict_PatientID_to_CancerType[PID] for PID in chen_metadata['PatientID'] ]\n",
    "\n",
    "chen_metadata = chen_metadata[(chen_metadata['CancerType']=='CRC') & (chen_metadata['Response']!='SD')]\n",
    "\n",
    "print(f'{len(set(chen_metadata['PatientID']))} MSI-H patients from:\\tChen et al. cohort')\n",
    "chen_metadata.to_csv('/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/metadata/Chen_et_al.MSI.CRC.woSD.run_meta.organized.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. Run Cell ranger of each sample (input mutliple FASTQ of same sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(64, 24)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(chen_metadata['SampleID'])), len(set(chen_metadata[chen_metadata['TreatmentStage']=='Pre']['SampleID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "PATH_cellranger_count_bin       = '/home/shared/program/cellranger-8.0.1/bin/cellranger'\n",
    "PATH_cellranger_transcriptome   = '/home/shared/cellranger_reference/refdata-gex-GRCh38-2020-A'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Total number of samples we need to process\n",
    "len(set(chen_metadata[(chen_metadata['TreatmentStage']=='Pre') & (chen_metadata['BiopsySite']=='Tumor')]['SampleID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Tumor\n",
    "DIR_merged_cellranger_out = '/node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged'\n",
    "DIR_FASTQ_out = '/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Tumor'\n",
    "\n",
    "run_cellranger_sh   = open(f'{DIR_merged_cellranger_out}/run_celleranger.sh', 'w')\n",
    "run_download_fq_sh  = open(f'{DIR_FASTQ_out}/download_fastq.sh', 'w')\n",
    "\n",
    "for SampleID, edf in chen_metadata[(chen_metadata['TreatmentStage']=='Pre') & (chen_metadata['BiopsySite']=='Tumor')].groupby(\"SampleID\"):\n",
    "    DIR_FASTQ = '/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Tumor'\n",
    "    allFastqExists = True\n",
    "    list_PATH_FASTQ = list()\n",
    "    for tup in edf.itertuples():\n",
    "        PATH_FASTQ = f'{DIR_FASTQ}/{tup.Accession}'\n",
    "        if os.path.exists(PATH_FASTQ)==False:\n",
    "            # print(f'{SampleID}\\t{tup.Accession}')\n",
    "            allFastqExists = False \n",
    "            break \n",
    "        else:\n",
    "            list_PATH_FASTQ.append( PATH_FASTQ )\n",
    "\n",
    "    if allFastqExists == True:\n",
    "        PATH_merged_cellranger_out = f'{DIR_merged_cellranger_out}/{SampleID}'\n",
    "        if os.path.exists(PATH_merged_cellranger_out)==True:\n",
    "            continue \n",
    "        cmd = f'{PATH_cellranger_count_bin} count --id {SampleID} --transcriptome {PATH_cellranger_transcriptome} --fastqs {\",\".join(list_PATH_FASTQ)} --create-bam true --nosecondary --output-dir {DIR_merged_cellranger_out}/{SampleID}'\n",
    "        print(cmd)\n",
    "        run_cellranger_sh.write(f'{cmd}\\n')\n",
    "    else:\n",
    "        print(f'FASTQ not intact: {SampleID} ({len(list_PATH_FASTQ)}/{len(edf)} found)')\n",
    "        for tup in edf.itertuples():\n",
    "            PATH_FASTQ = f'{DIR_FASTQ}/{tup.Accession}'\n",
    "            if os.path.exists(PATH_FASTQ)==False:\n",
    "                print(f'\\t{SampleID}\\t{tup.Accession}')\n",
    "                run_download_fq_sh.write( f'wget ftp://human.big.ac.cn:/HRA005546/{tup.Accession}/{tup.Accession}_f1.fastq.gz -O {DIR_FASTQ_out}/{tup.Accession}/{tup.Accession}_S1_L001_R1_001.fastq.gz --ftp-user=\\'jihwan.park@gist.ac.kr\\' --ftp-password=\\'Wlshalrtm_204!\\'\\n' )\n",
    "                run_download_fq_sh.write( f'wget ftp://human.big.ac.cn:/HRA005546/{tup.Accession}/{tup.Accession}_r2.fastq.gz -O {DIR_FASTQ_out}/{tup.Accession}/{tup.Accession}_S1_L001_R2_001.fastq.gz --ftp-user=\\'jihwan.park@gist.ac.kr\\' --ftp-password=\\'Wlshalrtm_204!\\'\\n' )\n",
    "\n",
    "run_cellranger_sh.close()\n",
    "run_download_fq_sh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/shared/program/cellranger-8.0.1/bin/cellranger count --id P11-N-I --transcriptome /home/shared/cellranger_reference/refdata-gex-GRCh38-2020-A --fastqs /node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373268,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373267,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373266,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373265 --create-bam true --nosecondary --output-dir /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue_merged/P11-N-I\n",
      "/home/shared/program/cellranger-8.0.1/bin/cellranger count --id P17-N-I --transcriptome /home/shared/cellranger_reference/refdata-gex-GRCh38-2020-A --fastqs /node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373512,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373511,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373510,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373509 --create-bam true --nosecondary --output-dir /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue_merged/P17-N-I\n",
      "/home/shared/program/cellranger-8.0.1/bin/cellranger count --id P18-N-I --transcriptome /home/shared/cellranger_reference/refdata-gex-GRCh38-2020-A --fastqs /node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373564,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373563,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373562,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373561 --create-bam true --nosecondary --output-dir /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue_merged/P18-N-I\n",
      "/home/shared/program/cellranger-8.0.1/bin/cellranger count --id P19-N-I --transcriptome /home/shared/cellranger_reference/refdata-gex-GRCh38-2020-A --fastqs /node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373588,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373587,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373586,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373585 --create-bam true --nosecondary --output-dir /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue_merged/P19-N-I\n",
      "/home/shared/program/cellranger-8.0.1/bin/cellranger count --id P20-N-I --transcriptome /home/shared/cellranger_reference/refdata-gex-GRCh38-2020-A --fastqs /node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373612,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373611,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373610,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373609 --create-bam true --nosecondary --output-dir /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue_merged/P20-N-I\n",
      "/home/shared/program/cellranger-8.0.1/bin/cellranger count --id P26-N-I --transcriptome /home/shared/cellranger_reference/refdata-gex-GRCh38-2020-A --fastqs /node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373820,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373819,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373818,/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/HRR1373817 --create-bam true --nosecondary --output-dir /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue_merged/P26-N-I\n"
     ]
    }
   ],
   "source": [
    "# NAT\n",
    "DIR_merged_cellranger_out = '/node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue_merged'\n",
    "DIR_FASTQ_out = '/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue'\n",
    "\n",
    "run_cellranger_sh   = open(f'{DIR_merged_cellranger_out}/run_celleranger.sh', 'w')\n",
    "run_download_fq_sh  = open(f'{DIR_FASTQ_out}/download_fastq.sh', 'w')\n",
    "\n",
    "# _S1_L001_R1_001\n",
    "for SampleID, edf in chen_metadata[(chen_metadata['TreatmentStage']=='Pre') & (chen_metadata['BiopsySite']!='Tumor')].groupby(\"SampleID\"):\n",
    "    DIR_FASTQ = '/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue'\n",
    "    allFastqExists = True\n",
    "    list_PATH_FASTQ = list()\n",
    "    for tup in edf.itertuples():\n",
    "        PATH_FASTQ = f'{DIR_FASTQ}/{tup.Accession}'\n",
    "        if os.path.exists(PATH_FASTQ)==False:\n",
    "            # print(f'{SampleID}\\t{tup.Accession}')\n",
    "            allFastqExists = False \n",
    "            break \n",
    "        else:\n",
    "            list_PATH_FASTQ.append( PATH_FASTQ )\n",
    "\n",
    "    if allFastqExists == True:\n",
    "        PATH_merged_cellranger_out = f'{DIR_merged_cellranger_out}/{SampleID}'\n",
    "        if os.path.exists(PATH_merged_cellranger_out)==True:\n",
    "            continue \n",
    "        cmd = f'{PATH_cellranger_count_bin} count --id {SampleID} --transcriptome {PATH_cellranger_transcriptome} --fastqs {\",\".join(list_PATH_FASTQ)} --create-bam true --nosecondary --output-dir {DIR_merged_cellranger_out}/{SampleID}'\n",
    "        run_cellranger_sh.write(f'{cmd}\\n')\n",
    "\n",
    "        print(cmd)\n",
    "    else:\n",
    "        print(f'FASTQ not intact: {SampleID} ({len(list_PATH_FASTQ)}/{len(edf)} found)')\n",
    "        for tup in edf.itertuples():\n",
    "            PATH_FASTQ = f'{DIR_FASTQ}/{tup.Accession}'\n",
    "            if os.path.exists(PATH_FASTQ)==False:\n",
    "                print(f'\\t{SampleID}\\t{tup.Accession}')\n",
    "                run_download_fq_sh.write(f'mkdir {PATH_FASTQ}\\n')\n",
    "                run_download_fq_sh.write( f'wget ftp://human.big.ac.cn:/HRA005546/{tup.Accession}/{tup.Accession}_f1.fastq.gz -O {PATH_FASTQ}/{tup.Accession}_S1_L001_R1_001.fastq.gz --ftp-user=\\'jihwan.park@gist.ac.kr\\' --ftp-password=\\'Wlshalrtm_204!\\'\\n' )\n",
    "                run_download_fq_sh.write( f'wget ftp://human.big.ac.cn:/HRA005546/{tup.Accession}/{tup.Accession}_r2.fastq.gz -O {PATH_FASTQ}/{tup.Accession}_S1_L001_R2_001.fastq.gz --ftp-user=\\'jihwan.park@gist.ac.kr\\' --ftp-password=\\'Wlshalrtm_204!\\'\\n' )\n",
    "\n",
    "run_cellranger_sh.close()\n",
    "run_download_fq_sh.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Run NanoMnT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P15-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P15-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P08-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P08-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P16-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P16-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P18-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P18-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P19-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P19-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P20-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P20-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P21-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P21-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P26-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P26-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P11-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P11-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P12-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P12-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P17-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P17-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P22-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P22-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P23-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P23-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P24-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P24-T-I/outs\n",
      "getAlleleTable -b /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P25-T-I/outs/possorted_genome_bam.bam -s /home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz -r /node200data/18parkky/datasets/reference/genome/hg38.fa -t 36 -f 12 -m 1 --sc -rf 1000 -out /node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged/P25-T-I/outs\n"
     ]
    }
   ],
   "source": [
    "# Tumor\n",
    "DIR_merged_cellranger_out = '/node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged'\n",
    "list_PATH_possorted_BAM = [ f'{DIR}/outs/possorted_genome_bam.bam' for DIR in glob.glob(f'{DIR_merged_cellranger_out}/*') if (os.path.exists(f'{DIR}/outs/possorted_genome_bam.bam')==True) and (os.path.exists(f'{DIR}/outs/possorted_genome_bam.STR_allele_table.tsv')==False) ]\n",
    "\n",
    "PATH_krait = '/home/18parkky/git/NanoMnT/ref/krait/raw/GRCh38/GRCh38-1bp_STR.gene.tsv.gz'\n",
    "PATH_ref_genome = '/node200data/18parkky/datasets/reference/genome/hg38.fa'\n",
    "num_threads = 36\n",
    "realignment_flanking = 1000 # no point in using 20000 bp flanking for short-read data \n",
    "\n",
    "# run_nanomnt_sh = open(f'{DIR_merged_cellranger_out}/run_nanomnt.sh', 'w')\n",
    "\n",
    "for PATH_possorted_BAM in list_PATH_possorted_BAM:\n",
    "    PATH_nanomnt_out = \"/\".join(PATH_possorted_BAM.split(\"/\")[:-1])\n",
    "    PATH_getAlleleTable_out     = f'{PATH_nanomnt_out}/possorted_genome_bam.STR_allele_table.tsv'\n",
    "    DIR_getAlleleTable_tmpdir   = f'{PATH_nanomnt_out}/possorted_genome_bam.multiprocessing_temp'\n",
    "    if os.path.exists(PATH_getAlleleTable_out)==False and os.path.exists(DIR_getAlleleTable_tmpdir)==False:\n",
    "        cmd = f'getAlleleTable -b {PATH_possorted_BAM} -s {PATH_krait} -r {PATH_ref_genome} -t {num_threads} -f 12 -m 1 --sc -rf {realignment_flanking} -out {PATH_nanomnt_out}'\n",
    "        # run_nanomnt_sh.write(f'{cmd}\\n')\n",
    "        print(cmd)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Merge data "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>PatientID</th>\n",
       "      <th>TreatmentStage</th>\n",
       "      <th>TreatmentPoint</th>\n",
       "      <th>BiopsySite</th>\n",
       "      <th>MMR</th>\n",
       "      <th>MSI</th>\n",
       "      <th>TMB</th>\n",
       "      <th>Response</th>\n",
       "      <th>LibraryType</th>\n",
       "      <th>TumorLocation</th>\n",
       "      <th>CancerType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>HRR1372976</td>\n",
       "      <td>P04-N-I</td>\n",
       "      <td>P04</td>\n",
       "      <td>Pre</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>Normal</td>\n",
       "      <td>dMMR</td>\n",
       "      <td>MSI</td>\n",
       "      <td>487</td>\n",
       "      <td>CR</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>CRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>HRR1372975</td>\n",
       "      <td>P04-N-I</td>\n",
       "      <td>P04</td>\n",
       "      <td>Pre</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>Normal</td>\n",
       "      <td>dMMR</td>\n",
       "      <td>MSI</td>\n",
       "      <td>487</td>\n",
       "      <td>CR</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>CRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HRR1372974</td>\n",
       "      <td>P04-N-I</td>\n",
       "      <td>P04</td>\n",
       "      <td>Pre</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>Normal</td>\n",
       "      <td>dMMR</td>\n",
       "      <td>MSI</td>\n",
       "      <td>487</td>\n",
       "      <td>CR</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>CRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>HRR1372973</td>\n",
       "      <td>P04-N-I</td>\n",
       "      <td>P04</td>\n",
       "      <td>Pre</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>Normal</td>\n",
       "      <td>dMMR</td>\n",
       "      <td>MSI</td>\n",
       "      <td>487</td>\n",
       "      <td>CR</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>CRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>HRR1373024</td>\n",
       "      <td>P04-N-II</td>\n",
       "      <td>P04</td>\n",
       "      <td>On</td>\n",
       "      <td>Post 2 cycle</td>\n",
       "      <td>Normal</td>\n",
       "      <td>dMMR</td>\n",
       "      <td>MSI</td>\n",
       "      <td>487</td>\n",
       "      <td>CR</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>CRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>HRR1373833</td>\n",
       "      <td>P26-T-I</td>\n",
       "      <td>P26</td>\n",
       "      <td>Pre</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>dMMR</td>\n",
       "      <td>MSI</td>\n",
       "      <td>75.85</td>\n",
       "      <td>CR</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>CRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>HRR1374144</td>\n",
       "      <td>P26-T-II</td>\n",
       "      <td>P26</td>\n",
       "      <td>Post</td>\n",
       "      <td>Post 3 cycle</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>dMMR</td>\n",
       "      <td>MSI</td>\n",
       "      <td>75.85</td>\n",
       "      <td>CR</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>CRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>HRR1374143</td>\n",
       "      <td>P26-T-II</td>\n",
       "      <td>P26</td>\n",
       "      <td>Post</td>\n",
       "      <td>Post 3 cycle</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>dMMR</td>\n",
       "      <td>MSI</td>\n",
       "      <td>75.85</td>\n",
       "      <td>CR</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>CRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>HRR1374142</td>\n",
       "      <td>P26-T-II</td>\n",
       "      <td>P26</td>\n",
       "      <td>Post</td>\n",
       "      <td>Post 3 cycle</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>dMMR</td>\n",
       "      <td>MSI</td>\n",
       "      <td>75.85</td>\n",
       "      <td>CR</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>CRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>HRR1374141</td>\n",
       "      <td>P26-T-II</td>\n",
       "      <td>P26</td>\n",
       "      <td>Post</td>\n",
       "      <td>Post 3 cycle</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>dMMR</td>\n",
       "      <td>MSI</td>\n",
       "      <td>75.85</td>\n",
       "      <td>CR</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>Ascending colon</td>\n",
       "      <td>CRC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>256 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Accession  SampleID PatientID TreatmentStage TreatmentPoint BiopsySite  \\\n",
       "24   HRR1372976   P04-N-I       P04            Pre       Baseline     Normal   \n",
       "25   HRR1372975   P04-N-I       P04            Pre       Baseline     Normal   \n",
       "26   HRR1372974   P04-N-I       P04            Pre       Baseline     Normal   \n",
       "27   HRR1372973   P04-N-I       P04            Pre       Baseline     Normal   \n",
       "32   HRR1373024  P04-N-II       P04             On   Post 2 cycle     Normal   \n",
       "..          ...       ...       ...            ...            ...        ...   \n",
       "979  HRR1373833   P26-T-I       P26            Pre       Baseline      Tumor   \n",
       "984  HRR1374144  P26-T-II       P26           Post   Post 3 cycle      Tumor   \n",
       "985  HRR1374143  P26-T-II       P26           Post   Post 3 cycle      Tumor   \n",
       "986  HRR1374142  P26-T-II       P26           Post   Post 3 cycle      Tumor   \n",
       "987  HRR1374141  P26-T-II       P26           Post   Post 3 cycle      Tumor   \n",
       "\n",
       "      MMR  MSI    TMB Response LibraryType    TumorLocation CancerType  \n",
       "24   dMMR  MSI    487       CR   scRNA-seq  Ascending colon        CRC  \n",
       "25   dMMR  MSI    487       CR   scRNA-seq  Ascending colon        CRC  \n",
       "26   dMMR  MSI    487       CR   scRNA-seq  Ascending colon        CRC  \n",
       "27   dMMR  MSI    487       CR   scRNA-seq  Ascending colon        CRC  \n",
       "32   dMMR  MSI    487       CR   scRNA-seq  Ascending colon        CRC  \n",
       "..    ...  ...    ...      ...         ...              ...        ...  \n",
       "979  dMMR  MSI  75.85       CR   scRNA-seq  Ascending colon        CRC  \n",
       "984  dMMR  MSI  75.85       CR   scRNA-seq  Ascending colon        CRC  \n",
       "985  dMMR  MSI  75.85       CR   scRNA-seq  Ascending colon        CRC  \n",
       "986  dMMR  MSI  75.85       CR   scRNA-seq  Ascending colon        CRC  \n",
       "987  dMMR  MSI  75.85       CR   scRNA-seq  Ascending colon        CRC  \n",
       "\n",
       "[256 rows x 13 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chen_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# NAT\n",
    "DIR_merged_NAT_cellranger_out = '/node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue_merged'\n",
    "\n",
    "for SampleID, edf in chen_metadata[(chen_metadata['TreatmentStage']=='Pre') & (chen_metadata['BiopsySite']=='Normal')].groupby('SampleID'):\n",
    "    PATH_cellranger_h5ad = f'{DIR_merged_NAT_cellranger_out}/{SampleID}/outs/filtered_feature_bc_matrix.h5'\n",
    "    PATH_cellranger_scrublet_out = f'{DIR_merged_NAT_cellranger_out}/{SampleID}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "\n",
    "    if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "        if os.path.exists( PATH_cellranger_scrublet_out ) == True:  # Add to list\n",
    "            adata_temp = sc.read_h5ad( PATH_cellranger_scrublet_out )\n",
    "            \n",
    "            adata_temp.obs['Accession'] = edf.iloc[0].Accession\n",
    "            adata_temp.obs['SampleID']  = SampleID\n",
    "            adata_temp.obs['PatientID'] = edf.iloc[0].PatientID\n",
    "            adata_temp.obs['BiopsySite'] = edf.iloc[0].BiopsySite\n",
    "\n",
    "            adata.append( adata_temp ) \n",
    "            pass\n",
    "            \n",
    "        else:   # Run scrublet, save to disk then add to list\n",
    "            adata_temp = sc.read_10x_h5( PATH_cellranger_h5ad )\n",
    "            adata_temp.obs_names_make_unique()\n",
    "            adata_temp.var_names_make_unique()\n",
    "            \n",
    "            sc.pp.scrublet(adata_temp)\n",
    "            \n",
    "            adata_temp.obs['Accession'] = edf.iloc[0].Accession\n",
    "            adata_temp.obs['SampleID']  = SampleID\n",
    "            adata_temp.obs['PatientID'] = edf.iloc[0].PatientID\n",
    "            adata_temp.obs['BiopsySite'] = edf.iloc[0].BiopsySite\n",
    "            \n",
    "            adata_temp.write(PATH_cellranger_scrublet_out)\n",
    "            adata.append( adata_temp )\n",
    "            pass\n",
    "    else:   # Report\n",
    "        print(f'{patientID}-{tup.Accession} has not been processed by CellRanger!')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Tumor\n",
    "DIR_merged_Tumor_cellranger_out = '/node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor_merged'\n",
    "\n",
    "for SampleID, edf in chen_metadata[(chen_metadata['TreatmentStage']=='Pre') & (chen_metadata['BiopsySite']=='Tumor')].groupby('SampleID'):\n",
    "    PATH_cellranger_h5ad = f'{DIR_merged_Tumor_cellranger_out}/{SampleID}/outs/filtered_feature_bc_matrix.h5'\n",
    "    PATH_cellranger_scrublet_out = f'{DIR_merged_Tumor_cellranger_out}/{SampleID}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "\n",
    "    if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "        if os.path.exists( PATH_cellranger_scrublet_out ) == True:  # Add to list\n",
    "            adata_temp = sc.read_h5ad( PATH_cellranger_scrublet_out )\n",
    "            \n",
    "            adata_temp.obs['Accession'] = edf.iloc[0].Accession\n",
    "            adata_temp.obs['SampleID']  = SampleID\n",
    "            adata_temp.obs['PatientID'] = edf.iloc[0].PatientID\n",
    "            adata_temp.obs['BiopsySite'] = edf.iloc[0].BiopsySite\n",
    "\n",
    "            adata.append( adata_temp ) \n",
    "            pass\n",
    "            \n",
    "        else:   # Run scrublet, save to disk then add to list\n",
    "            adata_temp = sc.read_10x_h5( PATH_cellranger_h5ad )\n",
    "            adata_temp.obs_names_make_unique()\n",
    "            adata_temp.var_names_make_unique()\n",
    "            \n",
    "            sc.pp.scrublet(adata_temp)\n",
    "            \n",
    "            adata_temp.obs['Accession'] = edf.iloc[0].Accession\n",
    "            adata_temp.obs['SampleID']  = SampleID\n",
    "            adata_temp.obs['PatientID'] = edf.iloc[0].PatientID\n",
    "            adata_temp.obs['BiopsySite'] = edf.iloc[0].BiopsySite\n",
    "            \n",
    "            adata_temp.write(PATH_cellranger_scrublet_out)\n",
    "            adata.append( adata_temp )\n",
    "            pass\n",
    "    else:   # Report\n",
    "        print(f'{patientID}-{tup.Accession} has not been processed by CellRanger!')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/18parkky/anaconda3-2021.11/envs/sc/lib/python3.12/site-packages/anndata/_core/anndata.py:1818: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"obs\")\n"
     ]
    }
   ],
   "source": [
    "adata = sc.concat( adata  )\n",
    "adata.obs_names_make_unique()\n",
    "\n",
    "# mitochondrial genes, \"MT-\" for human, \"Mt-\" for mouse\n",
    "adata.var[\"mt\"] = adata.var_names.str.startswith(\"MT-\")\n",
    "# # ribosomal genes\n",
    "# adata.var[\"ribo\"] = adata.var_names.str.startswith((\"RPS\", \"RPL\"))\n",
    "# # hemoglobin genes\n",
    "# adata.var[\"hb\"] = adata.var_names.str.contains(\"^HB[^(P)]\")\n",
    "\n",
    "sc.pp.calculate_qc_metrics(\n",
    "    adata, qc_vars=[\"mt\",], inplace=True, log1p=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "PATH_adata_save = '/node200data/18parkky/datasets/data/public/processed_data/Immunotherapy_CRC_Chen_et_al/1_raw_h5ad/Chen.MSI.raw.h5ad'\n",
    "adata.write(PATH_adata_save)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "raise ValueError"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = list()\n",
    "processed_sample_count = 0\n",
    "unprocessed_sample_count = 0\n",
    "\n",
    "for patientID, edf in chen_metadata.groupby('PatientID'):\n",
    "    print(patientID, flush=True)\n",
    "    for tup in edf.itertuples():\n",
    "    \n",
    "        if tup.BiopsySite == 'Normal':\n",
    "            DIR_Chen_cellranger_out_temp = f'{DIR_Chen_cellranger_out}/Adjacent_normal_tissue'\n",
    "        elif tup.BiopsySite == 'Tumor':\n",
    "            DIR_Chen_cellranger_out_temp = f'{DIR_Chen_cellranger_out}/Tumor'\n",
    "            \n",
    "        PATH_cellranger_h5ad = f'{DIR_Chen_cellranger_out_temp}/{tup.Accession}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_cellranger_scrublet_out = f'{DIR_Chen_cellranger_out_temp}/{tup.Accession}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "        \n",
    "        if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "            if os.path.exists( PATH_cellranger_scrublet_out ) == True:  # Add to list\n",
    "                adata_temp = sc.read_h5ad( PATH_cellranger_scrublet_out )\n",
    "                \n",
    "                adata_temp.obs['SampleID']  = tup.Accession\n",
    "                adata_temp.obs['PatientID'] = tup.PatientID\n",
    "                adata_temp.obs['BiopsySite'] = tup.BiopsySite\n",
    "\n",
    "                adata.append( adata_temp ) \n",
    "                pass\n",
    "                \n",
    "            else:   # Run scrublet, save to disk then add to list\n",
    "                adata_temp = sc.read_10x_h5( PATH_cellranger_h5ad )\n",
    "                adata_temp.obs_names_make_unique()\n",
    "                adata_temp.var_names_make_unique()\n",
    "                sc.pp.scrublet(adata_temp)\n",
    "                adata_temp.write(PATH_cellranger_scrublet_out)\n",
    "                adata.append( adata_temp )\n",
    "                pass\n",
    "        else:   # Report\n",
    "            unprocessed_sample_count += 1\n",
    "            print(f'{patientID}-{tup.Accession} has not been processed by CellRanger!')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# adata = sc.concat( adata  )\n",
    "# adata.obs_names_make_unique()\n",
    "\n",
    "# # mitochondrial genes, \"MT-\" for human, \"Mt-\" for mouse\n",
    "# adata.var[\"mt\"] = adata.var_names.str.startswith(\"MT-\")\n",
    "# # ribosomal genes\n",
    "# adata.var[\"ribo\"] = adata.var_names.str.startswith((\"RPS\", \"RPL\"))\n",
    "# # hemoglobin genes\n",
    "# adata.var[\"hb\"] = adata.var_names.str.contains(\"^HB[^(P)]\")\n",
    "\n",
    "# sc.pp.calculate_qc_metrics(\n",
    "#     adata, qc_vars=[\"mt\", \"ribo\", \"hb\"], inplace=True, log1p=True\n",
    "# )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# adata.obs['Accession'] = adata.obs['SampleID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# adata.write('/node200data/18parkky/datasets/data/public/processed_data/Immunotherapy_CRC_Chen_et_al/1_raw_h5ad/Chen.MSI.raw.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "raise ValueError"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Depr (Chen)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Optional) Summarize sample preprocessing status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "DIR_Chen_cellranger_out = '/node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_status_summary = list()\n",
    "\n",
    "for patientID, edf in chen_metadata.groupby('PatientID'):\n",
    "    for tup in edf.itertuples():\n",
    "    \n",
    "        if tup.BiopsySite == 'Normal':\n",
    "            DIR_Chen_cellranger_out_temp = f'{DIR_Chen_cellranger_out}/Adjacent_normal_tissue'\n",
    "        elif tup.BiopsySite == 'Tumor':\n",
    "            DIR_Chen_cellranger_out_temp = f'{DIR_Chen_cellranger_out}/Tumor'\n",
    "        else: raise ValueError    \n",
    "        \n",
    "        PATH_cellranger_h5ad = f'{DIR_Chen_cellranger_out_temp}/{tup.Accession}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_cellranger_scrublet_out = f'{DIR_Chen_cellranger_out_temp}/{tup.Accession}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "        \n",
    "        if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "            bool_cellranger = True\n",
    "            if os.path.exists( PATH_cellranger_scrublet_out ) == True:\n",
    "                bool_scrublet = True\n",
    "            else:\n",
    "                bool_scrublet = False\n",
    "            \n",
    "            if os.path.exists( f'{DIR_Chen_cellranger_out_temp}/{tup.Accession}/outs/possorted_genome_bam.STR_allele_table.tsv' ):\n",
    "                bool_nanomnt = True \n",
    "            else:\n",
    "                bool_nanomnt = False \n",
    "                \n",
    "        else:\n",
    "            bool_cellranger = False\n",
    "            bool_scrublet = False \n",
    "            bool_nanomnt = False\n",
    "        \n",
    "        sample_status_summary.append( [tup.Accession, tup.PatientID, tup.BiopsySite, bool_cellranger, bool_scrublet, bool_nanomnt] )\n",
    "\n",
    "sample_status_summary = pd.DataFrame(sample_status_summary, columns=['Accession', 'PatientID', 'BiopsySite', 'cellranger', 'scrublet', 'nanomnt'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Optional) Run cellranger for samples that have not been processed (create bash scripts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cell Ranger output directory already exists for: Normal-HRR1373904\n"
     ]
    }
   ],
   "source": [
    "PATH_cellranger_bin = '/home/shared/program/cellranger-8.0.1/bin/cellranger'\n",
    "PATH_cellranger_ref = '/home/shared/cellranger_reference/refdata-gex-GRCh38-2020-A'\n",
    "\n",
    "for BiopsySite, edf in sample_status_summary[sample_status_summary['cellranger']==False].groupby(\"BiopsySite\"):\n",
    "    with open(f'/node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/{BiopsySite}.run_cellranger.sh', 'w') as sh:\n",
    "        for tup in edf.itertuples():\n",
    "\n",
    "            if BiopsySite == 'Normal':\n",
    "                DIR_Chen_cellranger_out_temp = f'/node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/{tup.Accession}'\n",
    "                DIR_Chen_FASTQ_temp = f'/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Adjacent_normal_tissue/{tup.Accession}'\n",
    "            elif BiopsySite == 'Tumor':\n",
    "                DIR_Chen_cellranger_out_temp = f'/node200data/18parkky/datasets/data/public/BAM/Immunotherapy_CRC_Chen_et_al/MSI/Tumor/{tup.Accession}'\n",
    "                DIR_Chen_FASTQ_temp = f'/node200data/18parkky/datasets/data/public/FASTQ/Immunotherapy_CRC_Chen_et_al/MSI/Tumor/{tup.Accession}'\n",
    "            else: raise ValueError\n",
    "            \n",
    "            if os.path.exists(DIR_Chen_cellranger_out_temp):\n",
    "                print(f'Cell Ranger output directory already exists for: {BiopsySite}-{tup.Accession}')\n",
    "                continue\n",
    "            if os.path.exists(DIR_Chen_FASTQ_temp) == False:\n",
    "                print(f'Cannot find FASTQ directory for: {BiopsySite}-{tup.Accession}')\n",
    "                continue\n",
    "            elif len( glob.glob(f'{DIR_Chen_FASTQ_temp}/*.fastq.gz') ) != 2:\n",
    "                print(f'Cannot locate 2 FASTQ files for: {BiopsySite}-{tup.Accession}')\n",
    "                continue\n",
    "\n",
    "            cmd = f'echo {tup.Accession}'\n",
    "            sh.write(f'{cmd}\\n')\n",
    "            cmd = f'{PATH_cellranger_bin} count --id {tup.Accession} --transcriptome {PATH_cellranger_ref} --fastqs {DIR_Chen_FASTQ_temp} --create-bam true --output-dir {DIR_Chen_cellranger_out_temp}'\n",
    "            sh.write(f'{cmd}\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Optional) Run Scrublet to samples that have not been processed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/18parkky/anaconda3-2021.11/envs/sc/lib/python3.12/site-packages/anndata/_core/anndata.py:1820: UserWarning: Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"var\")\n",
      "/home/18parkky/anaconda3-2021.11/envs/sc/lib/python3.12/site-packages/anndata/_core/anndata.py:1820: UserWarning: Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"var\")\n"
     ]
    }
   ],
   "source": [
    "for tup in sample_status_summary[(sample_status_summary['cellranger']==True) & (sample_status_summary['scrublet']==False)].itertuples():\n",
    "    if tup.BiopsySite == 'Normal':\n",
    "        PATH_adata_e = f'{DIR_Chen_cellranger_out}/Adjacent_normal_tissue/{tup.Accession}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_scrublet_adata_e = f'{DIR_Chen_cellranger_out}/Adjacent_normal_tissue/{tup.Accession}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "    elif tup.BiopsySite == 'Tumor':\n",
    "        PATH_adata_e = f'{DIR_Chen_cellranger_out}/Tumor/{tup.Accession}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_scrublet_adata_e = f'{DIR_Chen_cellranger_out}/Tumor/{tup.Accession}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "    else:\n",
    "        print(f'Unidentified Biopsy site for {tup.Accession}: {tup.BiopsySite}')\n",
    "        raise ValueError\n",
    "    \n",
    "    if os.path.exists(PATH_adata_e)==False:\n",
    "        print(f'Cell Ranger has been processed for: {tup.Accession}')\n",
    "        raise ValueError\n",
    "    if os.path.exists(PATH_scrublet_adata_e)==True:\n",
    "        print(f'WARNING: Scrublet has been processed for: {tup.Accession}')\n",
    "        print(f'WARNING: Skipping\\n')\n",
    "        continue \n",
    "    else:\n",
    "        adata_e = sc.read_10x_h5( PATH_adata_e )\n",
    "        adata_e.obs_names_make_unique()\n",
    "        adata_e.var_names_make_unique()\n",
    "        sc.pp.scrublet(adata_e)\n",
    "        adata_e.write(PATH_scrublet_adata_e)    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge h5ad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "P04\n",
      "P08\n",
      "P11\n",
      "P12\n",
      "P15\n",
      "P16\n",
      "P17\n",
      "P18\n",
      "P19\n",
      "P20\n",
      "P21\n",
      "P22\n",
      "P23\n",
      "P24\n",
      "P25\n",
      "P26\n"
     ]
    }
   ],
   "source": [
    "adata = list()\n",
    "processed_sample_count = 0\n",
    "unprocessed_sample_count = 0\n",
    "\n",
    "for patientID, edf in chen_metadata.groupby('PatientID'):\n",
    "    print(patientID, flush=True)\n",
    "    for tup in edf.itertuples():\n",
    "    \n",
    "        if tup.BiopsySite == 'Normal':\n",
    "            DIR_Chen_cellranger_out_temp = f'{DIR_Chen_cellranger_out}/Adjacent_normal_tissue'\n",
    "        elif tup.BiopsySite == 'Tumor':\n",
    "            DIR_Chen_cellranger_out_temp = f'{DIR_Chen_cellranger_out}/Tumor'\n",
    "            \n",
    "        PATH_cellranger_h5ad = f'{DIR_Chen_cellranger_out_temp}/{tup.Accession}/outs/filtered_feature_bc_matrix.h5'\n",
    "        PATH_cellranger_scrublet_out = f'{DIR_Chen_cellranger_out_temp}/{tup.Accession}/outs/filtered_feature_bc_matrix.scrublet.h5'\n",
    "        \n",
    "        if os.path.exists( PATH_cellranger_h5ad ) == True:\n",
    "            if os.path.exists( PATH_cellranger_scrublet_out ) == True:  # Add to list\n",
    "                adata_temp = sc.read_h5ad( PATH_cellranger_scrublet_out )\n",
    "                \n",
    "                adata_temp.obs['SampleID']  = tup.Accession\n",
    "                adata_temp.obs['PatientID'] = tup.PatientID\n",
    "                adata_temp.obs['BiopsySite'] = tup.BiopsySite\n",
    "\n",
    "                adata.append( adata_temp ) \n",
    "                pass\n",
    "                \n",
    "            else:   # Run scrublet, save to disk then add to list\n",
    "                adata_temp = sc.read_10x_h5( PATH_cellranger_h5ad )\n",
    "                adata_temp.obs_names_make_unique()\n",
    "                adata_temp.var_names_make_unique()\n",
    "                sc.pp.scrublet(adata_temp)\n",
    "                adata_temp.write(PATH_cellranger_scrublet_out)\n",
    "                adata.append( adata_temp )\n",
    "                pass\n",
    "        else:   # Report\n",
    "            unprocessed_sample_count += 1\n",
    "            print(f'{patientID}-{tup.Accession} has not been processed by CellRanger!')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/18parkky/anaconda3-2021.11/envs/sc/lib/python3.12/site-packages/anndata/_core/anndata.py:1818: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"obs\")\n"
     ]
    }
   ],
   "source": [
    "adata = sc.concat( adata  )\n",
    "adata.obs_names_make_unique()\n",
    "\n",
    "# mitochondrial genes, \"MT-\" for human, \"Mt-\" for mouse\n",
    "adata.var[\"mt\"] = adata.var_names.str.startswith(\"MT-\")\n",
    "# ribosomal genes\n",
    "adata.var[\"ribo\"] = adata.var_names.str.startswith((\"RPS\", \"RPL\"))\n",
    "# hemoglobin genes\n",
    "adata.var[\"hb\"] = adata.var_names.str.contains(\"^HB[^(P)]\")\n",
    "\n",
    "sc.pp.calculate_qc_metrics(\n",
    "    adata, qc_vars=[\"mt\", \"ribo\", \"hb\"], inplace=True, log1p=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.obs['Accession'] = adata.obs['SampleID']\n",
    "adata.write('/node200data/18parkky/datasets/data/public/processed_data/Immunotherapy_CRC_Chen_et_al/1_raw_h5ad/Chen.MSI.raw.h5ad')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "sc",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
